<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Asthma project</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">funcFinemapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Asthma project</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>funcFinemapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210404code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210404code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210404)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree3df804aefbe1c267dc621149c910256f6b78e94ctargetblank3df804aa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/j3gu/funcFinemapping/tree/3df804aefbe1c267dc621149c910256f6b78e94c" target="_blank">3df804a</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree3df804aefbe1c267dc621149c910256f6b78e94ctargetblank3df804aa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/j3gu/funcFinemapping/tree/3df804aefbe1c267dc621149c910256f6b78e94c" target="_blank">3df804a</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/ldsc_results.nb.html
    Ignored:    analysis/results.nb.html
    Ignored:    analysis/snp_finemapping_results.nb.html
    Ignored:    analysis/splicing.nb.html

Untracked files:
    Untracked:  SNPs_categories,png
    Untracked:  SNPs_categories.png
    Untracked:  analysis/asthma_prelim_results.Rmd
    Untracked:  analysis/asthma_results_cp.Rmd
    Untracked:  analysis/enhancer_gene_feature.Rmd
    Untracked:  analysis/feedback.Rmd
    Untracked:  analysis/gene_finemapping_results.Rmd
    Untracked:  analysis/learn_susie.Rmd
    Untracked:  analysis/notes.Rmd
    Untracked:  analysis/snp_finemapping_results.Rmd
    Untracked:  analysis/splicing.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/ldsc_regression.sh
    Untracked:  code/make_plots.R
    Untracked:  code/run_ldsc.sh
    Untracked:  code/run_ldsc_with_bed.sh
    Untracked:  code/run_torus.sh
    Untracked:  code/split_vcf.sh
    Untracked:  data/num_overlaps_finemapped_SNPs_and_ctcf.txt
    Untracked:  data/scz_2018
    Untracked:  data/torus_enrichment_novel_annot.est
    Untracked:  data/torus_joint_enrichment.est
    Untracked:  data/torus_joint_refined_enrichment.est
    Untracked:  enhancer_gene_feature.rmd
    Untracked:  fig1_panels.pdf
    Untracked:  fig2.pdf
    Untracked:  fig_panel2.pdf
    Untracked:  gene_mapping.pdf
    Untracked:  output/background_SNPs_annotated_percent.txt
    Untracked:  panel_figure2.pdf
    Untracked:  test.txt

Unstaged changes:
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/asthma_results.Rmd</code>) and HTML (<code>docs/asthma_results.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/3df804aefbe1c267dc621149c910256f6b78e94c/analysis/asthma_results.Rmd" target="_blank">3df804a</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-04
</td>
<td>
wrote a more detailed documentation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/a52be14865ac86cc5c88e8ca7d1bc6c10ca3e8a4/docs/asthma_results.html" target="_blank">a52be14</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/e5d6f32829e9eb7e66be1e71b4fafb183d1e7ab1/analysis/asthma_results.Rmd" target="_blank">e5d6f32</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-23
</td>
<td>
added percent of SNPs with annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/53a87745f669f007b900afd4a5e34ebb051d14ab/docs/asthma_results.html" target="_blank">53a8774</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/2a6fc915e159af6504e1abdfdf30bd82b4ab4f75/docs/asthma_results.html" target="_blank">2a6fc91</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/bbb294489e21098fa679d51c82140965e185403d/analysis/asthma_results.Rmd" target="_blank">bbb2944</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
analyzed Asthma GWAS with functional priors from single cell data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/41f53c83d18a13da1b50ff9004b3abb011b9fdb1/docs/asthma_results.html" target="_blank">41f53c8</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/33d4e5af7c75ba3f7625c31c034f04d4e678a09c/analysis/asthma_results.Rmd" target="_blank">33d4e5a</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
analyzed Asthma GWAS with functional priors from single cell data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/b0b938bcb9c43790bb7bb0aa211ad03429e47934/docs/asthma_results.html" target="_blank">b0b938b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/8a5656cf110594c30de185f7ba5d12cc21a41e08/analysis/asthma_results.Rmd" target="_blank">8a5656c</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
analyzed Asthma GWAS with functional priors from single cell data
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="overall-goal" class="section level2">
<h2>Overall Goal</h2>
<p>Identify disease-relevant cell types and genes for Asthma</p>
</div>
<div id="enrichment-estimates-for-individual-annotation" class="section level2">
<h2>Enrichment estimates for individual annotation</h2>
<div id="adult-onset-asthma" class="section level3">
<h3>Adult-onset asthma</h3>
<ul>
<li>GWAS: <em>Zhu et al. 2019</em> (case N=22296, control N=347481)<br />
</li>
<li>Annotations: <em>Zhang et al.2021</em> - single-cell ATAC-Seq data for adult lungs
<ul>
<li>A union set of cCRE accessibility across cell types was provided in the following format: chr:start-end cell_type1 cell_type2...</li>
<li>Entropy based method used to identify cell-type specific peaks</li>
</ul></li>
</ul>
<p>As neither the number of peaks identified for each cell type nor the list of cell-type specific peaks was available, I followed their way of identifying cell-type restricted peaks. First, I took a subset of accessibility scores for lung cells only. Then I modeled the null distribution of the log2-based cell-type dependent fold changes from the avg level using a normal distribution, with mean 0 and standard deviation estimated by that of top 50% least variable peaks. By sampling from the null distribution, we can compute p-values for each site and for each cell type. A 0.1% FDR cutoff was set to decide which peaks are considered to be cell-type specific.</p>
<div id="cell-compositions-for-adult-lung-tissues" class="section level4">
<h4>Cell compositions for adult lung tissues</h4>
<p><img src="figure/asthma_results.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/b0b938bcb9c43790bb7bb0aa211ad03429e47934/docs/figure/asthma_results.Rmd/unnamed-chunk-1-1.png" target="_blank">b0b938b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="peak-summary-for-adult-lung-tissues" class="section level4">
<h4>peak summary for adult lung tissues</h4>
<p><img src="figure/asthma_results.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a52be14865ac86cc5c88e8ca7d1bc6c10ca3e8a4/docs/figure/asthma_results.Rmd/unnamed-chunk-2-1.png" target="_blank">a52be14</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/b0b938bcb9c43790bb7bb0aa211ad03429e47934/docs/figure/asthma_results.Rmd/unnamed-chunk-2-1.png" target="_blank">b0b938b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Check overlaps in peaks across cell types</strong><br />
We want to know to what extent are the cell-type restricted peaks across cell types overlapped with each other. The table shows the majority of the identified cell types overlap with each other by less than 15%. As Msc has the highest amount of peaks, it's expected the most overlapped cell-type is Msc for almost all other cell types. We also see high percent of overlaps between subclusters of one cell type, such as fibroblast, endothelial cells. Lastly, B cells share 9% of peaks with T-lymphocyte subcluster 1.</p>
<table>
<caption>Table 1-Summary of peak overlaps between a pair of cell types</caption>
<colgroup>
<col width="6%" />
<col width="18%" />
<col width="22%" />
<col width="24%" />
<col width="28%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">most_overlapped_per</th>
<th align="left">most_overlapped_celltype</th>
<th align="right">second_most_overlapped_per</th>
<th align="left">second_most_overlapped_celltype</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Agb</td>
<td align="right">0.12</td>
<td align="left">Msc</td>
<td align="right">0.08</td>
<td align="left">Mac.2</td>
</tr>
<tr class="even">
<td align="left">Bly</td>
<td align="right">0.10</td>
<td align="left">Msc</td>
<td align="right">0.07</td>
<td align="left">Mac.1</td>
</tr>
<tr class="odd">
<td align="left">End.1</td>
<td align="right">0.16</td>
<td align="left">End.2</td>
<td align="right">0.14</td>
<td align="left">Msc</td>
</tr>
<tr class="even">
<td align="left">End.2</td>
<td align="right">0.15</td>
<td align="left">Msc</td>
<td align="right">0.12</td>
<td align="left">Mac.2</td>
</tr>
<tr class="odd">
<td align="left">Fib.1</td>
<td align="right">0.15</td>
<td align="left">Msc</td>
<td align="right">0.14</td>
<td align="left">Fib.2</td>
</tr>
<tr class="even">
<td align="left">Fib.2</td>
<td align="right">0.18</td>
<td align="left">Fib.1</td>
<td align="right">0.15</td>
<td align="left">Fib.4</td>
</tr>
<tr class="odd">
<td align="left">Fib.3</td>
<td align="right">0.16</td>
<td align="left">Msc</td>
<td align="right">0.12</td>
<td align="left">Fib.1</td>
</tr>
<tr class="even">
<td align="left">Fib.4</td>
<td align="right">0.13</td>
<td align="left">Msc</td>
<td align="right">0.13</td>
<td align="left">Fib.1</td>
</tr>
<tr class="odd">
<td align="left">Gbl.2</td>
<td align="right">0.12</td>
<td align="left">Msc</td>
<td align="right">0.07</td>
<td align="left">Mac.2</td>
</tr>
<tr class="even">
<td align="left">Grn</td>
<td align="right">0.09</td>
<td align="left">Fib.2</td>
<td align="right">0.09</td>
<td align="left">Fib.1</td>
</tr>
<tr class="odd">
<td align="left">Mac.1</td>
<td align="right">0.12</td>
<td align="left">Msc</td>
<td align="right">0.07</td>
<td align="left">Fib.4</td>
</tr>
<tr class="even">
<td align="left">Mac.2</td>
<td align="right">0.16</td>
<td align="left">Msc</td>
<td align="right">0.06</td>
<td align="left">End.2</td>
</tr>
<tr class="odd">
<td align="left">Mes</td>
<td align="right">0.15</td>
<td align="left">Msc</td>
<td align="right">0.09</td>
<td align="left">Mac.2</td>
</tr>
<tr class="even">
<td align="left">Mfb.2</td>
<td align="right">0.16</td>
<td align="left">Msc</td>
<td align="right">0.10</td>
<td align="left">Mac.2</td>
</tr>
<tr class="odd">
<td align="left">Msc</td>
<td align="right">0.06</td>
<td align="left">Mac.2</td>
<td align="right">0.03</td>
<td align="left">Fib.3</td>
</tr>
<tr class="even">
<td align="left">Mst</td>
<td align="right">0.10</td>
<td align="left">Msc</td>
<td align="right">0.09</td>
<td align="left">Mac.1</td>
</tr>
<tr class="odd">
<td align="left">Pal</td>
<td align="right">0.15</td>
<td align="left">Msc</td>
<td align="right">0.10</td>
<td align="left">Mac.2</td>
</tr>
<tr class="even">
<td align="left">Smm.1</td>
<td align="right">0.14</td>
<td align="left">Msc</td>
<td align="right">0.12</td>
<td align="left">Fib.1</td>
</tr>
<tr class="odd">
<td align="left">Tly.1</td>
<td align="right">0.09</td>
<td align="left">Msc</td>
<td align="right">0.09</td>
<td align="left">Bly</td>
</tr>
<tr class="even">
<td align="left">Tly.2</td>
<td align="right">0.12</td>
<td align="left">Msc</td>
<td align="right">0.12</td>
<td align="left">Mac.2</td>
</tr>
<tr class="odd">
<td align="left">Vsm.2</td>
<td align="right">0.15</td>
<td align="left">Msc</td>
<td align="right">0.13</td>
<td align="left">Fib.1</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="enrichment-results" class="section level3">
<h3>Enrichment results</h3>
<ul>
<li>Tly.1(T_lymphocyte1) and Tly.2(T_lymphocyte2) cannot be distinguished simply based on known immune marker genes. Both of them show differential expression among genes such as CD3D, GZMA, IFNG. <img src="figure/asthma_results.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></li>
</ul>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a52be14865ac86cc5c88e8ca7d1bc6c10ca3e8a4/docs/figure/asthma_results.Rmd/unnamed-chunk-4-1.png" target="_blank">a52be14</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/b0b938bcb9c43790bb7bb0aa211ad03429e47934/docs/figure/asthma_results.Rmd/unnamed-chunk-4-1.png" target="_blank">b0b938b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Log2 scaled enrichment estimates for each cell type, which was individually run using Torus. The highlighted bars show significant enrichment, while the bars in grey color do not. The x-axis tickmark labels consist of the annotation name and percent of all SNPs that have that annotation. Asthma-associated variants are significantly enriched in certain sub-clusters of macrophages and fibroblasts, but not in others. The cell types that show the highest enrichment in the chromatin accessibility peaks are immune-related cells.</p>
</div>
<div id="children-onset-asthma" class="section level3">
<h3>Children-onset Asthma</h3>
<ul>
<li>GWAS: <em>Zhu et al. 2019</em><br />
</li>
<li>cell-type specific Annotations: <em>Domcke et al.2020</em> - single-cell ATAC-Seq data for fetal lungs</li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.4 (2021-02-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] knitr_1.31      ggplot2_3.3.3   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7        highr_0.8         pillar_1.5.0      compiler_4.0.4   
 [5] bslib_0.2.4       later_1.1.0.1     jquerylib_0.1.3   git2r_0.28.0     
 [9] tools_4.0.4       digest_0.6.27     jsonlite_1.7.2    evaluate_0.14    
[13] lifecycle_1.0.0   tibble_3.0.6      gtable_0.3.0      pkgconfig_2.0.3  
[17] rlang_0.4.11      DBI_1.1.1         yaml_2.2.1        xfun_0.21        
[21] withr_2.4.2       dplyr_1.0.4       stringr_1.4.0     generics_0.1.0   
[25] fs_1.5.0          vctrs_0.3.8       sass_0.3.1        tidyselect_1.1.1 
[29] rprojroot_2.0.2   grid_4.0.4        glue_1.4.2        R6_2.5.1         
[33] fansi_0.5.0       rmarkdown_2.7     farver_2.1.0      purrr_0.3.4      
[37] magrittr_2.0.1    whisker_0.4       scales_1.1.1      promises_1.2.0.1 
[41] ellipsis_0.3.2    htmltools_0.5.1.1 assertthat_0.2.1  colorspace_2.0-2 
[45] httpuv_1.5.5      labeling_0.4.2    utf8_1.2.2        stringi_1.5.3    
[49] munsell_0.5.0     crayon_1.4.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
