<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>asthma_prelim_results</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">funcFinemapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">asthma_prelim_results</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-11-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>funcFinemapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210404code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210404code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210404)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree093135b88dccaa3255cddcc23176f3caea927f00targetblank093135ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/j3gu/funcFinemapping/tree/093135b88dccaa3255cddcc23176f3caea927f00" target="_blank">093135b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree093135b88dccaa3255cddcc23176f3caea927f00targetblank093135ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/j3gu/funcFinemapping/tree/093135b88dccaa3255cddcc23176f3caea927f00" target="_blank">093135b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/ldsc_results.nb.html
    Ignored:    analysis/mtsplice_finemapping_results.nb.html
    Ignored:    analysis/results.nb.html
    Ignored:    analysis/snp_finemapping_results.nb.html
    Ignored:    analysis/splicing.nb.html

Untracked files:
    Untracked:  SNPs_categories,png
    Untracked:  SNPs_categories.png
    Untracked:  analysis/asthma_prelim_results_archive.Rmd
    Untracked:  analysis/asthma_prelim_results_cp.Rmd
    Untracked:  analysis/asthma_results_cp.Rmd
    Untracked:  analysis/enhancer_gene_feature.Rmd
    Untracked:  analysis/feedback.Rmd
    Untracked:  analysis/gene_finemapping_results.Rmd
    Untracked:  analysis/learn_susie.Rmd
    Untracked:  analysis/mtsplice_finemapping_results.Rmd
    Untracked:  analysis/notes.Rmd
    Untracked:  analysis/snp_finemapping_results.Rmd
    Untracked:  analysis/splicing.Rmd
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/ldsc_regression.sh
    Untracked:  code/make_plots.R
    Untracked:  code/run_ldsc.sh
    Untracked:  code/run_ldsc_with_bed.sh
    Untracked:  code/run_susie.R
    Untracked:  code/run_torus.sh
    Untracked:  code/split_vcf.sh
    Untracked:  data/num_overlaps_finemapped_SNPs_and_ctcf.txt
    Untracked:  data/scz_2018
    Untracked:  data/torus_enrichment_novel_annot.est
    Untracked:  data/torus_joint_enrichment.est
    Untracked:  data/torus_joint_refined_enrichment.est
    Untracked:  enhancer_gene_feature.rmd
    Untracked:  fig1_panels.pdf
    Untracked:  fig2.pdf
    Untracked:  fig_panel2.pdf
    Untracked:  gene_mapping.pdf
    Untracked:  output/background_SNPs_annotated_percent.txt
    Untracked:  panel_figure2.pdf
    Untracked:  test.txt

Unstaged changes:
    Modified:   analysis/asthma_results.Rmd
    Modified:   analysis/enrichment_analysis.Rmd
    Deleted:    output/AAD/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/AAD/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/AAD/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/AAD/asthma/diffe_adult_blood_torus.est
    Deleted:    output/AAD/asthma/joint_blood_immune_rest_vs_stimu.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/AAD/asthma/lung_clusters_dict.txt
    Deleted:    output/AAD/asthma/lung_clusters_info.txt
    Deleted:    output/AAD/asthma/stimu_adult_blood_torus.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_rest.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_stimulated.est
    Deleted:    output/AAD/asthma/zhang2021_annot_percent.txt
    Deleted:    output/AAD/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/AAD/asthma/zhang2021_peaks_per_celltype.txt
    Deleted:    output/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/asthma/diffe_adult_blood_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/asthma/lung_clusters_dict.txt
    Deleted:    output/asthma/lung_clusters_info.txt
    Deleted:    output/asthma/stimu_adult_blood_torus.est
    Deleted:    output/asthma/zhang2021_annot_percent.txt
    Deleted:    output/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/asthma/zhang2021_peaks_per_celltype.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/asthma_prelim_results.Rmd</code>) and HTML (<code>docs/asthma_prelim_results.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/093135b88dccaa3255cddcc23176f3caea927f00/analysis/asthma_prelim_results.Rmd" target="_blank">093135b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-17
</td>
<td>
estimate heritability
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/6ce4a8dc4fbaa9cf01a6597a28bfb254b21e9813/docs/asthma_prelim_results.html" target="_blank">6ce4a8d</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/0c42feed06f1cea9a7904334c1fc9624bb518e5f/analysis/asthma_prelim_results.Rmd" target="_blank">0c42fee</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
<td>
compare with lungs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/asthma_prelim_results.html" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/08ab4ff3a0f0b16d0509fea08bc387df10beba82/analysis/asthma_prelim_results.Rmd" target="_blank">08ab4ff</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
<td>
compare with lungs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/1bd2749c15dd518e7a023cc8680591adae591b77/docs/asthma_prelim_results.html" target="_blank">1bd2749</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/f375e53ce88639468e7b0c0584d6278c965fddf8/analysis/asthma_prelim_results.Rmd" target="_blank">f375e53</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-12
</td>
<td>
compare lung and blood
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/63b34726a622cc6fe7c4beac250dde014d18c026/docs/asthma_prelim_results.html" target="_blank">63b3472</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/912c9b64e9cce23e7d5d524213345c1ad563d88f/analysis/asthma_prelim_results.Rmd" target="_blank">912c9b6</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
<td>
analyzed lung immune cells
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/be4deded7c0d75d11b8304b1a05b6b719185cb55/docs/asthma_prelim_results.html" target="_blank">be4dede</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/dd720fe0861c0b141d2684c4500183ea9f2edcc8/analysis/asthma_prelim_results.Rmd" target="_blank">dd720fe</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-04
</td>
<td>
enrichment analysis for AAD traits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/8a43e8f8e060a960ba68d1c66572cca663872233/docs/asthma_prelim_results.html" target="_blank">8a43e8f</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/1892d30ab75bd608ccc0d393c00beacbb29fe3c8/analysis/asthma_prelim_results.Rmd" target="_blank">1892d30</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
<td>
wflow_publish(&quot;analysis/asthma_prelim_results.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/29453697bc84ef5935ad3ba843e6b8df8f6673aa/docs/asthma_prelim_results.html" target="_blank">2945369</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/b3259c5a06750b948cdb118eb6c9371b32b1dc3a/analysis/asthma_prelim_results.Rmd" target="_blank">b3259c5</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
<td>
wflow_publish(&quot;analysis/asthma_prelim_results.Rmd&quot;)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/5b65bd10f715e7c6d2f8dffdcac3cfb13feac2cf/docs/asthma_prelim_results.html" target="_blank">5b65bd1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/db3c6f96af3bb6ed4d4e210f7ac5585517152b8a/analysis/asthma_prelim_results.Rmd" target="_blank">db3c6f9</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
<td>
test enrichment for asthma risk variants
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="asthama-and-allergy-diseases" class="section level2">
<h2>Asthama and Allergy diseases</h2>
<p>The main goal is to identify causal variants, genes and cell types relevant to AAD by integrating omics data of lung samples. We hypothesize that open chromatin regions of lung-resident immune cells can explain broader heritability for AAD than those of blood immune cells. The disease associated variants annotated by these regulatory functions are more likely to contribute to disease risk, so this prior knowledge can be leveraged to prioritize risk variants in GWAS loci.</p>
</div>
<div id="dataset" class="section level2">
<h2>Dataset</h2>
<div id="atac-seq-data-for-blood-immune-cells-caldero2019" class="section level3">
<h3>ATAC-seq data for Blood immune cells (Caldero2019)</h3>
<p><strong>Dataset</strong>: ATAC-Seq profiles for FACS-sorted cells from the peripheral blood of up to 4 healthy donors</p>
<p><strong>Output files</strong>:<br />
1. An ATAC-seq count table with a union set of peaks as rows and individual cell as columns<br />
2. A Sample QC table that includes number of peaks and cell type identity for each cell<br />
3. Significant differentially accessible regions when compared to progenitor cells<br />
4. Significant differentially accessible regions under stimulation</p>
<p><strong>Procedure</strong>: I used the number of peaks after QC for each cell to extract its corresponding ATAC-Seq peaks from the count table. <details> <summary>Details of the procedure</summary> For the cells of interest, I sorted their corresponding columns of the count table and store the top N number of peaks according to the number of peaks from the QC table as cell type resolved peaks. </details></p>
</div>
<div id="atac-seq-data-for-hematopoietic-cells-ulirsch2019" class="section level3">
<h3>ATAC-seq data for hematopoietic cells (Ulirsch2019)</h3>
<p><strong>Dataset</strong>: ATAC-Seq profiles for FACS-sorted cells from human peripheral blood or bone marrow.</p>
<p><strong>Output files</strong>:<br />
1.peak files downloaded from: <a href="https://github.com/caleblareau/singlecell_bloodtraits/tree/master/data/bulk/ATAC/narrowpeaks" class="uri">https://github.com/caleblareau/singlecell_bloodtraits/tree/master/data/bulk/ATAC/narrowpeaks</a></p>
</div>
<div id="scatac-seq-data-for-lung-tissues-wang2020" class="section level3">
<h3>scATAC-seq data for lung tissues (Wang2020)</h3>
<p><strong>Dataset</strong>: scATAC-Seq and scRNA-seq profiles for small airway region of right middle lobe (RML) lung tissue from 3 donors at different ages</p>
<p><strong>Output files</strong>:<br />
1. peak files downloaded from web portal: <a href="https://www.lungepigenome.org/" class="uri">https://www.lungepigenome.org/</a></p>
</div>
<div id="scatac-seq-data-for-fetal-hematopoietic-cells-ronzoni2021" class="section level3">
<h3>scATAC-seq data for fetal hematopoietic cells (Ronzoni2021)</h3>
<p><strong>Dataset</strong>: scATAC-Seq and scRNA-seq profiles of human immunophenotypic blood cells from fetal liver and bone marrow</p>
<p><strong>Output files</strong>:<br />
Downloaded from gitlab page: to be added<br />
1. A merged normalized peak table with a union set of peaks as rows and individual cell as columns<br />
2. A meta table that includes number of peaks and predicted cell type identity for each cell<br />
3. A raw count table with peaks as rows and individual cell as columns</p>
<p><strong>Procedure</strong>: I used the number of peaks for each cell from the meta table to extract its corresponding ATAC-Seq peaks from the merged peak table. The number of peaks is equivalent to the amount of non-zero peaks.</p>
</div>
</div>
<div id="torus-run-for-individual-blood-annotation" class="section level2">
<h2>TORUS run for individual Blood annotation</h2>
<p><strong>Motivation</strong>:</p>
<ol style="list-style-type: decimal">
<li>perform QC check for the Caldero2019 dataset</li>
<li>have a senes of which cell types are potentially relevant to AAD</li>
</ol>
<p><strong>Results</strong>: Overall, the magnitude of enrichment estimates are much smaller than that in figure 5b. The discrepancy can be due to different GWAS datasets and enrichment method. They used LDSC to estimate enrichment coefficients, which have a better control on the overlapping peaks between many annotations. <img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-1-1.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/be4deded7c0d75d11b8304b1a05b6b719185cb55/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-1-1.png" target="_blank">be4dede</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/5b65bd10f715e7c6d2f8dffdcac3cfb13feac2cf/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-1-1.png" target="_blank">5b65bd1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="torus-run-for-individual-lung-annotation" class="section level2">
<h2>TORUS run for individual lung annotation</h2>
<p><strong>Motivation</strong>:</p>
<ol style="list-style-type: decimal">
<li>perform QC check for this dataset</li>
<li>have a senes of which cell types in lungs are potentially relevant to AAD</li>
</ol>
<p><strong>Results</strong>: Data look good as the open chromatin regions of most cell types in lungs are enriched with disease risk variants. <img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-2-1.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/63b34726a622cc6fe7c4beac250dde014d18c026/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-2-1.png" target="_blank">63b3472</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/be4deded7c0d75d11b8304b1a05b6b719185cb55/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-2-1.png" target="_blank">be4dede</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/5b65bd10f715e7c6d2f8dffdcac3cfb13feac2cf/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-2-1.png" target="_blank">5b65bd1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="torus-joint-run-for-lung-vs-blood-immune-cells" class="section level2">
<h2>TORUS joint run for lung vs blood immune cells</h2>
<p><strong>Motivation</strong>:<br />
Test the hypothesis that open chromatin regions of lung-resident immune cells can explain broader heritability for AAD than those of blood immune cells.</p>
<p><strong>Procedure</strong>: For each immune group (B cells, T cells, Myeloids, NK cells), I ran TORUS over pairs of annotations from lung and blood one at a time. Pairs of annotations as follows:</p>
<ul>
<li>B cells (lung) vs B cells (blood)</li>
<li>T cells (lung) vs CD4+ T (blood)</li>
<li>T cells (lung) vs CD8+ T (blood)</li>
<li>NK cells (lung) vs NK cells (blood)</li>
<li>Fibroblast cells (lung) vs Myeloids (blood)</li>
</ul>
<p><strong>Results</strong>: Overall, we see lung-resident immune cells show significant enrichemnt conditional on blood immune cells in AAD, but not in RA. Compared with a control set of traits, there is a large difference between blood and lung enrichment estimates for Myeloid, CD4+ and CD8+ T cells in AAD.</p>
<div id="caldero2019-dataset" class="section level3">
<h3>Caldero2019 dataset</h3>
<p><img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-3-1.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/1bd2749c15dd518e7a023cc8680591adae591b77/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-3-1.png" target="_blank">1bd2749</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/63b34726a622cc6fe7c4beac250dde014d18c026/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-3-1.png" target="_blank">63b3472</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/be4deded7c0d75d11b8304b1a05b6b719185cb55/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-3-1.png" target="_blank">be4dede</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/5b65bd10f715e7c6d2f8dffdcac3cfb13feac2cf/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-3-1.png" target="_blank">5b65bd1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-3-2.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ulirsch2019-dataset" class="section level3">
<h3>Ulirsch2019 dataset</h3>
<p><strong>Motivation</strong>: Replicate the above result in a different blood dataset</p>
<p><strong>Results</strong>:<br />
Overall, the enrichment of lung annotations dwindles in a great extent when jointly run with blood annotations. There is one exception that CD8+ T cells in lungs show significant enrichment with child-onset asthma.</p>
<p><img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-4-1.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/63b34726a622cc6fe7c4beac250dde014d18c026/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-4-1.png" target="_blank">63b3472</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/be4deded7c0d75d11b8304b1a05b6b719185cb55/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-4-1.png" target="_blank">be4dede</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/5b65bd10f715e7c6d2f8dffdcac3cfb13feac2cf/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-4-1.png" target="_blank">5b65bd1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-2">
Past versions of unnamed-chunk-4-2.png
</button>
</p>
<div id="fig-unnamed-chunk-4-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-4-2.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="joint-torus-run-for-annotation-sets" class="section level2">
<h2>Joint TORUS run for annotation sets</h2>
<div id="major-clusters-in-lung-dataset" class="section level3">
<h3>major clusters in lung dataset</h3>
<ul>
<li>Mesenchymal cells</li>
<li>Immune cells</li>
<li>epithelial cells</li>
<li>endothelial cells</li>
<li>others(neuroendocrine cells, type I pneumocyte, ciliated cells)</li>
</ul>
<p><strong>Motivation</strong>: determine how each cluster with similar chromatin accessibility patterns contribute to AAD heritability.</p>
<p><strong>Results</strong>:<br />
We observed some non-immune cell types such as epithelial cells, endothelia cells do contribute to genetic risks of AAD, though not as much as immune cells. <img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-5-1.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/63b34726a622cc6fe7c4beac250dde014d18c026/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-5-1.png" target="_blank">63b3472</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/be4deded7c0d75d11b8304b1a05b6b719185cb55/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-5-1.png" target="_blank">be4dede</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-04
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/8a43e8f8e060a960ba68d1c66572cca663872233/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-5-1.png" target="_blank">8a43e8f</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/5b65bd10f715e7c6d2f8dffdcac3cfb13feac2cf/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-5-1.png" target="_blank">5b65bd1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-5-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-5-2.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="annotation-sets-for-caldero2019" class="section level3">
<h3>annotation sets for Caldero2019</h3>
<div id="grouped-by-lineages" class="section level4">
<h4>grouped by lineages</h4>
<p><strong>Motivation</strong>: Running annotation sets in a joint model enables us to idenity the relevant contribution in open chromatin regions of each immune cell type to traits. LSDC is a better tool to use as we expected many overlaps of acccessible peaks across the sub-clusters of immune cells, which can make the estimation of TORUS's joint model unstable. Here we grouped the cell types and used TORUS to get a quick run of which group of cell types have significant enrichment.</p>
<p><strong>Procedure</strong>:<br />
For Caldero2019 dataset: Immune cells were grouped into six main categories and merged across two conditions. The peaks in these groups can be overlapped.<br />
For Ronzoni2021 dataset, I took a union set of peaks from all cells that were predicted to be granulocytes progenitors.</p>
<p><strong>Results</strong>:<br />
Consistent with prior knowledge, we see enrichment of granulocyte progenitors with genetic risk of allergy. <img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-6-1.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/63b34726a622cc6fe7c4beac250dde014d18c026/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-6-1.png" target="_blank">63b3472</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/be4deded7c0d75d11b8304b1a05b6b719185cb55/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-6-1.png" target="_blank">be4dede</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Summary</strong>:</p>
<ol style="list-style-type: decimal">
<li>Overall, we see the differences in immune cell components that contribute to these three autoimmune diseases.</li>
<li>GPs and CD4+ T cels are consistently significant in enrichment of risk variants for all three diseases.</li>
</ol>
</div>
<div id="grouped-by-lineages-and-conditions" class="section level4">
<h4>grouped by lineages and conditions</h4>
<p><strong>Procedure</strong>: Immune cells were grouped first into six main categories and then separated by conditions. This set of 12 annotations were jointly tested via TORUS. <img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-7-1.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/1bd2749c15dd518e7a023cc8680591adae591b77/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-7-1.png" target="_blank">1bd2749</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/63b34726a622cc6fe7c4beac250dde014d18c026/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-7-1.png" target="_blank">63b3472</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="grouped-by-disjoint-peaks" class="section level4">
<h4>grouped by disjoint peaks</h4>
<p><strong>Motivation</strong>: Using disjoint groups of peaks from immune cell types to estimate separate contributions of immune components to disease heritability. These annotations are ideal predictors for the linear model used in either LDSC or TORUS to obtain unbiased enrichment estimators.</p>
<p><strong>Procedure</strong>: The disjoint peaks were directly downloaded from the paper. <img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-8-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a6ca0c73567b2d1b0a44ca905bdae63dec2ef442/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-8-1.png" target="_blank">a6ca0c7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-16
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/1bd2749c15dd518e7a023cc8680591adae591b77/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-8-1.png" target="_blank">1bd2749</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-12
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/63b34726a622cc6fe7c4beac250dde014d18c026/docs/figure/asthma_prelim_results.Rmd/unnamed-chunk-8-1.png" target="_blank">63b3472</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>explained Heritability</strong></p>
</div>
</div>
<div id="disjoint-immune-peaks-in-blood-vs.-immune-groups-in-lung" class="section level3">
<h3>Disjoint immune peaks in blood vs. immune groups in lung</h3>
<p><strong>Motivation</strong>:<br />
We will do a joint analysis on three disjoint peak sets - only in blood, only in lung and shared. In this way, we can both estimate enrichments and compute for percent of genetic signals explained by each set.</p>
<p><strong>Procedure</strong>:<br />
I started from the cell types defined with disjoint sets of immune peaks from Caldero2019 dataset. For each cell type, I use <code>bedtools intersect</code> to call the overlapped peaks and peaks that are unique to either side for lung and blood peaks. Then, the overlapped peaks will be merged.</p>
<p><img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /> <strong>Heritability estimation</strong> <img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/asthma_prelim_results.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<!-- ### [Archive] Disjoint immune peaks in blood vs. immune groups in lung -->
<!-- **Motivation**:  -->
<!-- There is a large fraction of overlaps in peaks across blood's immune groups. Thus we used disjoint peaks to build blood annotations and then ran TORUS together with main immune groups in lung to better estimate their enrichment separately.  -->
<!-- **Procedure**:   -->
<!-- I merged peaks from three groups of T-cells, which are T-stimulated, T-resting and BandT_stimulated cells. Same process for B cells. Then I ran TORUS on merged T cells, B cells, NK resting cells, myeloid resting cells from Caldero2019 with main immune groups in lungs.  -->
<!-- **Results**:     -->
<!-- NK disjoint peaks in blood was not displayed due to very large standard error. We can still see there is  difference in T cells between blood and lung and the extent of difference varies across traits. Other immune cell types in blood do not show significance probably due to small number of disjoint peaks. Since the peaks in lungs are not disjoint, it is expected to see a bias towards lung. Nonetheless, compared with controls, we observe a larger difference in enrichment of risk variants in T cells for AAD.  -->
<!-- ```{r echo=F, fig.width=12} -->
<!-- tbl<-data.frame() -->
<!-- for(trait in traits){ -->
<!--   df<-read.table(sprintf("output/AAD/%s/Caldero2019_disjoint_compare.est", trait), header=T, skip=1) -->
<!--   colnames(df)<-c("raw_term", "estimate", "low", "high") -->
<!--   df["term"] = unlist(lapply(df$raw_term, function(i){strsplit(i[1], "[.]")[[1]][1]})) -->
<!--   tbl<-data.frame(rbind(tbl, cbind(df, trait))) -->
<!-- } -->
<!-- tbl<-tbl[tbl$term!="NK_disjoint_compare_blood",] -->
<!-- snp_enrichment_plot(tbl[tbl$trait %in% test,], trait="",  -->
<!--                     y.label = c("B_cell_blood", "B_cell_lung", "Myeloid_blood", -->
<!--                                 "Myeloid_lung","NK_cell_lung", "T_cell_blood",  -->
<!--                                 "T_cell_lung" -->
<!--                                 )) + facet_grid(. ~ trait, scale="free_x") -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.4 (2021-02-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.0.4     knitr_1.31      ggplot2_3.3.3   workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7        highr_0.8         pillar_1.5.0      compiler_4.0.4   
 [5] bslib_0.2.4       later_1.1.0.1     jquerylib_0.1.3   git2r_0.28.0     
 [9] tools_4.0.4       digest_0.6.27     jsonlite_1.7.2    evaluate_0.14    
[13] lifecycle_1.0.0   tibble_3.0.6      gtable_0.3.0      pkgconfig_2.0.3  
[17] rlang_0.4.11      DBI_1.1.1         yaml_2.2.1        xfun_0.21        
[21] withr_2.4.2       stringr_1.4.0     generics_0.1.0    fs_1.5.0         
[25] vctrs_0.3.8       sass_0.3.1        tidyselect_1.1.1  rprojroot_2.0.2  
[29] grid_4.0.4        glue_1.4.2        R6_2.5.1          fansi_0.5.0      
[33] rmarkdown_2.7     farver_2.1.0      purrr_0.3.4       magrittr_2.0.1   
[37] whisker_0.4       scales_1.1.1      promises_1.2.0.1  ellipsis_0.3.2   
[41] htmltools_0.5.1.1 assertthat_0.2.1  colorspace_2.0-2  httpuv_1.5.5     
[45] labeling_0.4.2    utf8_1.2.2        stringi_1.5.3     munsell_0.5.0    
[49] crayon_1.4.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
