<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Fine-mapping Pipeline</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">funcFinemapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Fine-mapping Pipeline</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 1 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>funcFinemapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you'll want to first commit it to the Git repo. If you're still working on the analysis, you can ignore this warning. When you're finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree1da45e4498f93d511e9048cef7818ad3525c8890targetblank1da45e4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/j3gu/funcFinemapping/tree/1da45e4498f93d511e9048cef7818ad3525c8890" target="_blank">1da45e4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree1da45e4498f93d511e9048cef7818ad3525c8890targetblank1da45e4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/j3gu/funcFinemapping/tree/1da45e4498f93d511e9048cef7818ad3525c8890" target="_blank">1da45e4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/build_annotations_for_single_cell_data.nb.html
    Ignored:    analysis/lab4_prepare.nb.html
    Ignored:    analysis/ldsc_results.nb.html
    Ignored:    analysis/learn_archR.nb.html
    Ignored:    analysis/mtsplice_finemapping_results.nb.html
    Ignored:    analysis/results.nb.html
    Ignored:    analysis/snp_finemapping_results.nb.html
    Ignored:    analysis/splicing.nb.html
    Ignored:    analysis/susie_tutorial.nb.html

Untracked files:
    Untracked:  SNPs_categories,png
    Untracked:  SNPs_categories.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/annotations_PTR.Rmd
    Untracked:  bmi_locus1410.pdf
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/.snakemake/
    Untracked:  code/Euro_LD_Chunks.RData
    Untracked:  code/Snakefile
    Untracked:  code/config.yaml
    Untracked:  code/environment.yml
    Untracked:  code/ldsc.log
    Untracked:  code/ldsc.results
    Untracked:  code/ldsc_regression.sh
    Untracked:  code/make_plots.R
    Untracked:  code/run_ldsc.sh
    Untracked:  code/run_ldsc_with_bed.sh
    Untracked:  code/run_ldsc_with_bed_v2.sh
    Untracked:  code/run_susie.R
    Untracked:  code/run_torus.sh
    Untracked:  code/sctype/
    Untracked:  code/slurm-20815237.out
    Untracked:  code/slurm-20832976.out
    Untracked:  code/slurm-20832977.out
    Untracked:  code/slurm-20832978.out
    Untracked:  code/slurm-20832979.out
    Untracked:  code/slurm-20832980.out
    Untracked:  code/slurm-21213166.out
    Untracked:  code/slurm-21213167.out
    Untracked:  code/slurm-21213168.out
    Untracked:  code/slurm-21213169.out
    Untracked:  code/slurm-21213170.out
    Untracked:  code/slurm-21213181.out
    Untracked:  code/slurm-21213183.out
    Untracked:  code/slurm-21213184.out
    Untracked:  code/slurm-21213188.out
    Untracked:  code/slurm-21213195.out
    Untracked:  code/slurm-21213200.out
    Untracked:  code/slurm-21213204.out
    Untracked:  code/slurm-21213207.out
    Untracked:  code/slurm-21213212.out
    Untracked:  code/slurm-21213217.out
    Untracked:  code/slurm-21213224.out
    Untracked:  code/slurm-21213230.out
    Untracked:  code/slurm-21213231.out
    Untracked:  code/slurm-21213236.out
    Untracked:  code/slurm-21213240.out
    Untracked:  code/slurm-21213244.out
    Untracked:  code/slurm-21213246.out
    Untracked:  code/slurm-21213601.out
    Untracked:  code/slurm-21213602.out
    Untracked:  code/slurm-21213603.out
    Untracked:  code/slurm-21213604.out
    Untracked:  code/slurm-21214302.out
    Untracked:  code/slurm-21214303.out
    Untracked:  code/slurm-21214359.out
    Untracked:  code/slurm-21216841.out
    Untracked:  code/slurm-21216842.out
    Untracked:  code/slurm-21216843.out
    Untracked:  code/slurm-21216844.out
    Untracked:  code/slurm-21216845.out
    Untracked:  code/slurm-21217091.out
    Untracked:  code/slurm-21217092.out
    Untracked:  code/slurm-21217093.out
    Untracked:  code/slurm-21217094.out
    Untracked:  code/slurm-21217095.out
    Untracked:  code/slurm-21217137.out
    Untracked:  code/slurm-21217138.out
    Untracked:  code/slurm-21217139.out
    Untracked:  code/slurm-21217140.out
    Untracked:  code/slurm-21217141.out
    Untracked:  code/slurm-21217343.out
    Untracked:  code/slurm-21217344.out
    Untracked:  code/slurm-21217345.out
    Untracked:  code/slurm-21217346.out
    Untracked:  code/slurm-21217347.out
    Untracked:  code/slurm-21217382.out
    Untracked:  code/slurm-21217383.out
    Untracked:  code/slurm-21217384.out
    Untracked:  code/slurm-21217385.out
    Untracked:  code/slurm-21217386.out
    Untracked:  code/slurm-21217411.out
    Untracked:  code/slurm-21217412.out
    Untracked:  code/slurm-21217413.out
    Untracked:  code/slurm-21217414.out
    Untracked:  code/slurm-21217416.out
    Untracked:  code/slurm-21217455.out
    Untracked:  code/slurm-21217456.out
    Untracked:  code/slurm-21217457.out
    Untracked:  code/slurm-21217458.out
    Untracked:  code/slurm-21217459.out
    Untracked:  code/slurm-21217482.out
    Untracked:  code/slurm-21217489.out
    Untracked:  code/slurm-21217490.out
    Untracked:  code/slurm-21217491.out
    Untracked:  code/slurm-21217492.out
    Untracked:  code/slurm-21217531.out
    Untracked:  code/slurm-21217532.out
    Untracked:  code/slurm-21217533.out
    Untracked:  code/slurm-21217534.out
    Untracked:  code/slurm-21217536.out
    Untracked:  code/slurm-21217574.out
    Untracked:  code/slurm-21217575.out
    Untracked:  code/slurm-21217576.out
    Untracked:  code/slurm-21217577.out
    Untracked:  code/slurm-21217578.out
    Untracked:  code/slurm-21217601.out
    Untracked:  code/slurm-21217602.out
    Untracked:  code/slurm-21217603.out
    Untracked:  code/slurm-21217604.out
    Untracked:  code/slurm-21217607.out
    Untracked:  code/slurm-21217627.out
    Untracked:  code/slurm-21217628.out
    Untracked:  code/slurm-21217629.out
    Untracked:  code/slurm-21217630.out
    Untracked:  code/slurm-21217633.out
    Untracked:  code/slurm-21217669.out
    Untracked:  code/slurm-21217671.out
    Untracked:  code/slurm-21217672.out
    Untracked:  code/slurm-21217673.out
    Untracked:  code/slurm-21217674.out
    Untracked:  code/slurm-21217700.out
    Untracked:  code/slurm-21217701.out
    Untracked:  code/slurm-21217702.out
    Untracked:  code/slurm-21217703.out
    Untracked:  code/slurm-21217707.out
    Untracked:  code/slurm-21217747.out
    Untracked:  code/slurm-21217748.out
    Untracked:  code/slurm-21217749.out
    Untracked:  code/slurm-21217750.out
    Untracked:  code/slurm-21217751.out
    Untracked:  code/slurm-21217773.out
    Untracked:  code/slurm-21217783.out
    Untracked:  code/slurm-21217784.out
    Untracked:  code/slurm-21217785.out
    Untracked:  code/slurm-21217786.out
    Untracked:  code/slurm-21217975.out
    Untracked:  code/slurm-21217976.out
    Untracked:  code/slurm-21217977.out
    Untracked:  code/slurm-21217979.out
    Untracked:  code/slurm-21217980.out
    Untracked:  code/slurm-21218015.out
    Untracked:  code/slurm-21218016.out
    Untracked:  code/slurm-21218017.out
    Untracked:  code/slurm-21218018.out
    Untracked:  code/slurm-21218019.out
    Untracked:  code/slurm-21218038.out
    Untracked:  code/slurm-21218039.out
    Untracked:  code/slurm-21218040.out
    Untracked:  code/slurm-21218046.out
    Untracked:  code/slurm-21218047.out
    Untracked:  code/slurm-21218071.out
    Untracked:  code/slurm-21218072.out
    Untracked:  code/slurm-21218073.out
    Untracked:  code/slurm-21218076.out
    Untracked:  code/slurm-21218077.out
    Untracked:  code/slurm-21218081.out
    Untracked:  code/slurm-21218082.out
    Untracked:  code/slurm-21218083.out
    Untracked:  code/slurm-21218084.out
    Untracked:  code/slurm-21218085.out
    Untracked:  code/slurm-21218117.out
    Untracked:  code/slurm-21218118.out
    Untracked:  code/slurm-21218119.out
    Untracked:  code/slurm-21218120.out
    Untracked:  code/slurm-21218125.out
    Untracked:  code/slurm-21218129.out
    Untracked:  code/split_vcf.sh
    Untracked:  code_backup/
    Untracked:  data/ScTypeDB_full.xlsx
    Untracked:  data/hg19_gtf_genomic_annots_ver2.gr.rds
    Untracked:  data/num_overlaps_finemapped_SNPs_and_ctcf.txt
    Untracked:  data/qqplot_SNPs_high_spliceAI.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_SCZ.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_aFib.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_allergy.png
    Untracked:  data/torus_enrichment_novel_annot.est
    Untracked:  data/torus_joint_enrichment.est
    Untracked:  data/torus_joint_refined_enrichment.est
    Untracked:  enhancer_gene_feature.rmd
    Untracked:  fig1_panels.pdf
    Untracked:  fig2.pdf
    Untracked:  fig_panel2.pdf
    Untracked:  gene_mapping.pdf
    Untracked:  output/AAD/GMP_merge_stats.txt
    Untracked:  output/AAD/Wang2020_joint.results
    Untracked:  output/AAD/Wang2020_joint_T.results
    Untracked:  output/AAD/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/allergy/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/allergy/Ulirsch2019_disjoint_snps.sumstats
    Untracked:  output/AAD/allergy/Wang2020_T_subsets.est
    Untracked:  output/AAD/allergy/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/allergy/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/allergy/Wang2020_joint_T.results
    Untracked:  output/AAD/allergy/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/allergy/Wang2020_tissueResT.est
    Untracked:  output/AAD/allergy/torus_enrichment_CD4.est
    Untracked:  output/AAD/allergy/torus_enrichment_CD8.est
    Untracked:  output/AAD/allergy/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD4_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD8_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_subsets.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/asthma_adult/Wang2020_joint_T.results
    Untracked:  output/AAD/asthma_adult/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/asthma_adult/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/AAD/asthma_child/CD4_compare.est
    Untracked:  output/AAD/asthma_child/CD8_compare.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_subsets.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/asthma_child/Wang2020_joint_T.results
    Untracked:  output/AAD/asthma_child/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/asthma_child/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/LDL_ukb_L10.gif
    Untracked:  output/LDL_ukb_L10.pdf
    Untracked:  output/background_SNPs_annotated_percent.txt
    Untracked:  output/ldsc
    Untracked:  output/locus_1452.gif
    Untracked:  output/locus_1452.pdf
    Untracked:  output/splicing/PTR_across_traits_annotations.results
    Untracked:  output/splicing/header.txt
    Untracked:  output/splicing/prior/
    Untracked:  output/splicing/scz_PTR_annotations.results
    Untracked:  output/splicing/scz_neuOCR_m6a_DMR.results
    Untracked:  output/splicing/scz_spliceAI0.03_hist.png
    Untracked:  output/splicing/scz_spliceAI0.03_scatterplot.png
    Untracked:  output/splicing/scz_spliceai_binary0.03.results
    Untracked:  output/splicing/torus_afib_spliceai.est
    Untracked:  output/splicing/torus_annotations_spliceai0.01.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.03.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.05.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.07.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.1.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.2.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai9.txt.gz
    Untracked:  output/splicing/torus_enrichment_joint_scz_mtsplice0.6_hypothalamus-brain.est
    Untracked:  output/splicing/torus_enrichment_joint_scz_spliceAI.est
    Untracked:  output/splicing/torus_spliceai0.01.enrichment
    Untracked:  output/splicing/torus_spliceai0.03.enrichment
    Untracked:  output/splicing/torus_spliceai0.05.enrichment
    Untracked:  output/splicing/torus_spliceai0.07.enrichment
    Untracked:  output/splicing/torus_spliceai0.1.enrichment
    Untracked:  output/splicing/torus_spliceai0.2.enrichment
    Untracked:  output/splicing/torus_zscores.txt.gz
    Untracked:  output/torus
    Untracked:  panel_figure2.pdf
    Untracked:  test.txt

Unstaged changes:
    Deleted:    .Rprofile
    Modified:   analysis/index.Rmd
    Modified:   analysis/ldsc_PTR_results.Rmd
    Deleted:    output/AAD/Caldero2019_disjoint_snps.sumstats
    Modified:   output/AAD/allergy/Caldero2019_disjoint_snps.sumstats
    Modified:   output/AAD/allergy/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/allergy/Wang2020_indiv.est
    Modified:   output/AAD/allergy/Wang2020_joint.results
    Deleted:    output/AAD/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/AAD/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/AAD/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/AAD/asthma/diffe_adult_blood_torus.est
    Deleted:    output/AAD/asthma/joint_blood_immune_rest_vs_stimu.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/AAD/asthma/lung_clusters_dict.txt
    Deleted:    output/AAD/asthma/lung_clusters_info.txt
    Deleted:    output/AAD/asthma/stimu_adult_blood_torus.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_rest.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_stimulated.est
    Deleted:    output/AAD/asthma/zhang2021_annot_percent.txt
    Deleted:    output/AAD/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/AAD/asthma/zhang2021_peaks_per_celltype.txt
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD4_compare.est
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD8_compare.est
    Deleted:    output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/asthma_adult/Wang2020_indiv.est
    Modified:   output/AAD/asthma_adult/Wang2020_joint.results
    Modified:   output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/asthma_child/Wang2020_indiv.est
    Modified:   output/AAD/asthma_child/Wang2020_joint.results
    Deleted:    output/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/asthma/diffe_adult_blood_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/asthma/lung_clusters_dict.txt
    Deleted:    output/asthma/lung_clusters_info.txt
    Deleted:    output/asthma/stimu_adult_blood_torus.est
    Deleted:    output/asthma/zhang2021_annot_percent.txt
    Deleted:    output/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/asthma/zhang2021_peaks_per_celltype.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/d96f416503f721035dc5859438ed502212b24fe1/docs/index.html" target="_blank">d96f416</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-06-15
</td>
<td>
estimate_h2g
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/ccdd16704665569a95268c4ee8b49bcc7b75a03d/docs/index.html" target="_blank">ccdd167</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-06-15
</td>
<td>
estimate_h2g
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/8c82e64f0adb7bc792923eebf141e885482e7f69/docs/index.html" target="_blank">8c82e64</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-06-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/611b343e10b567848bb9882e1ddd2d8daecce0cf/analysis/index.Rmd" target="_blank">611b343</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-06-15
</td>
<td>
wflow_publish(files = c(&quot;analysis/index.Rmd&quot;, &quot;docs/ldsc_PTR_results.html&quot;))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/27f0ead9addb42034091ab7c7e861c3a9ab288fe/docs/index.html" target="_blank">27f0ead</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-06-04
</td>
<td>
wflow_publish(files = c(&quot;docs/index.html&quot;, &quot;docs/exploring_lung_scATAC_R.html&quot;,
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/a71a34d30a87f58def191391281a94d439718719/docs/index.html" target="_blank">a71a34d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-03-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/5ea9a98a4bb2001e7e7be6c4449e5dd91265b0ba/analysis/index.Rmd" target="_blank">5ea9a98</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-03-24
</td>
<td>
update spliceAI results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/506928dbbe159d921df0303c8ee6284f63fcc76e/docs/index.html" target="_blank">506928d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-03-23
</td>
<td>
analyzed snATAC-seq lung dataset
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/5b65bd10f715e7c6d2f8dffdcac3cfb13feac2cf/docs/index.html" target="_blank">5b65bd1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/db3c6f96af3bb6ed4d4e210f7ac5585517152b8a/analysis/index.Rmd" target="_blank">db3c6f9</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-10-26
</td>
<td>
test enrichment for asthma risk variants
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/53a87745f669f007b900afd4a5e34ebb051d14ab/docs/index.html" target="_blank">53a8774</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/2a6fc915e159af6504e1abdfdf30bd82b4ab4f75/docs/index.html" target="_blank">2a6fc91</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/41f53c83d18a13da1b50ff9004b3abb011b9fdb1/docs/index.html" target="_blank">41f53c8</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/b0b938bcb9c43790bb7bb0aa211ad03429e47934/docs/index.html" target="_blank">b0b938b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/8a5656cf110594c30de185f7ba5d12cc21a41e08/analysis/index.Rmd" target="_blank">8a5656c</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-22
</td>
<td>
analyzed Asthma GWAS with functional priors from single cell data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/639aacfea2387dc7ad5dd85854b4666912d3c0c1/docs/index.html" target="_blank">639aacf</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/c046e9e96de0bcaf5382954783d57e660e42354d/docs/index.html" target="_blank">c046e9e</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/05f73e942320177f10cffd317526a654adc06b06/docs/index.html" target="_blank">05f73e9</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/bed838e6d4f2f222fbcb9bc8cde0d2e5e66bf48a/analysis/index.Rmd" target="_blank">bed838e</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-09-01
</td>
<td>
update evaluations on slicing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/bbfe1d41681fcf52636e941ef2b1372fab8568c1/docs/index.html" target="_blank">bbfe1d4</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/540e287734a85d462236518a91a70d1e90c91e88/docs/index.html" target="_blank">540e287</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/e597d4ed8f7db6c2146a6c55ef6ff6801c76197f/docs/index.html" target="_blank">e597d4e</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/491ca5ccff37db99cae0f1dc9263beaded759acd/docs/index.html" target="_blank">491ca5c</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/ba251cb4b94ee3d0e77190320421211dc34b99b5/docs/index.html" target="_blank">ba251cb</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/66c75b9b528c238fb339e8d7bececbdbacf3a1c1/docs/index.html" target="_blank">66c75b9</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/1a76b294e45c0bb495ac2108d8c2b021d430577d/docs/index.html" target="_blank">1a76b29</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ae12e8aa24b22a009ea11c30c06102393df7ff28/analysis/index.Rmd" target="_blank">ae12e8a</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-06
</td>
<td>
evaluate sequence constraints
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/752cb393786591800d7a9477093e2f731cd88433/docs/index.html" target="_blank">752cb39</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/f1c5950cd9ff555c31e73394072eb624a8254955/docs/index.html" target="_blank">f1c5950</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/5e70eccfdf2eb1d13b5fd97e7abe51a662b54894/analysis/index.Rmd" target="_blank">5e70ecc</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
edit index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/d16a5e0ba9cc157232914ff14a755f7df7a1bf5b/docs/index.html" target="_blank">d16a5e0</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/6f7214b5258820d1c3a3ea16ebe150474514f9ec/analysis/index.Rmd" target="_blank">6f7214b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
edit index page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/4d851a879f76ce9a8479ade95826e6f7c7d48ee4/docs/index.html" target="_blank">4d851a8</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/4cc091d4f1fa642f9503e463782157e779a7bc21/analysis/index.Rmd" target="_blank">4cc091d</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
start my phd project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/12fa9ffdf0845454855a4fbee5c489d9e6deb92f/docs/index.html" target="_blank">12fa9ff</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/c00a877f4881e8d1bdfc11250bcb1b5bb75a757b/analysis/index.Rmd" target="_blank">c00a877</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-04
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="welcome-to-the-project-website." class="section level1">
<h1>Welcome to the project website.</h1>
<p>My thesis project is about improving functional fine mapping procedures that enable us to identify causal variants underlying complex traits such as obsesity and asthma.</p>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This document will provide an overview of my thesis project, including the aims for the project, summary of previous work, details of the method, and preliminary results. By constantly reviewing this document, I can keep track of my research aims without getting diverted too far.</p>
<div id="literature-reviews" class="section level2">
<h2>Literature reviews</h2>
<ul>
<li>[Current challenges in functional fine-mapping]</li>
<li>[Existing pipelines or methods]
<ul>
<li>Polyfun + SuSiE</li>
<li>TORUS + SuSiE</li>
</ul></li>
<li><a href="biology_bkg.html">Biology backgrounds</a></li>
<li><a href="method_bkg.html">Prediction methods</a></li>
</ul>
</div>
<div id="results-by-methods" class="section level2">
<h2>Results by Methods</h2>
<ul>
<li>[feedbacks]</li>
<li><a href="enrichment_analysis.html">Torus enrichment results</a></li>
<li><a href="ldsc_results.html">LDSC enrichment results</a></li>
<li><a href="snp_finemapping_results.html">fine mapping results</a></li>
<li>[gene mapping results]</li>
</ul>
</div>
<div id="results-by-projects" class="section level2">
<h2>Results by Projects</h2>
<ul>
<li><a href="asthma_prelim_results.html">Asthma Projects</a>
<ul>
<li><a href="identifying_TRMs_in_lungs.html">identifying TRMs</a></li>
<li><a href="lung_Wang2020_scRNA.html">scRNA-seq analysis</a></li>
<li><a href="exploring_lung_scATAC_R.html">scATAC-seq analysis</a></li>
</ul></li>
<li>[Functional annotation project]
<ul>
<li><a href="annotations_PTR.html">PTR_results</a></li>
</ul></li>
</ul>
</div>
<div id="sample-qcs-by-dataset" class="section level2">
<h2>Sample QCs by dataset</h2>
<ul>
<li>[Wang2020 et al.]</li>
</ul>
</div>
<div id="references" class="section level2">
<h2>References</h2>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
