<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Lab 4: Fine-mapping with SuSiE</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">funcFinemapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Lab 4: Fine-mapping with SuSiE</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-02-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>funcFinemapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you'll want to first commit it to the Git repo. If you're still working on the analysis, you can ignore this warning. When you're finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210404code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210404code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210404)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree2532745b6af2af708d5c2101d123c8bac41fb41etargetblank2532745a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/j3gu/funcFinemapping/tree/2532745b6af2af708d5c2101d123c8bac41fb41e" target="_blank">2532745</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree2532745b6af2af708d5c2101d123c8bac41fb41etargetblank2532745a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/j3gu/funcFinemapping/tree/2532745b6af2af708d5c2101d123c8bac41fb41e" target="_blank">2532745</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/figure/
    Ignored:    analysis/lab4_prepare.nb.html
    Ignored:    analysis/ldsc_results.nb.html
    Ignored:    analysis/mtsplice_finemapping_results.nb.html
    Ignored:    analysis/results.nb.html
    Ignored:    analysis/snp_finemapping_results.nb.html
    Ignored:    analysis/splicing.nb.html
    Ignored:    analysis/susie_tutorial.nb.html

Untracked files:
    Untracked:  SNPs_categories,png
    Untracked:  SNPs_categories.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/asthma_prelim_results_archive.Rmd
    Untracked:  analysis/asthma_prelim_results_cp.Rmd
    Untracked:  analysis/asthma_results_cp.Rmd
    Untracked:  analysis/enhancer_gene_feature.Rmd
    Untracked:  analysis/feedback.Rmd
    Untracked:  analysis/gene_finemapping_results.Rmd
    Untracked:  analysis/lab4.Rmd
    Untracked:  analysis/lab4_prepare.Rmd
    Untracked:  analysis/learn_susie.Rmd
    Untracked:  analysis/mtsplice_finemapping_results.Rmd
    Untracked:  analysis/notes.Rmd
    Untracked:  analysis/snp_finemapping_results.Rmd
    Untracked:  analysis/splicing.Rmd
    Untracked:  analysis/susie_tutorial.Rmd
    Untracked:  bmi_locus1410.pdf
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/ldsc_regression.sh
    Untracked:  code/make_plots.R
    Untracked:  code/run_ldsc.sh
    Untracked:  code/run_ldsc_with_bed.sh
    Untracked:  code/run_ldsc_with_bed_v2.sh
    Untracked:  code/run_susie.R
    Untracked:  code/run_torus.sh
    Untracked:  code/split_vcf.sh
    Untracked:  data/hg19_gtf_genomic_annots_ver2.gr.rds
    Untracked:  data/num_overlaps_finemapped_SNPs_and_ctcf.txt
    Untracked:  data/scz_2018
    Untracked:  data/torus_enrichment_novel_annot.est
    Untracked:  data/torus_joint_enrichment.est
    Untracked:  data/torus_joint_refined_enrichment.est
    Untracked:  enhancer_gene_feature.rmd
    Untracked:  fig1_panels.pdf
    Untracked:  fig2.pdf
    Untracked:  fig_panel2.pdf
    Untracked:  gene_mapping.pdf
    Untracked:  lab4
    Untracked:  output/AAD/GMP_merge_stats.txt
    Untracked:  output/AAD/allergy/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD4_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD8_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/LDL_ukb_L10.gif
    Untracked:  output/LDL_ukb_L10.pdf
    Untracked:  output/background_SNPs_annotated_percent.txt
    Untracked:  output/locus_1452.gif
    Untracked:  output/locus_1452.pdf
    Untracked:  panel_figure2.pdf
    Untracked:  test.txt
    Untracked:  transfer/

Unstaged changes:
    Deleted:    .Rprofile
    Modified:   analysis/asthma_prelim_results.Rmd
    Modified:   analysis/asthma_results.Rmd
    Modified:   analysis/enrichment_analysis.Rmd
    Modified:   output/AAD/allergy/Ulirsch2019/GMP_merge_compare.est
    Deleted:    output/AAD/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/AAD/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/AAD/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/AAD/asthma/diffe_adult_blood_torus.est
    Deleted:    output/AAD/asthma/joint_blood_immune_rest_vs_stimu.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/AAD/asthma/lung_clusters_dict.txt
    Deleted:    output/AAD/asthma/lung_clusters_info.txt
    Deleted:    output/AAD/asthma/stimu_adult_blood_torus.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_rest.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_stimulated.est
    Deleted:    output/AAD/asthma/zhang2021_annot_percent.txt
    Deleted:    output/AAD/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/AAD/asthma/zhang2021_peaks_per_celltype.txt
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD4_compare.est
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD8_compare.est
    Deleted:    output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare.est
    Deleted:    output/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/asthma/diffe_adult_blood_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/asthma/lung_clusters_dict.txt
    Deleted:    output/asthma/lung_clusters_info.txt
    Deleted:    output/asthma/stimu_adult_blood_torus.est
    Deleted:    output/asthma/zhang2021_annot_percent.txt
    Deleted:    output/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/asthma/zhang2021_peaks_per_celltype.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p><em>Created by Jing Gu</em></p>
<p>SuSiE is a state-of-art fine-mapping method based on Bayesian variable selection regression. The name of &quot;SuSiE&quot; means the sum of single effects, which detects single causal effect one at a time but also extends to multiple effects through conditional analysis.</p>
<p>SusieR package was developed to run fine-mapping procedures over GWAS summary statistics or invidiual level genotype/phenotype data. More details about the SusieR package can be found at [SusieR github page] [<a href="https://stephenslab.github.io/susieR/index.html" class="uri">https://stephenslab.github.io/susieR/index.html</a>].</p>
<p>By the end of the lab, you should be able to:</p>
<ul>
<li><strong>Understand the types of input files for running susieR</strong></li>
<li><strong>Interpret output results from susieR</strong></li>
<li><strong>Evaluate the effects of setting different numbers of causal variants on the results</strong></li>
<li><strong>Compare the uses of different LD reference panels on the results</strong></li>
</ul>
<div id="setup" class="section level1">
<h1>Setup</h1>
<div id="download-packages" class="section level2">
<h2>download packages</h2>
<p><strong>First, let's download necessary packages and load data for today's lab session</strong></p>
<pre><code>Loading required package: susieR</code></pre>
<pre><code>Loading required package: data.table</code></pre>
<pre><code>Loading required package: dplyr</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre><code>Loading required package: bigsnpr</code></pre>
<pre><code>Loading required package: bigstatsr</code></pre>
<pre><code>Loading required package: ggplot2</code></pre>
<pre><code>Warning in data(&quot;lab4/susie.RData&quot;): data set &#39;lab4/susie.RData&#39; not found</code></pre>
</div>
<div id="dataset" class="section level2">
<h2>Dataset</h2>
<ul>
<li>gwas summary statistics (after QC):
<ul>
<li>Body mass index as the phenotype</li>
<li>A meta-analysis of Locke et al 2015 and analyses from UK Biobank</li>
<li>No individual genotypes are provided.</li>
<li>Chr, pos, rsID, effect/non-effect alleles, beta, se(beta), p-values, odds ratio, N, etc.</li>
</ul></li>
<li>reference LD panels from 1000 genome or UK Biobank
<ul>
<li>chr, pos, rsID, major/minor alleles, etc.</li>
<li>LD matrix: pair-wise LD coefficients across each LD block</li>
</ul></li>
<li>LD blocks:
<ul>
<li>Based on the European haplotypes, researchers have built 1704 LD blocks, where variants across blocks are not correlated.</li>
</ul></li>
</ul>
</div>
<div id="input-files-for-susier" class="section level2">
<h2>Input files for susieR</h2>
<ul>
<li>individual level phenotypes/genotypes: <code>susie()</code> or Summary statistics: <code>susie_rss()</code></li>
<li>LD coefficients (R)</li>
<li>expected number of causal variants (L)</li>
</ul>
</div>
</div>
<div id="run-susier-on-the-simulated-example" class="section level1">
<h1>Run susieR on the simulated example</h1>
<p>Let's first test the performance of susieR on the simulated dataset. There are two simulation replicates in the dataset. We will focus on the first one.</p>
<p><strong>load N3finemapping from susieR package</strong></p>
<pre class="r"><code>data(N3finemapping)
attach(N3finemapping)
str(N3finemapping)</code></pre>
<pre><code>List of 8
 $ X                : num [1:574, 1:1001] -0.0209 -0.0209 -0.0209 -0.0209 -0.0209 ...
 $ chrom            : chr &quot;chr19&quot;
 $ pos              : int [1:8750(1d)] 7302164 7302332 7302591 7303825 7304182 7304408 7305504 7306114 7306459 7306697 ...
 $ true_coef        : num [1:1001, 1:2] 0 0 0 0 0 0 0 0 0 0 ...
 $ residual_variance: num [1:2(1d)] 6.29 1.65
 $ Y                : num [1:574, 1:2] 0.2284 -0.6509 0.0768 -3.1507 2.338 ...
 $ allele_freq      : num [1:1001, 1] 0.0105 0.2631 0.419 0.0261 0.0819 ...
 $ V                : num [1:2, 1:2] 7.842 -0.221 -0.221 2.131</code></pre>
<p>We can identify which variables are simulated as truly causal by searching for <span class="math inline">\(b\neq0\)</span>.</p>
<pre class="r"><code>which(true_coef[,1] !=0)</code></pre>
<pre><code>[1] 403 653 773</code></pre>
<pre class="r"><code>plot(true_coef[,2], ylab = &quot;true effect size&quot;)</code></pre>
<p><img src="figure/lab4.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /> <strong>Run linear regression to obtain p-values, similar as running GWAS</strong></p>
<pre class="r"><code>sumstats &lt;- univariate_regression(X, Y[,1])
z_scores &lt;- sumstats$betahat / sumstats$sebetahat
susie_plot(z_scores, y = &quot;z&quot;, b = true_coef[,1])</code></pre>
<p><img src="figure/lab4.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /> <strong>Fine-mapping with susieR at L = 1</strong></p>
<p>The susie_plot() function is a scatter plot of PIP or z scores against variant index. The variables selected into credible sets are colored by groups.</p>
<pre class="r"><code>fitted.L1 &lt;- susie(X, Y[,1],
                L = 1,
        verbose = TRUE)</code></pre>
<pre><code>For an X with many columns, please consider installing  the Rfast package for more efficient credible set (CS)  calculations.</code></pre>
<pre><code>[1] &quot;objective:-1386.58122361469&quot;
[1] &quot;objective:-1385.18416210208&quot;
[1] &quot;objective:-1385.17881826275&quot;
[1] &quot;objective:-1385.17879793437&quot;
[1] &quot;objective:-1385.17879786253&quot;</code></pre>
<pre class="r"><code>susie_plot(fitted.L1, y=&quot;PIP&quot;)</code></pre>
<p><img src="figure/lab4.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /> <strong>Fine-mapping with susieR at L = 10</strong></p>
<pre class="r"><code>fitted.L10 &lt;- susie(X, Y[,1],
                L = 10,
        verbose = TRUE)</code></pre>
<pre><code>For an X with many columns, please consider installing  the Rfast package for more efficient credible set (CS)  calculations.</code></pre>
<pre><code>[1] &quot;objective:-1380.57545244462&quot;
[1] &quot;objective:-1377.48660917726&quot;
[1] &quot;objective:-1375.85777207947&quot;
[1] &quot;objective:-1375.80892301654&quot;
[1] &quot;objective:-1370.33949333218&quot;
[1] &quot;objective:-1370.19677277173&quot;
[1] &quot;objective:-1370.10919739198&quot;
[1] &quot;objective:-1370.10918017465&quot;
[1] &quot;objective:-1370.10901872279&quot;</code></pre>
<pre class="r"><code>susie_plot(fitted.L10, y=&quot;PIP&quot;)</code></pre>
<p><img src="figure/lab4.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /> By default, a credible set is defined as the minimum set of variables of which PIPs sum up to be at least 95%, and the minimum mean correlations within a set to be 0.5.</p>
<p>Problem 1: With seting L = 10, are the causal variables included in the credible sets?</p>
<pre class="r"><code>print(fitted.L10$sets)</code></pre>
<pre><code>$cs
$cs$L2
[1] 653

$cs$L1
[1] 773 777

$cs$L3
 [1] 362 365 372 373 374 379 381 383 384 386 387 388 389 391 392 396 397 398 399
[20] 400 401 403 404 405 407 408 415


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L2    1.0000000     1.0000000       1.0000000
L1    0.9815726     0.9815726       0.9815726
L3    0.8686309     0.9640176       0.9720711

$cs_index
[1] 2 1 3

$coverage
[1] 0.9998236 0.9988858 0.9539811

$requested_coverage
[1] 0.95</code></pre>
<p>It has been shown through simulations that the finemapping results are robust to the choices of L.</p>
</div>
<div id="fine-mapping-on-gwas-summary-statistics" class="section level1">
<h1>Fine-mapping on GWAS summary statistics</h1>
<p>Let's first take a look at GWAS summary statistics file. There is one additional column that specifies the LD block that each variant belongs.</p>
<pre class="r"><code>head(gwas)</code></pre>
<pre><code>  chr       pos a0 a1    beta     se        snp  pval     zscore block
1   6 108467024  A  G  0.0007 0.0018   rs218291 0.700  0.3888889   704
2   6 108469920  T  G -0.0009 0.0019   rs170125 0.630 -0.4736842   704
3   6 108470257  A  G -0.0088 0.0038 rs12173906 0.020 -2.3157895   704
4   6 108471948  C  T  0.0008 0.0018   rs660532 0.660  0.4444444   704
5   6 108474665  G  T -0.0092 0.0038  rs9400209 0.015 -2.4210526   704
6   6 108474935  C  T -0.0093 0.0038  rs9400210 0.015 -2.4473684   704</code></pre>
<pre class="r"><code>gwas&lt;-split(gwas, gwas$block)
names(gwas)</code></pre>
<pre><code>[1] &quot;704&quot;  &quot;1452&quot;</code></pre>
<p>By plotting -log10(p) against genomic positions, we see many SNPs near the lead SNP passing the significance threashold. To help prioritize SNPs for downstream analysis and validation, fine-mapping provides a statistical approach to evaluate each SNP's causal probability.</p>
<pre class="r"><code>orig.gwas&lt;-gwas[[&quot;1452&quot;]]
plot(x = orig.gwas$pos, y = -log10(orig.gwas$pval), 
     xlab = &quot;pos&quot;, ylab=&quot;-log10(p)&quot;)</code></pre>
<p><img src="figure/lab4.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /> ## Step 1: match variants between GWAS summary stats and LD panel Bigsnpr is a very useful R packpage to process genotype data. We will use its &quot;snp_match&quot; function to match variants in GWAS summary stats and LD panels. This function outputs a summary table with previous GWAS summary statistics for the matched SNPs and additional columns such as the &quot;bigSNP_index&quot;, the index number for each matched variant in the LD panels. We will further utilize these index numbers to quickly extract the LD coefficients from the LD matrix.</p>
<pre class="r"><code>matched.gwas&lt;-as_tibble(bigsnpr::snp_match(sumstats = gwas[[&quot;1452&quot;]], 
                          info_snp = bigSNP[[&quot;1452&quot;]]$map) %&gt;% 
                          dplyr::rename(og_index = `_NUM_ID_.ss`) %&gt;% 
                          dplyr::rename(bigSNP_index = `_NUM_ID_`))</code></pre>
<pre><code>537 variants to be matched.</code></pre>
<pre><code>0 ambiguous SNPs have been removed.</code></pre>
<pre><code>532 variants have been matched; 0 were flipped and 162 were reversed.</code></pre>
<p>Problem 2: From the help page of &quot;snp_match&quot; function, we know when the alleles are reversed the directions of beta values will be reversed as well. Let's make a scatter plot comparing the beta values from original gwas against those from the matched gwas.</p>
<p>Problem 3: Make the same plot for z-scores. The directions of z scores can have a big impact on susie's performance. So be careful about specifying the right z-scores used in susie. Otherwise, you can directly compute zscores by dividing beta from se(beta). This relationship also implies the direction of z scores should be the same as beta.</p>
<p>Knowing the directions of z scores were not updated, we manually computed the z scores in the following codes and repeated this step by looping over all the LD blocks.</p>
<pre class="r"><code>for(i in names(gwas)){
  gwas[[i]]&lt;-as_tibble(bigsnpr::snp_match(sumstats = gwas[[i]], 
                          info_snp = bigSNP[[i]]$map) %&gt;%  
                          dplyr::rename(og_index = `_NUM_ID_.ss`) %&gt;% 
                          dplyr::rename(bigSNP_index = `_NUM_ID_`)) %&gt;%
                          dplyr::mutate(z_matched = beta/se)
}</code></pre>
<pre><code>1,174 variants to be matched.</code></pre>
<pre><code>0 ambiguous SNPs have been removed.</code></pre>
<pre><code>1,172 variants have been matched; 0 were flipped and 284 were reversed.</code></pre>
<pre><code>537 variants to be matched.</code></pre>
<pre><code>0 ambiguous SNPs have been removed.</code></pre>
<pre><code>532 variants have been matched; 0 were flipped and 162 were reversed.</code></pre>
<div id="step2-extract-ld-coefficients-for-the-tested-snps" class="section level2">
<h2>step2: Extract LD coefficients for the tested SNPs</h2>
<p>The given LD matrix for each LD block is a N by N matrix, where N is the number of common variants within the block. The row and column index match with the &quot;bigSNP_index&quot; in the summary table output from snp_match function.</p>
<pre class="r"><code>snps.vec&lt;-gwas[[&quot;1452&quot;]]$bigSNP_index
r2.mat&lt;-bigSNP[[&quot;1452&quot;]]$LD_matrix[snps.vec, snps.vec]
#check if the bigSNP_index matches between GWAS and the LD panel
sum(gwas[[&quot;1452&quot;]]$snp != bigSNP[[&quot;1452&quot;]]$map[snps.vec,&quot;id&quot;])</code></pre>
<pre><code>[1] 0</code></pre>
</div>
<div id="step-3-run-susie_rss-for-summary-statistics" class="section level2">
<h2>step 3: run susie_rss() for summary statistics</h2>
<p>Let's first try L=1. Here we sepcified z scores as the column of &quot;z_matched&quot;, which was corrected by directions.</p>
<pre class="r"><code>res.L1 &lt;- susieR::susie_rss(z = gwas[[&quot;1452&quot;]]$z_matched, R = r2.mat, L = 1, verbose = F)
summary(res.L1)</code></pre>
<pre><code>
Variables in credible sets:

 variable variable_prob cs
      492     0.4999607  1
      482     0.4999607  1

Credible sets summary:

 cs cs_log10bf cs_avg_r2 cs_min_r2 variable
  1   55.70593 0.9973469 0.9973469  482,492</code></pre>
<pre class="r"><code>res.L1[[&quot;z&quot;]]&lt;-gwas[[&quot;1452&quot;]]$z_matched
par(mfrow = c(1,2))
susie_plot(res.L1, &quot;PIP&quot;, add_legend = &quot;topleft&quot;)
susie_plot(res.L1, y=&quot;z&quot;)</code></pre>
<p><img src="figure/lab4.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /> From the output summary, we know one credible set was identified. There are two variants, index 482 and 492, included in the credible set, each with around 0.5 posterior probability. As their average LD correlation is very close to one, Susie was not able to distinguish between them. Therefore, the probability of being causal given data were equally distributed between these two variables. Problem</p>
<p>Problem 4: Run susie at L=10 and then report the number of variables selected in each credible set. Additionally, use susie_plot() to visualize the distributions of PIPs and -log10(pval).</p>
<p>Now we are done with the general procedures for running susie. Let's put them together and loop over all the blocks.</p>
<pre class="r"><code>susie.ukb.L10&lt;-list()
for(i in names(gwas)){
  snps.vec&lt;-gwas[[i]]$bigSNP_index
  r2.mat&lt;-bigSNP[[i]]$LD_matrix[snps.vec, snps.vec]
  susie.ukb.L10[[i]]&lt;-susie_rss(z = gwas[[i]]$z_matched, R = r2.mat, L = 10, verbose = F)
  susie.ukb.L10[[i]][[&quot;z&quot;]]&lt;-gwas[[i]]$z_matched #add the original z scores to susie outputs
}</code></pre>
<pre><code>For large R or large XtX, consider installing the  Rfast package for better performance.</code></pre>
</div>
</div>
<div id="compare-different-ld-panels" class="section level1">
<h1>Compare different LD panels</h1>
<p>Lastly, we will examine how the mismatch of LD matrix can affect how variants are selected into credible sets . Our previous analysis was based on the LD matrix computed over around 50K British individuals provided by UK Biobank. In the dataset, we also provided the fine-mapped results using the LD matrix from samples in 1000 genome consortium. The LD matrix for the 1kg LD reference panel was computed over 500 Eurpean individuals.</p>
<p>When comparing the fine-mapped results over all 97 blocks that contain GWAS lead loci, we found using 1kg LD reference panel tends to capture many more variants with non-significant p-values in credible sets. This looks suspicious and deserves some examinations. Let's focus on one specific LD block 704 and see how the results differ due to the use of different LD reference panels.</p>
<p>Problem 5. Compare the credible sets obtained from running susie using ukbb LD matrix and 1kg LD matrix. The susie outputs from using 1kg LD matrix are stored in a variable called <code>susie.1kg.L10</code>. Also make the plots of PIPs or -log10(p-val) to show the difference.</p>
<p><strong>Check the LD patterns relative to the confident SNP in credible sets</strong></p>
<p>We can also visualize LD patterns by coloring the data points by the LD coefficients relative to the SNP of interest. Particularly, we examined the SNP with the highest SNP in the additional credible set identified by susie using 1kg LD reference panel. From the plot below, LD coefficients relative to the top snp rs12204742 in that credible set overall look stronger in UKB LD panel shown by darker color. When we add up the PIPs for all SNPs in the credible set identified by 1kg LD reference, the total PIP does not pass the default threshold. Thus, the fine-mapped results can be affected by the differences in LD reference panels. This differences may be sutble in this example, but more spurious results have been seen with mismatched LD references.</p>
<pre class="r"><code>ggplot(susie.1kg[[&quot;704&quot;]]$gwas, aes(x=pos,y=-log10(pval), color=susie.1kg[[&quot;704&quot;]]$LD.coef[789,])) +
       geom_point() + scale_color_gradient(low = alpha(&quot;yellow&quot;, 0.2), high =&quot;red&quot;) + 
       labs(color=&quot;LD coefficient&quot;) + labs(title = &quot;1kg&quot;) + 
       ylab(&quot;-log10(p-val)&quot;) +  xlab(&quot;variable&quot;) </code></pre>
<p><img src="figure/lab4.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(gwas[[&quot;704&quot;]], aes(x=pos,y=-log10(pval), color=bigSNP[[&quot;704&quot;]]$LD_matrix[gwas[[&quot;704&quot;]][788,]$bigSNP_index, gwas[[&quot;704&quot;]]$bigSNP_index])) +
       geom_point() + scale_color_gradient(low = alpha(&quot;yellow&quot;, 0.2), high =&quot;red&quot;) + 
       labs(color=&quot;LD coefficient&quot;) + labs(title = &quot;UKB&quot;) + 
       ylab(&quot;-log10(p-val)&quot;) +  xlab(&quot;variable&quot;) </code></pre>
<p><img src="figure/lab4.Rmd/unnamed-chunk-15-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.4 (2021-02-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.3     bigsnpr_1.8.1     bigstatsr_1.5.1   dplyr_1.0.4      
[5] data.table_1.14.2 susieR_0.11.92   

loaded via a namespace (and not attached):
 [1] tidyselect_1.1.1   xfun_0.21          bslib_0.2.4        purrr_0.3.4       
 [5] lattice_0.20-41    bigassertr_0.1.4   colorspace_2.0-2   vctrs_0.3.8       
 [9] generics_0.1.0     htmltools_0.5.1.1  yaml_2.2.1         utf8_1.2.2        
[13] rlang_0.4.11       mixsqp_0.3-43      jquerylib_0.1.3    later_1.1.0.1     
[17] pillar_1.5.0       withr_2.4.2        glue_1.4.2         DBI_1.1.1         
[21] matrixStats_0.58.0 foreach_1.5.1      lifecycle_1.0.0    plyr_1.8.6        
[25] stringr_1.4.0      munsell_0.5.0      gtable_0.3.0       workflowr_1.6.2   
[29] codetools_0.2-18   evaluate_0.14      labeling_0.4.2     knitr_1.31        
[33] doParallel_1.0.16  httpuv_1.5.5       parallel_4.0.4     irlba_2.3.3       
[37] fansi_0.5.0        highr_0.8          Rcpp_1.0.7         promises_1.2.0.1  
[41] scales_1.1.1       bigsparser_0.4.4   jsonlite_1.7.2     farver_2.1.0      
[45] fs_1.5.0           digest_0.6.27      bigparallelr_0.3.1 stringi_1.5.3     
[49] cowplot_1.1.1      grid_4.0.4         rprojroot_2.0.2    tools_4.0.4       
[53] magrittr_2.0.1     sass_0.3.1         tibble_3.0.6       crayon_1.4.1      
[57] pkgconfig_2.0.3    ellipsis_0.3.2     Matrix_1.3-4       assertthat_0.2.1  
[61] rmarkdown_2.7      reshape_0.8.8      iterators_1.0.13   R6_2.5.1          
[65] flock_0.7          git2r_0.28.0       compiler_4.0.4    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
