<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>PTR:RNA-modifications</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">funcFinemapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">PTR:RNA-modifications</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>funcFinemapping/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210404code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210404code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210404)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtreeba2c2b6ca1323e7df30c0257e57e6e771cc9a094targetblankba2c2b6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/funcFinemapping/tree/ba2c2b6ca1323e7df30c0257e57e6e771cc9a094" target="_blank">ba2c2b6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtreeba2c2b6ca1323e7df30c0257e57e6e771cc9a094targetblankba2c2b6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/funcFinemapping/tree/ba2c2b6ca1323e7df30c0257e57e6e771cc9a094" target="_blank">ba2c2b6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/PTR_m6A.nb.html
    Ignored:    analysis/build_annotations_for_single_cell_data.nb.html
    Ignored:    analysis/lab4_prepare.nb.html
    Ignored:    analysis/ldsc_results.nb.html
    Ignored:    analysis/learn_archR.nb.html
    Ignored:    analysis/mtsplice_finemapping_results.nb.html
    Ignored:    analysis/results.nb.html
    Ignored:    analysis/snp_finemapping_results.nb.html
    Ignored:    analysis/splicing.nb.html
    Ignored:    analysis/susie_tutorial.nb.html

Untracked files:
    Untracked:  SCZ_pval_vs_MAF.png
    Untracked:  SNPs_categories,png
    Untracked:  SNPs_categories.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/PTR_m6A.Rmd
    Untracked:  analysis/PTR_splicing_mtsplice.Rmd
    Untracked:  analysis/PTR_splicing_spliceAI.Rmd
    Untracked:  analysis/explore_mtsplice.Rmd
    Untracked:  analysis/snp_finemapping_spliceAI_prior.Rmd
    Untracked:  analysis/snp_finemapping_with_priors.Rmd
    Untracked:  bmi_locus1410.pdf
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/.snakemake/
    Untracked:  code/Euro_LD_Chunks.RData
    Untracked:  code/Snakefile
    Untracked:  code/config.yaml
    Untracked:  code/environment.yml
    Untracked:  code/extract_mmsplice_predictions_bychr.R
    Untracked:  code/extract_mtsplice_maxPred.sh
    Untracked:  code/get_indiv_annotation_from_predictions.R
    Untracked:  code/heatmap_torus_enrichment.ipynb
    Untracked:  code/ldsc.log
    Untracked:  code/ldsc.results
    Untracked:  code/ldsc_regression.sh
    Untracked:  code/make_joint_annotations.R
    Untracked:  code/make_plots.R
    Untracked:  code/out/
    Untracked:  code/plot_torus_enrichment.R
    Untracked:  code/prepare_torus_inputs.R
    Untracked:  code/run_ldsc.sh
    Untracked:  code/run_ldsc_with_bed.sh
    Untracked:  code/run_ldsc_with_bed_v2.sh
    Untracked:  code/run_susie.R
    Untracked:  code/run_susie_in_parallel.R
    Untracked:  code/run_torus.sh
    Untracked:  code/run_torus/
    Untracked:  code/sctype/
    Untracked:  code/slurm-22619162.out
    Untracked:  code/slurm-22619163.out
    Untracked:  code/slurm-22619164.out
    Untracked:  code/slurm-22619165.out
    Untracked:  code/slurm-22619166.out
    Untracked:  code/slurm-22619199.out
    Untracked:  code/slurm-22619200.out
    Untracked:  code/slurm-22619201.out
    Untracked:  code/slurm-22619202.out
    Untracked:  code/slurm-22619203.out
    Untracked:  code/slurm-22619228.out
    Untracked:  code/slurm-22619229.out
    Untracked:  code/slurm-22619230.out
    Untracked:  code/slurm-22619231.out
    Untracked:  code/slurm-22619232.out
    Untracked:  code/slurm-22619234.out
    Untracked:  code/slurm-22619235.out
    Untracked:  code/slurm-22619236.out
    Untracked:  code/slurm-22619237.out
    Untracked:  code/slurm-22619238.out
    Untracked:  code/slurm-22619289.out
    Untracked:  code/slurm-22619290.out
    Untracked:  code/split_vcf.sh
    Untracked:  code/test_mtsplice.txt.gz
    Untracked:  code_backup/
    Untracked:  data/ASD_denovo_variants.1.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.10.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.11.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.12.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.13.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.14.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.15.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.16.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.17.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.18.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.19.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.2.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.20.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.21.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.22.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.3.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.4.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.5.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.6.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.7.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.8.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.9.sorted.vcf.gz
    Untracked:  data/ASD_denovo_variants.vcf.gz
    Untracked:  data/ScTypeDB_full.xlsx
    Untracked:  data/compare_spliceai_and_mmsplice.RData
    Untracked:  data/header.txt.gz
    Untracked:  data/hg19_gtf_genomic_annots_ver2.gr.rds
    Untracked:  data/mmsplice_mtsplice_cutoffs.M5.txt
    Untracked:  data/mmsplice_mtsplice_cutoffs.txt
    Untracked:  data/num_overlaps_finemapped_SNPs_and_ctcf.txt
    Untracked:  data/process_vcf.sh
    Untracked:  data/qqplot_SNPs_high_spliceAI.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_SCZ.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_aFib.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_allergy.png
    Untracked:  data/spliceAIandMAF.txt.gz
    Untracked:  data/torus_enrichment_novel_annot.est
    Untracked:  data/torus_joint_enrichment.est
    Untracked:  data/torus_joint_refined_enrichment.est
    Untracked:  data/vcf/
    Untracked:  enhancer_gene_feature.rmd
    Untracked:  fig1_panels.pdf
    Untracked:  fig2.pdf
    Untracked:  fig_panel2.pdf
    Untracked:  gene_mapping.pdf
    Untracked:  output/AAD/GMP_merge_stats.txt
    Untracked:  output/AAD/Wang2020_joint.results
    Untracked:  output/AAD/Wang2020_joint_T.results
    Untracked:  output/AAD/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/allergy/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/allergy/Ulirsch2019_disjoint_snps.sumstats
    Untracked:  output/AAD/allergy/Wang2020_T_subsets.est
    Untracked:  output/AAD/allergy/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/allergy/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/allergy/Wang2020_joint_T.results
    Untracked:  output/AAD/allergy/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/allergy/Wang2020_tissueResT.est
    Untracked:  output/AAD/allergy/torus_enrichment_CD4.est
    Untracked:  output/AAD/allergy/torus_enrichment_CD8.est
    Untracked:  output/AAD/allergy/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD4_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD8_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_subsets.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/asthma_adult/Wang2020_joint_T.results
    Untracked:  output/AAD/asthma_adult/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/asthma_adult/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/AAD/asthma_child/CD4_compare.est
    Untracked:  output/AAD/asthma_child/CD8_compare.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_subsets.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/asthma_child/Wang2020_joint_T.results
    Untracked:  output/AAD/asthma_child/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/asthma_child/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/LDL_ukb_L10.gif
    Untracked:  output/LDL_ukb_L10.pdf
    Untracked:  output/background_SNPs_annotated_percent.txt
    Untracked:  output/ldsc
    Untracked:  output/locus_1452.gif
    Untracked:  output/locus_1452.pdf
    Untracked:  output/spliceAI_vs_MAF.png
    Untracked:  output/splicing/.ipynb_checkpoints/
    Untracked:  output/splicing/PTR_across_traits_annotations.results
    Untracked:  output/splicing/PTR_joint.pdf
    Untracked:  output/splicing/PTR_joint.png
    Untracked:  output/splicing/QQplot_mmsplice.png
    Untracked:  output/splicing/QQplot_mmsplice_top15.png
    Untracked:  output/splicing/QQplot_mmsplice_vs_mtsplice.png
    Untracked:  output/splicing/QQplot_mmsplice_vs_mtsplice_SCZ.png
    Untracked:  output/splicing/TSplice_scores_distribution.png
    Untracked:  output/splicing/aFib_joint_comparison.pdf
    Untracked:  output/splicing/allergy_joint_comparison.pdf
    Untracked:  output/splicing/header.txt
    Untracked:  output/splicing/m6A_enrichment_across_traits.pdf
    Untracked:  output/splicing/mmsplice_vs_mtsplice_AA_heart.png
    Untracked:  output/splicing/mmsplice_vs_mtsplice_Hypothalamus_brain.png
    Untracked:  output/splicing/mmsplice_vs_mtsplice_LV_heart.png
    Untracked:  output/splicing/mmsplice_vs_mtsplice_across_tissues.png
    Untracked:  output/splicing/mtsplice_enrichment_across_traits.pdf
    Untracked:  output/splicing/prior/
    Untracked:  output/splicing/scz_PTR_annotations.results
    Untracked:  output/splicing/scz_joint_comparison.pdf
    Untracked:  output/splicing/scz_neuOCR_m6a_DMR.results
    Untracked:  output/splicing/scz_spliceAI0.03_hist.png
    Untracked:  output/splicing/scz_spliceAI0.03_scatterplot.png
    Untracked:  output/splicing/scz_spliceai_binary0.03.results
    Untracked:  output/splicing/summarizing_aFib_PIPs.pdf
    Untracked:  output/splicing/torus_afib_spliceai.est
    Untracked:  output/splicing/torus_annotations_spliceai0.01.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.03.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.05.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.07.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.1.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.2.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai9.txt.gz
    Untracked:  output/splicing/torus_enrichment_joint_scz_mtsplice0.6_hypothalamus-brain.est
    Untracked:  output/splicing/torus_enrichment_joint_scz_spliceAI.est
    Untracked:  output/splicing/torus_spliceai0.01.enrichment
    Untracked:  output/splicing/torus_spliceai0.03.enrichment
    Untracked:  output/splicing/torus_spliceai0.05.enrichment
    Untracked:  output/splicing/torus_spliceai0.07.enrichment
    Untracked:  output/splicing/torus_spliceai0.1.enrichment
    Untracked:  output/splicing/torus_spliceai0.2.enrichment
    Untracked:  output/splicing/torus_zscores.txt.gz
    Untracked:  output/torus
    Untracked:  panel_figure2.pdf
    Untracked:  test.txt

Unstaged changes:
    Deleted:    .Rprofile
    Modified:   analysis/index.Rmd
    Modified:   analysis/lab4_prepare.Rmd
    Modified:   analysis/ldsc_PTR_results.Rmd
    Deleted:    output/AAD/Caldero2019_disjoint_snps.sumstats
    Modified:   output/AAD/allergy/Caldero2019_disjoint_snps.sumstats
    Modified:   output/AAD/allergy/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/allergy/Wang2020_indiv.est
    Modified:   output/AAD/allergy/Wang2020_joint.results
    Deleted:    output/AAD/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/AAD/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/AAD/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/AAD/asthma/diffe_adult_blood_torus.est
    Deleted:    output/AAD/asthma/joint_blood_immune_rest_vs_stimu.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/AAD/asthma/lung_clusters_dict.txt
    Deleted:    output/AAD/asthma/lung_clusters_info.txt
    Deleted:    output/AAD/asthma/stimu_adult_blood_torus.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_rest.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_stimulated.est
    Deleted:    output/AAD/asthma/zhang2021_annot_percent.txt
    Deleted:    output/AAD/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/AAD/asthma/zhang2021_peaks_per_celltype.txt
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD4_compare.est
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD8_compare.est
    Deleted:    output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/asthma_adult/Wang2020_indiv.est
    Modified:   output/AAD/asthma_adult/Wang2020_joint.results
    Modified:   output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/asthma_child/Wang2020_indiv.est
    Modified:   output/AAD/asthma_child/Wang2020_joint.results
    Deleted:    output/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/asthma/diffe_adult_blood_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/asthma/lung_clusters_dict.txt
    Deleted:    output/asthma/lung_clusters_info.txt
    Deleted:    output/asthma/stimu_adult_blood_torus.est
    Deleted:    output/asthma/zhang2021_annot_percent.txt
    Deleted:    output/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/asthma/zhang2021_peaks_per_celltype.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/PTR_m6A.Rmd</code>) and HTML
(<code>docs/PTR_m6A.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/05a300925c8875e96215026e8c7533a41c09f236/docs/PTR_m6A.html" target="_blank">05a3009</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
<td>
PTR_m6A
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    between, first, last</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<div id="backgrounds" class="section level2">
<h2>Backgrounds</h2>
<p>Post-transcriptional regulatory (PTR) processes have been implicated
in development and diseases, however, it is largely unknown how genetic
variations are mediated through PTR processes. We propose to annotate
GWAS variants using both experimental measurements and computational
predictions. With this prior knowledge, we can further identify most
likely causal variants through fine-mapping and then link them to
genes.</p>
<p>Several post-transcriptonal features will be explored:</p>
<ul>
<li>alternative splicing</li>
<li><strong>RNA modification: m6A</strong></li>
<li>RNA binding</li>
<li>Polyadenylation</li>
</ul>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="ldsc-enrichment-across-multiple-traits" class="section level3">
<h3>LDSC enrichment across multiple traits</h3>
<p><img src="figure/PTR_m6A.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/05a300925c8875e96215026e8c7533a41c09f236/docs/figure/PTR_m6A.Rmd/unnamed-chunk-3-1.png" target="_blank">05a3009</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/PTR_m6A.Rmd/unnamed-chunk-4-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/05a300925c8875e96215026e8c7533a41c09f236/docs/figure/PTR_m6A.Rmd/unnamed-chunk-4-1.png" target="_blank">05a3009</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="adjusting-for-transcript-level-variations-across-tissues"
class="section level3">
<h3>Adjusting for transcript-level variations across tissues</h3>
<p>We compared the enrichment estimates for m6A peaks in fetal tissues
conditional on all coding regions or coding regions from differentially
expressed genes in each tissue.</p>
<p><strong>Fetal cell atlas for transcriptomics</strong> Cao et
al.Science 2020:A human cell atlas of fetal gene expression.</p>
<p>The list DE genes obtained from DEG analysis on pseudobulk
transcriptomes, comparing the most expressed tissue and the second most
expressed tissue.</p>
<p>For heart tissue, 1011 genes show significantly higher expression
than that in the second mostly expressed tissue at FDR 0.05.</p>
<p><img src="figure/PTR_m6A.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>Loading required package: GenomicRanges</code></pre>
<pre><code>Loading required package: stats4</code></pre>
<pre><code>Loading required package: BiocGenerics</code></pre>
<pre><code>
Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    combine, intersect, setdiff, union</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    IQR, mad, sd, var, xtabs</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    Filter, Find, Map, Position, Reduce, anyDuplicated, append,
    as.data.frame, basename, cbind, colnames, dirname, do.call,
    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,
    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,
    tapply, union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>Loading required package: S4Vectors</code></pre>
<pre><code>
Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    first, rename</code></pre>
<pre><code>The following objects are masked from &#39;package:data.table&#39;:

    first, second</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    I, expand.grid, unname</code></pre>
<pre><code>Loading required package: IRanges</code></pre>
<pre><code>
Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    collapse, desc, slice</code></pre>
<pre><code>The following object is masked from &#39;package:data.table&#39;:

    shift</code></pre>
<pre><code>Loading required package: GenomeInfoDb</code></pre>
<pre><code>Warning: replacing previous import &#39;utils::download.file&#39; by
&#39;restfulr::download.file&#39; when loading &#39;rtracklayer&#39;</code></pre>
<p><strong>Joint analysis via LDSC</strong></p>
<p>Note: The extended region of $$250 bp away from the coding regions
was included to adjust for the unsure boundary.
<img src="figure/PTR_m6A.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" />
7.8% of GWAS SNPs are within coding and the extended regions.
<img src="figure/PTR_m6A.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" />
Only 0.2% of GWAS SNPs are within heart-specific DEGs among fetal
samples.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
[1] GenomicRanges_1.48.0 GenomeInfoDb_1.32.2  IRanges_2.30.1      
[4] S4Vectors_0.34.0     BiocGenerics_0.42.0  dplyr_1.0.9         
[7] data.table_1.14.2    ggplot2_3.3.6       

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3                lattice_0.20-45            
 [3] Rsamtools_2.12.0            Biostrings_2.64.0          
 [5] assertthat_0.2.1            rprojroot_2.0.3            
 [7] digest_0.6.29               utf8_1.2.2                 
 [9] R6_2.5.1                    evaluate_0.15              
[11] highr_0.9                   pillar_1.7.0               
[13] zlibbioc_1.42.0             rlang_1.0.2                
[15] rstudioapi_0.13             whisker_0.4                
[17] jquerylib_0.1.4             Matrix_1.4-1               
[19] rmarkdown_2.14              labeling_0.4.2             
[21] plyranges_1.16.0            BiocParallel_1.30.3        
[23] stringr_1.4.0               RCurl_1.98-1.7             
[25] munsell_0.5.0               DelayedArray_0.22.0        
[27] rtracklayer_1.56.0          compiler_4.2.0             
[29] httpuv_1.6.5                xfun_0.30                  
[31] pkgconfig_2.0.3             htmltools_0.5.2            
[33] tidyselect_1.1.2            SummarizedExperiment_1.26.1
[35] tibble_3.1.7                GenomeInfoDbData_1.2.8     
[37] workflowr_1.7.0             codetools_0.2-18           
[39] matrixStats_0.62.0          XML_3.99-0.9               
[41] fansi_1.0.3                 crayon_1.5.1               
[43] withr_2.5.0                 later_1.3.0                
[45] GenomicAlignments_1.32.0    bitops_1.0-7               
[47] grid_4.2.0                  jsonlite_1.8.0             
[49] gtable_0.3.0                lifecycle_1.0.1            
[51] DBI_1.1.2                   git2r_0.30.1               
[53] magrittr_2.0.3              scales_1.2.0               
[55] cli_3.3.0                   stringi_1.7.6              
[57] farver_2.1.0                XVector_0.36.0             
[59] fs_1.5.2                    promises_1.2.0.1           
[61] bslib_0.3.1                 ellipsis_0.3.2             
[63] generics_0.1.2              vctrs_0.4.1                
[65] rjson_0.2.21                restfulr_0.0.14            
[67] tools_4.2.0                 Biobase_2.56.0             
[69] glue_1.6.2                  purrr_0.3.4                
[71] MatrixGenerics_1.8.0        parallel_4.2.0             
[73] fastmap_1.1.0               yaml_2.3.5                 
[75] colorspace_2.0-3            knitr_1.39                 
[77] BiocIO_1.6.0                sass_0.4.1                 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
