<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Predicting splicing effects with MTSplice</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">funcFinemapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Predicting splicing effects with
MTSplice</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>funcFinemapping/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210404code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210404code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210404)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/projects/funcFinemapping/docs/assets/splicing/compare_MAFs_mtsplice_scz_top1-2.png
</td>
<td style="text-align:left;">
docs/assets/splicing/compare_MAFs_mtsplice_scz_top1-2.png
</td>
</tr>
<tr>
<td style="text-align:left;">
~/projects/funcFinemapping/output/splicing/PTR_joint.pdf
</td>
<td style="text-align:left;">
output/splicing/PTR_joint.pdf
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtreeab4948d39589a1263eb3989c6bfcc793639c53b3targetblankab4948da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/j3gu/funcFinemapping/tree/ab4948d39589a1263eb3989c6bfcc793639c53b3" target="_blank">ab4948d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtreeab4948d39589a1263eb3989c6bfcc793639c53b3targetblankab4948da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/j3gu/funcFinemapping/tree/ab4948d39589a1263eb3989c6bfcc793639c53b3" target="_blank">ab4948d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/PTR_m6A.nb.html
    Ignored:    analysis/build_annotations_for_single_cell_data.nb.html
    Ignored:    analysis/lab4_prepare.nb.html
    Ignored:    analysis/ldsc_results.nb.html
    Ignored:    analysis/learn_archR.nb.html
    Ignored:    analysis/mtsplice_finemapping_results.nb.html
    Ignored:    analysis/results.nb.html
    Ignored:    analysis/snp_finemapping_results.nb.html
    Ignored:    analysis/splicing.nb.html
    Ignored:    analysis/susie_tutorial.nb.html

Untracked files:
    Untracked:  SCZ_pval_vs_MAF.png
    Untracked:  SNPs_categories,png
    Untracked:  SNPs_categories.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/PTR_m6A.Rmd
    Untracked:  analysis/PTR_splicing_mtsplice.Rmd
    Untracked:  analysis/PTR_splicing_spliceAI.Rmd
    Untracked:  analysis/snp_finemapping_spliceAI_prior.Rmd
    Untracked:  bmi_locus1410.pdf
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/.snakemake/
    Untracked:  code/Euro_LD_Chunks.RData
    Untracked:  code/Snakefile
    Untracked:  code/config.yaml
    Untracked:  code/environment.yml
    Untracked:  code/extract_mmsplice_predictions_bychr.R
    Untracked:  code/extract_mtsplice_maxPred.sh
    Untracked:  code/get_indiv_annotation_from_predictions.R
    Untracked:  code/heatmap_torus_enrichment.ipynb
    Untracked:  code/ldsc.log
    Untracked:  code/ldsc.results
    Untracked:  code/ldsc_regression.sh
    Untracked:  code/make_joint_annotations.R
    Untracked:  code/make_plots.R
    Untracked:  code/out/
    Untracked:  code/plot_torus_enrichment.R
    Untracked:  code/prepare_torus_inputs.R
    Untracked:  code/run_ldsc.sh
    Untracked:  code/run_ldsc_with_bed.sh
    Untracked:  code/run_ldsc_with_bed_v2.sh
    Untracked:  code/run_susie.R
    Untracked:  code/run_susie_in_parallel.R
    Untracked:  code/run_torus.sh
    Untracked:  code/run_torus/
    Untracked:  code/sctype/
    Untracked:  code/split_vcf.sh
    Untracked:  code/test_mtsplice.txt.gz
    Untracked:  code_backup/
    Untracked:  data/ScTypeDB_full.xlsx
    Untracked:  data/compare_spliceai_and_mmsplice.RData
    Untracked:  data/hg19_gtf_genomic_annots_ver2.gr.rds
    Untracked:  data/mmsplice_mtsplice_cutoffs.M5.txt
    Untracked:  data/mmsplice_mtsplice_cutoffs.txt
    Untracked:  data/num_overlaps_finemapped_SNPs_and_ctcf.txt
    Untracked:  data/qqplot_SNPs_high_spliceAI.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_SCZ.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_aFib.png
    Untracked:  data/qqplot_SNPs_high_spliceAI_scores_allergy.png
    Untracked:  data/spliceAIandMAF.txt.gz
    Untracked:  data/torus_enrichment_novel_annot.est
    Untracked:  data/torus_joint_enrichment.est
    Untracked:  data/torus_joint_refined_enrichment.est
    Untracked:  enhancer_gene_feature.rmd
    Untracked:  fig1_panels.pdf
    Untracked:  fig2.pdf
    Untracked:  fig_panel2.pdf
    Untracked:  gene_mapping.pdf
    Untracked:  output/AAD/GMP_merge_stats.txt
    Untracked:  output/AAD/Wang2020_joint.results
    Untracked:  output/AAD/Wang2020_joint_T.results
    Untracked:  output/AAD/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/allergy/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/allergy/Ulirsch2019_disjoint_snps.sumstats
    Untracked:  output/AAD/allergy/Wang2020_T_subsets.est
    Untracked:  output/AAD/allergy/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/allergy/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/allergy/Wang2020_joint_T.results
    Untracked:  output/AAD/allergy/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/allergy/Wang2020_tissueResT.est
    Untracked:  output/AAD/allergy/torus_enrichment_CD4.est
    Untracked:  output/AAD/allergy/torus_enrichment_CD8.est
    Untracked:  output/AAD/allergy/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/allergy/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD4_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD8_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_subsets.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/asthma_adult/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/asthma_adult/Wang2020_joint_T.results
    Untracked:  output/AAD/asthma_adult/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/asthma_adult/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/asthma_adult/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/AAD/asthma_child/CD4_compare.est
    Untracked:  output/AAD/asthma_child/CD8_compare.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_subsets.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_subsets_indiv.est
    Untracked:  output/AAD/asthma_child/Wang2020_T_tissueRes.est
    Untracked:  output/AAD/asthma_child/Wang2020_joint_T.results
    Untracked:  output/AAD/asthma_child/Wang2020_joint_tissueResT.results
    Untracked:  output/AAD/asthma_child/torus_enrichment_CD4.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_CD8.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_non_tissueRes_T.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueMigraT.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResT_C6.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResT_C8.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueRes_T.est
    Untracked:  output/AAD/asthma_child/torus_enrichment_tissueResident_T_cells.est
    Untracked:  output/LDL_ukb_L10.gif
    Untracked:  output/LDL_ukb_L10.pdf
    Untracked:  output/background_SNPs_annotated_percent.txt
    Untracked:  output/ldsc
    Untracked:  output/locus_1452.gif
    Untracked:  output/locus_1452.pdf
    Untracked:  output/spliceAI_vs_MAF.png
    Untracked:  output/splicing/.ipynb_checkpoints/
    Untracked:  output/splicing/PTR_across_traits_annotations.results
    Untracked:  output/splicing/PTR_joint.pdf
    Untracked:  output/splicing/PTR_joint.png
    Untracked:  output/splicing/QQplot_mmsplice.png
    Untracked:  output/splicing/QQplot_mmsplice_top15.png
    Untracked:  output/splicing/QQplot_mmsplice_vs_mtsplice.png
    Untracked:  output/splicing/QQplot_mmsplice_vs_mtsplice_SCZ.png
    Untracked:  output/splicing/TSplice_scores_distribution.png
    Untracked:  output/splicing/aFib_joint_comparison.pdf
    Untracked:  output/splicing/allergy_joint_comparison.pdf
    Untracked:  output/splicing/header.txt
    Untracked:  output/splicing/m6A_enrichment_across_traits.pdf
    Untracked:  output/splicing/mmsplice_vs_mtsplice_AA_heart.png
    Untracked:  output/splicing/mmsplice_vs_mtsplice_Hypothalamus_brain.png
    Untracked:  output/splicing/mmsplice_vs_mtsplice_LV_heart.png
    Untracked:  output/splicing/mmsplice_vs_mtsplice_across_tissues.png
    Untracked:  output/splicing/mtsplice_enrichment_across_traits.pdf
    Untracked:  output/splicing/prior/
    Untracked:  output/splicing/scz_PTR_annotations.results
    Untracked:  output/splicing/scz_joint_comparison.pdf
    Untracked:  output/splicing/scz_neuOCR_m6a_DMR.results
    Untracked:  output/splicing/scz_spliceAI0.03_hist.png
    Untracked:  output/splicing/scz_spliceAI0.03_scatterplot.png
    Untracked:  output/splicing/scz_spliceai_binary0.03.results
    Untracked:  output/splicing/summarizing_aFib_PIPs.pdf
    Untracked:  output/splicing/torus_afib_spliceai.est
    Untracked:  output/splicing/torus_annotations_spliceai0.01.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.03.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.05.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.07.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.1.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai0.2.txt.gz
    Untracked:  output/splicing/torus_annotations_spliceai9.txt.gz
    Untracked:  output/splicing/torus_enrichment_joint_scz_mtsplice0.6_hypothalamus-brain.est
    Untracked:  output/splicing/torus_enrichment_joint_scz_spliceAI.est
    Untracked:  output/splicing/torus_spliceai0.01.enrichment
    Untracked:  output/splicing/torus_spliceai0.03.enrichment
    Untracked:  output/splicing/torus_spliceai0.05.enrichment
    Untracked:  output/splicing/torus_spliceai0.07.enrichment
    Untracked:  output/splicing/torus_spliceai0.1.enrichment
    Untracked:  output/splicing/torus_spliceai0.2.enrichment
    Untracked:  output/splicing/torus_zscores.txt.gz
    Untracked:  output/torus
    Untracked:  panel_figure2.pdf
    Untracked:  test.txt

Unstaged changes:
    Deleted:    .Rprofile
    Modified:   analysis/index.Rmd
    Modified:   analysis/lab4_prepare.Rmd
    Modified:   analysis/ldsc_PTR_results.Rmd
    Deleted:    output/AAD/Caldero2019_disjoint_snps.sumstats
    Modified:   output/AAD/allergy/Caldero2019_disjoint_snps.sumstats
    Modified:   output/AAD/allergy/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/allergy/Wang2020_indiv.est
    Modified:   output/AAD/allergy/Wang2020_joint.results
    Deleted:    output/AAD/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/AAD/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/AAD/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/AAD/asthma/diffe_adult_blood_torus.est
    Deleted:    output/AAD/asthma/joint_blood_immune_rest_vs_stimu.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/AAD/asthma/lung_clusters_dict.txt
    Deleted:    output/AAD/asthma/lung_clusters_info.txt
    Deleted:    output/AAD/asthma/stimu_adult_blood_torus.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_rest.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_stimulated.est
    Deleted:    output/AAD/asthma/zhang2021_annot_percent.txt
    Deleted:    output/AAD/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/AAD/asthma/zhang2021_peaks_per_celltype.txt
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD4_compare.est
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD8_compare.est
    Deleted:    output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/asthma_adult/Wang2020_indiv.est
    Modified:   output/AAD/asthma_adult/Wang2020_joint.results
    Modified:   output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/asthma_child/Wang2020_indiv.est
    Modified:   output/AAD/asthma_child/Wang2020_joint.results
    Deleted:    output/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/asthma/diffe_adult_blood_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/asthma/lung_clusters_dict.txt
    Deleted:    output/asthma/lung_clusters_info.txt
    Deleted:    output/asthma/stimu_adult_blood_torus.est
    Deleted:    output/asthma/zhang2021_annot_percent.txt
    Deleted:    output/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/asthma/zhang2021_peaks_per_celltype.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/PTR_splicing_mtsplice.Rmd</code>)
and HTML (<code>docs/PTR_splicing_mtsplice.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/PTR_splicing_mtsplice.html" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
<td>
update MTSplice enrichment results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/171a0dbca399ed126abff9d132d924451a454c17/docs/PTR_splicing_mtsplice.html" target="_blank">171a0db</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-17
</td>
<td>
PTR_mtsplice
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/10deffed239e4acf3ae3776e4229b4c177d66981/docs/PTR_splicing_mtsplice.html" target="_blank">10deffe</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-15
</td>
<td>
PTR_mtsplice
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/aa0b52db2ecbe64556ee21cfba61da626151c5ec/docs/PTR_splicing_mtsplice.html" target="_blank">aa0b52d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-10
</td>
<td>
PTR_mtsplice
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/03e831fdd5f5f4ee800608286a9c629a07f4dca5/docs/PTR_splicing_mtsplice.html" target="_blank">03e831f</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
<td>
PTR_splicing
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/abda215c4d1a7632a96916798c0d9cb5cf58066a/docs/PTR_splicing_mtsplice.html" target="_blank">abda215</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
<td>
PTR_mtsplice
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="backgrounds" class="section level2">
<h2>Backgrounds</h2>
<p>Post-transcriptional regulatory (PTR) processes have been implicated
in development and diseases, however, it is largely unknown how genetic
variations are mediated through PTR processes. We propose to annotate
GWAS variants using both experimental measurements and computational
predictions. With this prior knowledge, we can further identify most
likely causal variants through fine-mapping and then link them to
genes.</p>
<p>Several post-transcriptonal features will be explored:</p>
<ul>
<li><strong>Alternative splicing</strong></li>
<li>RNA modification: m6A</li>
<li>RNA binding</li>
<li>Polyadenylation</li>
</ul>
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<p><strong>MMSplice/MTSplice</strong><br />
A CNN method that aims to provide tissue-average or tissue-specific
combined with the average predictions on how likely a given variant can
alter splicing patterns based upon its sequence alone.</p>
<p><strong>SNP effect predictions</strong><br />
The predictions of variant effects on post-transcriptional regulation
were performed on 10 million SNPs after some QC criteria, from 1000
genome phase 3 project.</p>
<p><strong>Enrichment analysis</strong><br />
We first tested annotations one at a time using both TORUS and LDSC.Then
we jointly assessed a set of annotations.</p>
<p>Notably, we ran Torus without being limited to 1M hapmap SNPs.
However, all analyses with LDSC were limited to 1M hapmap SNPs.</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="characteristics-for-mtsplice-predictions"
class="section level3">
<h3>Characteristics for MTSplice predictions</h3>
<p>Out of 10M SNPs (MAF&gt;=0.005), around 2000 SNPs have predicted
score bigger than 2 or smaller than -2, which can be considered
strong.</p>
<p><strong>Compare MMSplice and MTSplice predictions </strong></p>
<div class="figure">
<img
src="assets/splicing/mmsplice_vs_mtsplice_predictions_across_tissues.png"
alt="" />
<p class="caption">compare predictions</p>
</div>
<p><strong>Table of prediction cutoffs</strong></p>
<p><strong>Understand the distribution of predicted scores wrt.
MAFs</strong></p>
<p><img src="assets/splicing/compare_MAFs_mtsplice_scz_top1-2.png"
alt="MAFs distribution" /> <strong>Examine MAF distribution for SNPs
above or below threshold</strong></p>
<p><strong>QQ Plots - comparing SNPs within annotations and the
rest</strong></p>
<p>Color scheme: * red - tissue specific * blue - mmsplice * black -
background</p>
<p>Top-left: SCZ gwas<br />
Top-right and Down-left: aFib gwas; AA - Atrial Appendage; LV -Left
ventrical</p>
<div class="figure">
<img src="assets/splicing/QQplot_mmsplice_vs_mtsplice_top1.png"
alt="" />
<p class="caption">QQ plots for both SCZ and aFib traits</p>
</div>
</div>
<div id="enrichment-analysis-on-scz-gwas" class="section level3">
<h3>Enrichment analysis on SCZ GWAS</h3>
<p><em>Legends for the plots</em>:</p>
<ul>
<li>y-axis - annotations(% of SNPs within annotations)</li>
<li>x-axis - fold of enrichment</li>
<li>label on the plot - percent of SNP heritability or enrichment
p-value</li>
<li>dashed line - no enrichment</li>
</ul>
<div id="evaluate-features-with-ldsc" class="section level4">
<h4>Evaluate features with LDSC</h4>
<p>The defined set of baseline annotations for running LDSC are coding,
promoter, 3’UTR, 5’UTR, each with a 500-bp extended region.</p>
<p><em>MMSplice vs. Brain-specific MTSplice</em></p>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-10-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/171a0dbca399ed126abff9d132d924451a454c17/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-10-1.png" target="_blank">171a0db</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/10deffed239e4acf3ae3776e4229b4c177d66981/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-10-1.png" target="_blank">10deffe</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/abda215c4d1a7632a96916798c0d9cb5cf58066a/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-10-1.png" target="_blank">abda215</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 1.1 Enrichment of SCZ risk variants in tissue-average (mmsplice)
versus Hypothalamus-specific predictions for altering splicing patterns
(mtsplice) via LDSC.
</summary>
The variants with predicted score within top 5%, 10% and 15% were
compared for their enrichment of risk variants between the two methods.
</details>
<p><strong>Examine baseline annotations</strong></p>
<p>test annotation:top 5% Hypothalamus-specific MTSplice predictions</p>
<pre><code>Warning: Removed 1 rows containing missing values (geom_text).</code></pre>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-11-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/171a0dbca399ed126abff9d132d924451a454c17/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-11-1.png" target="_blank">171a0db</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/10deffed239e4acf3ae3776e4229b4c177d66981/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-11-1.png" target="_blank">10deffe</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/abda215c4d1a7632a96916798c0d9cb5cf58066a/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-11-1.png" target="_blank">abda215</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 2.1 Baseline enrichment of SCZ risk variants from LDSC.
</summary>
Conditional on coding, introns, promoters and UTR annotations, MTSplice
predictions specific to Hypothalams shows significant enrichment with
SCZ risk variants. The enrichment estimate of MTSplice prediction was
not affected much by whether or not including introns.
</details>
</div>
<div id="evaluate-features-with-torus" class="section level4">
<h4>Evaluate features with Torus</h4>
<p><strong>Individual run at different cutoffs</strong>
<img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-12-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/171a0dbca399ed126abff9d132d924451a454c17/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-12-1.png" target="_blank">171a0db</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/10deffed239e4acf3ae3776e4229b4c177d66981/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-12-1.png" target="_blank">10deffe</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/aa0b52db2ecbe64556ee21cfba61da626151c5ec/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-12-1.png" target="_blank">aa0b52d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/03e831fdd5f5f4ee800608286a9c629a07f4dca5/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-12-1.png" target="_blank">03e831f</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/abda215c4d1a7632a96916798c0d9cb5cf58066a/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-12-1.png" target="_blank">abda215</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 3.1 Enrichment of SCZ risk variants in individual annotation from
Torus.
</summary>
No baselines was provided for this run of analysis. For each of the two
features, we ran torus at different cutoffs, from top 2% to top 5%.
</details>
<p><strong>Joint run with other annotations</strong></p>
<p>Sequentially adding annotation one at a time with the MTSplice
prediction and the baseline set.
<img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-13-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/171a0dbca399ed126abff9d132d924451a454c17/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-13-1.png" target="_blank">171a0db</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/10deffed239e4acf3ae3776e4229b4c177d66981/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-13-1.png" target="_blank">10deffe</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/aa0b52db2ecbe64556ee21cfba61da626151c5ec/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-13-1.png" target="_blank">aa0b52d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 3.2 Enrichment of SCZ risk variants jointly run with baselines and
annotations of interest added one after another via Torus.
</summary>
After jointly run with adult brain OCRs, iPSC derived OCRs,
differentially methylated sites across brain regions,and brain m6A
sites, we observed slightly decrease in the enrichment of mtsplice
predictions. The biggest decrease occurs when brain differentially
methylated regions were added.
</details>
</div>
</div>
<div id="enrichment-analysis-on-afib-gwas" class="section level3">
<h3>Enrichment analysis on AFib GWAS</h3>
<div id="evaluate-features-with-ldsc-1" class="section level4">
<h4>Evaluate features with LDSC</h4>
<p><strong>AFib</strong></p>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-15-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/abda215c4d1a7632a96916798c0d9cb5cf58066a/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-15-1.png" target="_blank">abda215</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 4.1a-d Enrichment of aFib risk variants jointly run with baselines
with LDSC.
</summary>
Top 5% MTSplice predictions for left ventricle and atrial appendage
heart shows higher enrichment of risk variants with large standard
errors.
</details>
<p><strong>SCZ</strong></p>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-17-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/171a0dbca399ed126abff9d132d924451a454c17/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-17-1.png" target="_blank">171a0db</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/10deffed239e4acf3ae3776e4229b4c177d66981/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-17-1.png" target="_blank">10deffe</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/aa0b52db2ecbe64556ee21cfba61da626151c5ec/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-17-1.png" target="_blank">aa0b52d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>LDL/HDL</strong></p>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-19-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/171a0dbca399ed126abff9d132d924451a454c17/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-19-1.png" target="_blank">171a0db</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/10deffed239e4acf3ae3776e4229b4c177d66981/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-19-1.png" target="_blank">10deffe</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/aa0b52db2ecbe64556ee21cfba61da626151c5ec/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-19-1.png" target="_blank">aa0b52d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Asthma</strong></p>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-21-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/aa0b52db2ecbe64556ee21cfba61da626151c5ec/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-21-1.png" target="_blank">aa0b52d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="evaluate-features-with-torus-1" class="section level4">
<h4>Evaluate features with Torus</h4>
<p><strong>Torus - with the baseline set</strong> SNPs with MAF&gt;=0.05
<img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-22-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/171a0dbca399ed126abff9d132d924451a454c17/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-22-1.png" target="_blank">171a0db</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/10deffed239e4acf3ae3776e4229b4c177d66981/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-22-1.png" target="_blank">10deffe</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-08-15
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 4.2 Torus individual test on 8M aFib GWAS SNPs
</summary>
The Torus run shows top 5% of predictions average across tissues or
specific to blood or aorta artery show significant enrichment.
</details>
<p><strong>Torus joint analysis</strong><br />
TBD - top 5% predictions</p>
<details>
<summary>
Fig 4.2 Torus joint run on aFib GWAS SNPs with MAF&gt;=5%
</summary>
The Torus run shows top 5% of predictions average across tissues or
specific to blood or aorta artery show significant enrichment.
</details>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-24-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 4.3 Torus joint run on COA GWAS SNPs with MAF&gt;=5%
</summary>
SplcieAI at above 0.05 threshold shows enrichment after conditional on
other annotations and baselines.
</details>
<p><strong>Torus - Joint analysis with other annotations</strong>
<img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-25-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 4.4 Torus joint run with other types of annotations for AA-specific
MTSplice predictions
</summary>
With CM-specific OCRs and m6A sites, AA-MTSplice predictions no longer
shows significant enrichment.
</details>
</div>
</div>
<div
id="enrichment-for-tissue-agnostic-predictions-across-multiple-traits"
class="section level3">
<h3>Enrichment for tissue-agnostic predictions across multiple
traits</h3>
<p><strong>Torus</strong>
<img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-26-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-26-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-26-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-2">
Past versions of unnamed-chunk-26-2.png
</button>
</p>
<div id="fig-unnamed-chunk-26-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-26-2.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 5.1 Examine the enrichment of MMSplice-predicted SNPs that may alter
splicing across traits
</summary>
All traits do not display signal of enrichment except for AOA and COA.
</details>
<p><strong>LDSC results across multiple traits</strong></p>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-28-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 5.2 Examine the enrichment of MMSplice predicted SNPs conditional on
spliceAI predictions via LDSC
</summary>
</details>
</div>
<div id="ptr-enrichment-conditional-on-transcriptional-features"
class="section level3">
<h3>PTR enrichment conditional on transcriptional features</h3>
<p><strong>scz</strong></p>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-30-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-30-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<details>
<summary>
Fig 6.1 Conditional on transcriptional effects, the enrichment of PTR
features remains.
</summary>
</details>
<p><strong>asthma</strong></p>
<p><strong>allergy</strong>
<img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-32-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-32-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>aFib</strong></p>
<p><img src="figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-34-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-34-1">
Past versions of unnamed-chunk-34-1.png
</button>
</p>
<div id="fig-unnamed-chunk-34-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ab4948d39589a1263eb3989c6bfcc793639c53b3/docs/figure/PTR_splicing_mtsplice.Rmd/unnamed-chunk-34-1.png" target="_blank">ab4948d</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-09-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_1.0.9       gridExtra_2.3     data.table_1.14.2 ggplot2_3.3.6    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3     pillar_1.7.0     compiler_4.2.0   bslib_0.3.1     
 [5] later_1.3.0      jquerylib_0.1.4  git2r_0.30.1     highr_0.9       
 [9] workflowr_1.7.0  tools_4.2.0      digest_0.6.29    gtable_0.3.0    
[13] jsonlite_1.8.0   evaluate_0.15    lifecycle_1.0.1  tibble_3.1.7    
[17] pkgconfig_2.0.3  rlang_1.0.2      DBI_1.1.2        cli_3.3.0       
[21] rstudioapi_0.13  yaml_2.3.5       xfun_0.30        fastmap_1.1.0   
[25] withr_2.5.0      stringr_1.4.0    knitr_1.39       generics_0.1.2  
[29] fs_1.5.2         vctrs_0.4.1      sass_0.4.1       tidyselect_1.1.2
[33] grid_4.2.0       rprojroot_2.0.3  glue_1.6.2       R6_2.5.1        
[37] fansi_1.0.3      rmarkdown_2.14   farver_2.1.0     purrr_0.3.4     
[41] magrittr_2.0.3   whisker_0.4      scales_1.2.0     promises_1.2.0.1
[45] ellipsis_0.3.2   htmltools_0.5.2  assertthat_0.2.1 colorspace_2.0-3
[49] httpuv_1.6.5     labeling_0.4.2   utf8_1.2.2       stringi_1.7.6   
[53] munsell_0.5.0    crayon_1.5.1    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
