<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>results</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">funcFinemapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">results</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>funcFinemapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210404code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210404code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210404)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtreea29395c4071992e029fb1f75669377c932e58fcctargetblanka29395ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/j3gu/funcFinemapping/tree/a29395c4071992e029fb1f75669377c932e58fcc" target="_blank">a29395c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtreea29395c4071992e029fb1f75669377c932e58fcctargetblanka29395ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/j3gu/funcFinemapping/tree/a29395c4071992e029fb1f75669377c932e58fcc" target="_blank">a29395c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    analysis/results.nb.html

Untracked files:
    Untracked:  analysis/feedback.Rmd
    Untracked:  analysis/notes.Rmd
    Untracked:  output/OCR_peaks_summary_stats.txt
    Untracked:  output/constrained_and_OPC.txt
    Untracked:  output/prop_constrained_variants_with_OCR.txt

Unstaged changes:
    Modified:   analysis/biology_bkg.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/method_bkg.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/results.Rmd</code>) and HTML (<code>docs/results.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/a29395c4071992e029fb1f75669377c932e58fcc/analysis/results.Rmd" target="_blank">a29395c</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-05-04
</td>
<td>
exmain DMR feature
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/a7da05c023660196b2706ffd640f53c23e168e5a/docs/results.html" target="_blank">a7da05c</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/d4f860e7a279603f2e9a883846cb3b4bcc2dc106/analysis/results.Rmd" target="_blank">d4f860e</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-28
</td>
<td>
updated tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/92c58f9a7c08bf0d030da80c863d08c626b3817b/docs/results.html" target="_blank">92c58f9</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/588a9b3ca7f22d081b8f916a1111e9825e64b2da/analysis/results.Rmd" target="_blank">588a9b3</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-28
</td>
<td>
updated tables
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/14315fdfa5a9ec201f5675b726076d267d4b7d50/docs/results.html" target="_blank">14315fd</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/2a16e42b7e6c42335840d61edf4a9ce07e2ffe72/analysis/results.Rmd" target="_blank">2a16e42</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-28
</td>
<td>
compare CDTS with other annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/621b19b5f203007d0d0b3b08994d4bb693522bd8/docs/results.html" target="_blank">621b19b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/16ea9f9ba5a41f7cd87cfd522646bdb47eac6a4b/analysis/results.Rmd" target="_blank">16ea9f9</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-22
</td>
<td>
compare annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/bbfe1d41681fcf52636e941ef2b1372fab8568c1/docs/results.html" target="_blank">bbfe1d4</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/4dd58d7655b3ba9bc70417363dd7df4c7189f73d/analysis/results.Rmd" target="_blank">4dd58d7</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-22
</td>
<td>
compare annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/540e287734a85d462236518a91a70d1e90c91e88/docs/results.html" target="_blank">540e287</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/e597d4ed8f7db6c2146a6c55ef6ff6801c76197f/docs/results.html" target="_blank">e597d4e</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-21
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/b93bab117119ddffe9059af3f25f436d534565f1/analysis/results.Rmd" target="_blank">b93bab1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-21
</td>
<td>
compare with other conservation annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/491ca5ccff37db99cae0f1dc9263beaded759acd/docs/results.html" target="_blank">491ca5c</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/1ed77cbdc6156c983d9e92f7caa1e352ec976158/analysis/results.Rmd" target="_blank">1ed77cb</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-14
</td>
<td>
characterize annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/1ed77cbdc6156c983d9e92f7caa1e352ec976158/docs/results.html" target="_blank">1ed77cb</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-14
</td>
<td>
characterize annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/ba251cb4b94ee3d0e77190320421211dc34b99b5/docs/results.html" target="_blank">ba251cb</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/76883f139cd9e9b844506b528fc2bb4e02258c1c/analysis/results.Rmd" target="_blank">76883f1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-14
</td>
<td>
characterize annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/76883f139cd9e9b844506b528fc2bb4e02258c1c/docs/results.html" target="_blank">76883f1</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-14
</td>
<td>
characterize annotations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/1a76b294e45c0bb495ac2108d8c2b021d430577d/docs/results.html" target="_blank">1a76b29</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-06
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/ae12e8aa24b22a009ea11c30c06102393df7ff28/analysis/results.Rmd" target="_blank">ae12e8a</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-06
</td>
<td>
evaluate sequence constraints
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/752cb393786591800d7a9477093e2f731cd88433/docs/results.html" target="_blank">752cb39</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/f1c5950cd9ff555c31e73394072eb624a8254955/docs/results.html" target="_blank">f1c5950</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/j3gu/funcFinemapping/d16a5e0ba9cc157232914ff14a755f7df7a1bf5b/docs/results.html" target="_blank">d16a5e0</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/6f7214b5258820d1c3a3ea16ebe150474514f9ec/analysis/results.Rmd" target="_blank">6f7214b</a>
</td>
<td>
Jing Gu
</td>
<td>
2021-04-05
</td>
<td>
edit index page
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Fine-mapping with functional annotations as priors has shown improved results in identifying causal variants. This project is to evaluate the utility of novel annotation features and adopt ones that can improve fine mapping results.</p>
<div id="evaluation" class="section level2">
<h2>Evaluation</h2>
<div id="gwas-summary-statistics" class="section level3">
<h3>GWAS summary statistics</h3>
<p>Schizopherenia - Pardinas et al., 2018</p>
<ul>
<li>40675 cases and 64643 controls</li>
<li>CLOZUK sample + PGC sample (independent)</li>
<li>179 independent GWAS significant SNPs mapped to 145 independent loci</li>
<li>SNPs were imputed using a combination of the 1KGPp3 and UK 10K datasets.</li>
<li>SNPs were filtered by NFO &gt; 0.6 and MAF &gt; 0.01</li>
<li>LD-score regression analysis: An LD reference was generated from 1KGPp3 after restricting this dataset to strictly unrelated individuals and retaining only markers with MAF &gt; 0.01.</li>
</ul>
</div>
<div id="gwas-qc-procedures" class="section level3">
<h3>GWAS QC Procedures</h3>
<ul>
<li>Current procedures was based on Alan's finemappeR pipeline</li>
<li>Criteria for filtering gwas SNPs</li>
</ul>
<ol style="list-style-type: decimal">
<li>Remove all non-biallelic SNPs</li>
<li>Remove all SNPs with strand-ambiguous alleles (SNPs with A/T, C/G alleles)</li>
<li>Removed SNPs without rs IDs, duplicated rs IDs or base pair position.</li>
<li>Removed SNPs not in the reference panels</li>
<li>Removed SNPs whose base pair positions or alleles doesn’t match the reference panels</li>
<li>Removed all SNPs on chromosome X, Y, and MT</li>
</ol>
<p>After filtering, there are around 6 million variants remained. <img src="assets/gwas_summary_stats_hist_plot.png" alt="Plots for GWAS summary statistics" /></p>
</div>
<div id="features" class="section level3">
<h3>Features</h3>
<ol style="list-style-type: decimal">
<li>Sequence constraints:
<ul>
<li>context-dependent tolerance scores(CDTS) in percentiles</li>
<li>A score was computed for each 10bp bin in the genome.</li>
<li>The lower the score is, the more intolerant to variation is the bin.</li>
</ul></li>
</ol>
</div>
<div id="procedures" class="section level3">
<h3>Procedures</h3>
<ol style="list-style-type: decimal">
<li>GWAS summary statistics was pre-processed to remove sex chromosomes, indels, ambiguous and duplicated SNPs.</li>
<li>Currently, genotypes from 1kg European samples are used to compute LD between SNPs.</li>
<li>SNPs in GWAS summary statistics were matched with the reference panel and assigned to in total 1687 independent LD blocks.</li>
<li>Run TORUS to perform genome-wide enrichment analyses.</li>
</ol>
<p>All variants were catogrized into whether or not they occur in genomic bins with CDTS up to 1 percentile or 5 percentile.</p>
</div>
<div id="examine-the-cdts-feature" class="section level3">
<h3>Examine the CDTS feature</h3>
<p><em>Check the percent of constrained sequences that overlaps with open chromatin regions in brain</em></p>
<ul>
<li>Overlaps between two sets of genomic features were identified using <code>bedtools intersect</code>. The constrained sequences were counted to be overlapped when at least 20%(&gt;=2 bp) intersect with peaks called from ATAC-Seq profiles. No reciprocal option used here due to the much shorter length of CDTS sequence.</li>
</ul>
<table>
<caption>Summary of the overlaps between constrained sequences and open chromatin regions across tissues</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">b_cell</th>
<th align="left">iN_Dopa</th>
<th align="left">iN_GABA</th>
<th align="left">iN_Glut</th>
<th align="left">iPSC</th>
<th align="left">NPC</th>
<th align="left">t_cell</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CDTS_1%</td>
<td align="left">11.3%</td>
<td align="left">46.9%</td>
<td align="left">55.6%</td>
<td align="left">45.4%</td>
<td align="left">62.2%</td>
<td align="left">44.2%</td>
<td align="left">9.4%</td>
</tr>
<tr class="even">
<td align="left">CDTS_5%</td>
<td align="left">3.5%</td>
<td align="left">19%</td>
<td align="left">21.9%</td>
<td align="left">15.7%</td>
<td align="left">25.9%</td>
<td align="left">14.9%</td>
<td align="left">2.5%</td>
</tr>
<tr class="odd">
<td align="left">CDTS_50%</td>
<td align="left">0.4%</td>
<td align="left">5.8%</td>
<td align="left">6.6%</td>
<td align="left">3.4%</td>
<td align="left">7.8%</td>
<td align="left">2.9%</td>
<td align="left">0.1%</td>
</tr>
</tbody>
</table>
<p><em>Example of top 1% CDTS sequences and genes nearby</em><br />
Conserved sequences found in the noncoding regions nearby gene YPEL4, which is a member of the highly conserved YPEL gene family.<br />
<img src="assets/chr11_57405334_57407554.png" alt="chr11:57405334-57407554" /></p>
<p><strong>Summary</strong>:<br />
Overall, CDTS at top percentiles are more enriched in open chromatin regions from brain than immune cells. CDTS at 50% threshold can be regarded as a negative control, which implies that less constrained sequence bins have much less overlapping with OCRs in brain.</p>
<p><em>Check how highly conserved CDTS bins overlap with genomic annotations</em><br />
The coordinates for the genomic regions was generated by Shicheng Guo, which I borrowed here to quickly get some results for how CDTS overlap with genomic annotations. I used <code>bedtools intersect</code> to obtain the pair of CDTS bin and genomic segment that share by at least 1 base. Then I summarized the proportion of CDTS bins that have paired with genomic segments extracted from each annotation file.</p>
<table>
<caption>Summary of overlapping between CDTS and genomic annotations</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">b_cell</th>
<th align="right">iN_Dopa</th>
<th align="right">iN_GABA</th>
<th align="right">iN_Glut</th>
<th align="right">iPSC</th>
<th align="right">NPC</th>
<th align="right">pancreas</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CDTS_1%</td>
<td align="right">11.3</td>
<td align="right">46.9</td>
<td align="right">55.6</td>
<td align="right">45.4</td>
<td align="right">62.2</td>
<td align="right">44.2</td>
<td align="right">9.4</td>
</tr>
<tr class="even">
<td align="left">CDTS_5%</td>
<td align="right">3.5</td>
<td align="right">19.0</td>
<td align="right">21.9</td>
<td align="right">15.7</td>
<td align="right">25.9</td>
<td align="right">14.9</td>
<td align="right">2.5</td>
</tr>
<tr class="odd">
<td align="left">CDTS_50%</td>
<td align="right">0.4</td>
<td align="right">5.8</td>
<td align="right">6.6</td>
<td align="right">3.4</td>
<td align="right">7.8</td>
<td align="right">2.9</td>
<td align="right">0.1</td>
</tr>
</tbody>
</table>
<p><strong>Summary</strong>: The overlap of CDTS with open chromain regions in neurons much higher than immure cells. In our analysis, CDTS shows high overlapping with exons, introns and promoters. Almost 50% of CDTS at 1 percentile overlap with introns, which is much higher than 20% value in paper. The sum of the proportions for exons, promoters and introns exceeds one, which also shows the results need further examination.</p>
<p><strong>Enrichment analysis for sequence constraints</strong><br />
<em>QCs for brain OCR ATAC-seq peaks</em><br />
There are very few peaks with size larger than 5kb across different brain ATAC-Seq profiles. The removal of large peaks did not improve the enrichment results.</p>
<table>
<caption>ATAC-seq peaks summary statistics</caption>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">total_peaks</th>
<th align="right">total_peak_coverage</th>
<th align="left">percent_of_genome</th>
<th align="right">large_peaks</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CDTS_1%</td>
<td align="right">2056917</td>
<td align="right">20574135</td>
<td align="left">0.07%</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">CDTS_5%</td>
<td align="right">10582523</td>
<td align="right">105837405</td>
<td align="left">0.35%</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">iN_Dopa</td>
<td align="right">270779</td>
<td align="right">170466315</td>
<td align="left">0.57%</td>
<td align="right">69</td>
</tr>
<tr class="even">
<td align="left">iN_GABA</td>
<td align="right">276475</td>
<td align="right">184786667</td>
<td align="left">0.62%</td>
<td align="right">14</td>
</tr>
<tr class="odd">
<td align="left">iN_Glut</td>
<td align="right">245252</td>
<td align="right">106380822</td>
<td align="left">0.35%</td>
<td align="right">36</td>
</tr>
<tr class="even">
<td align="left">iPSC</td>
<td align="right">334401</td>
<td align="right">232096455</td>
<td align="left">0.77%</td>
<td align="right">46</td>
</tr>
<tr class="odd">
<td align="left">NPC</td>
<td align="right">244359</td>
<td align="right">96144030</td>
<td align="left">0.32%</td>
<td align="right">44</td>
</tr>
</tbody>
</table>
<p><img src="figure/results.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /> The enrichment estimate has a confience level above zero for CDTS and positive controls. This shows SNPs associated with SCZ are on average ~ 9 fold enriched in genomic bins with up to 5 percentile of CDTS.</p>
<p><strong>Compare with other conservation annotations</strong></p>
<ul>
<li>LINSIGHT
<ul>
<li>predict how noncoding nucleotide sites are likely to have deleterious fitness consequences and hence be phenotypically important</li>
<li>genome-wide average of LINSIGHT scores was ~0.07 (range: 0.03-0.99)</li>
<li>Estimated mean LINSIGHT score for conserved TFBSs was 0.24-&gt;used as cutoff for whether the nucelotide site is conserved</li>
<li>2.5% of GWAS SNPs are above LINSIGHT threshold.</li>
</ul></li>
<li>CADD - Combined Annotation–Dependent Depletion,
<ul>
<li>provides metrics of deleteriousness</li>
<li>scaled PHRED score [-10log10(P)]</li>
<li>5 percent chosen as a cutoff, which represents top 5% of all possible reference genome SNVs</li>
</ul></li>
<li>GERP - Genomic Evolutionary Rate Profiling
<ul>
<li>produce position-specific estimates of evolutionary constraint</li>
<li>constraint intensity quantified as a &quot;rejection score&quot; range from -12.3 to 6.17</li>
<li>UCSC suggests a RS score threshold of 2 which provides high sensitivity and strongly enriched for true constraint sites</li>
</ul></li>
</ul>
<p><em>Compare top k-th percentile of CDTS bins vs top k-th percentile scores of other annotations</em></p>
<pre><code>           top5%_CDTS top5%_CADD
top5%_CDTS      1.000      0.108
top5%_CADD      0.154      1.000</code></pre>
<pre><code>                top10%_CDTS top10%_LINSIGHT top10%_GERP
top10%_CDTS            1.00            0.01       0.087
top10%_LINSIGHT        0.12            1.00          NA
top10%_GERP            0.12              NA       1.000</code></pre>
<p><strong>Summary</strong>:</p>
<ol style="list-style-type: decimal">
<li><p>Overall, there is a maximum of 15.4% overlapping between the bases within the top CDTS bins and those ranked among top annotation scores.</p></li>
<li><p>Only 1% of bases from top 10% CDTS bins have LINSIGHT scores above 10%, probably due to many missing predictions in LINSIGHT.</p></li>
</ol>
<p>The correlation table shows the pair-wise correlations between ranks for the conservation scores on GWAS SNPs across all the annotations. CDTS seems to be uncorrelated to other conservation scores. CADD has a moderate correlation separately with LINSIGHT and GERP. A low correlation observed between GERP and LINSIGHT.</p>
<p><strong>joint TORUS enrichment analysis over conservation-related annotations</strong> <img src="figure/results.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /> With other conservation annotations as predictors in the model, we can see CDTS within top 5 percentile still shows around 8 fold enrichment.</p>
</div>
<div id="dna-methylation" class="section level3">
<h3>DNA methylation</h3>
<p><strong>procedure</strong></p>
<ol style="list-style-type: decimal">
<li>concatenate and sort CG DNA methylation regions across 21 clusters<br />
</li>
<li>use <code>bedtools merge -i merged.bed</code> to merge overlapped regions in the concatenated bed file<br />
</li>
<li>check the distribution of peak lengths<br />
<img src="assets/human_CG_DMR_peak_length.png" alt="Histogram for the sizes of DNA methylation regions" /></li>
</ol>
<p><strong>TORUS enrichment results</strong> <img src="figure/results.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.4 (2021-02-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] ggplot2_3.3.3     knitr_1.31        data.table_1.14.0 workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.6        pillar_1.5.0      compiler_4.0.4    bslib_0.2.4      
 [5] later_1.1.0.1     jquerylib_0.1.3   git2r_0.28.0      highr_0.8        
 [9] tools_4.0.4       digest_0.6.27     gtable_0.3.0      jsonlite_1.7.2   
[13] evaluate_0.14     lifecycle_1.0.0   tibble_3.0.6      pkgconfig_2.0.3  
[17] rlang_0.4.10      DBI_1.1.1         yaml_2.2.1        xfun_0.21        
[21] withr_2.4.1       dplyr_1.0.4       stringr_1.4.0     generics_0.1.0   
[25] fs_1.5.0          vctrs_0.3.6       sass_0.3.1        tidyselect_1.1.0 
[29] rprojroot_2.0.2   grid_4.0.4        glue_1.4.2        R6_2.5.0         
[33] fansi_0.4.2       rmarkdown_2.7     farver_2.0.3      purrr_0.3.4      
[37] magrittr_2.0.1    whisker_0.4       scales_1.1.1      promises_1.2.0.1 
[41] ellipsis_0.3.1    htmltools_0.5.1.1 assertthat_0.2.1  colorspace_2.0-0 
[45] httpuv_1.5.5      labeling_0.4.2    utf8_1.1.4        stringi_1.5.3    
[49] munsell_0.5.0     crayon_1.4.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
