<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>SNP_finemapping_results</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">funcFinemapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SNP_finemapping_results</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-03-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>funcFinemapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210404code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210404)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210404code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210404)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree5ea9a98a4bb2001e7e7be6c4449e5dd91265b0batargetblank5ea9a98a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/j3gu/funcFinemapping/tree/5ea9a98a4bb2001e7e7be6c4449e5dd91265b0ba" target="_blank">5ea9a98</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomj3gufuncFinemappingtree5ea9a98a4bb2001e7e7be6c4449e5dd91265b0batargetblank5ea9a98a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/j3gu/funcFinemapping/tree/5ea9a98a4bb2001e7e7be6c4449e5dd91265b0ba" target="_blank">5ea9a98</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .ipynb_checkpoints/
    Ignored:    analysis/lab4_prepare.nb.html
    Ignored:    analysis/ldsc_results.nb.html
    Ignored:    analysis/learn_archR.nb.html
    Ignored:    analysis/mtsplice_finemapping_results.nb.html
    Ignored:    analysis/results.nb.html
    Ignored:    analysis/snp_finemapping_results.nb.html
    Ignored:    analysis/splicing.nb.html
    Ignored:    analysis/susie_tutorial.nb.html

Untracked files:
    Untracked:  SNPs_categories,png
    Untracked:  SNPs_categories.png
    Untracked:  analysis/.ipynb_checkpoints/
    Untracked:  analysis/asthma_bkg.Rmd
    Untracked:  analysis/asthma_prelim_results_archive.Rmd
    Untracked:  analysis/asthma_prelim_results_cp.Rmd
    Untracked:  analysis/asthma_results_cp.Rmd
    Untracked:  analysis/enhancer_gene_feature.Rmd
    Untracked:  analysis/feedback.Rmd
    Untracked:  analysis/gene_finemapping_results.Rmd
    Untracked:  analysis/lab4.Rmd
    Untracked:  analysis/lab4_prepare.Rmd
    Untracked:  analysis/learn_archR.Rmd
    Untracked:  analysis/learn_susie.Rmd
    Untracked:  analysis/mtsplice_finemapping_results.Rmd
    Untracked:  analysis/notes.Rmd
    Untracked:  analysis/splicing.Rmd
    Untracked:  analysis/susie_tutorial.Rmd
    Untracked:  bmi_locus1410.pdf
    Untracked:  code/.ipynb_checkpoints/
    Untracked:  code/Euro_LD_Chunks.RData
    Untracked:  code/ldsc_regression.sh
    Untracked:  code/make_plots.R
    Untracked:  code/run_ldsc.sh
    Untracked:  code/run_ldsc_with_bed.sh
    Untracked:  code/run_ldsc_with_bed_v2.sh
    Untracked:  code/run_susie.R
    Untracked:  code/run_torus.sh
    Untracked:  code/split_vcf.sh
    Untracked:  data/hg19_gtf_genomic_annots_ver2.gr.rds
    Untracked:  data/num_overlaps_finemapped_SNPs_and_ctcf.txt
    Untracked:  data/scz_2018
    Untracked:  data/scz_spliceAI_0.01_prior_output.RData
    Untracked:  data/scz_spliceAI_0.01_ukb_out.RData
    Untracked:  data/torus_enrichment_novel_annot.est
    Untracked:  data/torus_joint_enrichment.est
    Untracked:  data/torus_joint_refined_enrichment.est
    Untracked:  enhancer_gene_feature.rmd
    Untracked:  fig1_panels.pdf
    Untracked:  fig2.pdf
    Untracked:  fig_panel2.pdf
    Untracked:  gene_mapping.pdf
    Untracked:  lab4
    Untracked:  output/AAD/GMP_merge_stats.txt
    Untracked:  output/AAD/allergy/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD4_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/CD8_compare_old.est
    Untracked:  output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare_old.est
    Untracked:  output/LDL_ukb_L10.gif
    Untracked:  output/LDL_ukb_L10.pdf
    Untracked:  output/background_SNPs_annotated_percent.txt
    Untracked:  output/locus_1452.gif
    Untracked:  output/locus_1452.pdf
    Untracked:  output/splicing/torus_enrichment_joint_scz_spliceAI.est
    Untracked:  panel_figure2.pdf
    Untracked:  test.txt
    Untracked:  transfer/

Unstaged changes:
    Deleted:    .Rprofile
    Modified:   analysis/asthma_prelim_results.Rmd
    Modified:   analysis/asthma_results.Rmd
    Modified:   analysis/enrichment_analysis.Rmd
    Modified:   output/AAD/allergy/Ulirsch2019/GMP_merge_compare.est
    Deleted:    output/AAD/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/AAD/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/AAD/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/AAD/asthma/diffe_adult_blood_torus.est
    Deleted:    output/AAD/asthma/joint_blood_immune_rest_vs_stimu.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/AAD/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/AAD/asthma/lung_clusters_dict.txt
    Deleted:    output/AAD/asthma/lung_clusters_info.txt
    Deleted:    output/AAD/asthma/stimu_adult_blood_torus.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_rest.est
    Deleted:    output/AAD/asthma/torus_enrichment_all_stimulated.est
    Deleted:    output/AAD/asthma/zhang2021_annot_percent.txt
    Deleted:    output/AAD/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/AAD/asthma/zhang2021_peaks_per_celltype.txt
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD4_compare.est
    Modified:   output/AAD/asthma_adult/Ulirsch2019/CD8_compare.est
    Deleted:    output/AAD/asthma_adult/Ulirsch2019/GMP_merge_compare.est
    Modified:   output/AAD/asthma_child/Ulirsch2019/GMP_merge_compare.est
    Deleted:    output/asthma/Caldero2019_diffDA_annot_percent.txt
    Deleted:    output/asthma/Caldero2019_stimuDA_annot_percent.txt
    Deleted:    output/asthma/celltype_specific_adult_lungs_torus.est
    Deleted:    output/asthma/diffe_adult_blood_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_diff_torus.est
    Deleted:    output/asthma/joint_lung_vs_blood_immune_stimu_torus.est
    Deleted:    output/asthma/lung_clusters_dict.txt
    Deleted:    output/asthma/lung_clusters_info.txt
    Deleted:    output/asthma/stimu_adult_blood_torus.est
    Deleted:    output/asthma/zhang2021_annot_percent.txt
    Deleted:    output/asthma/zhang2021_cell_type_overlaps.txt
    Deleted:    output/asthma/zhang2021_peaks_per_celltype.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/snp_finemapping_results.Rmd</code>) and HTML (<code>docs/snp_finemapping_results.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/j3gu/funcFinemapping/blob/5ea9a98a4bb2001e7e7be6c4449e5dd91265b0ba/analysis/snp_finemapping_results.Rmd" target="_blank">5ea9a98</a>
</td>
<td>
Jing Gu
</td>
<td>
2022-03-24
</td>
<td>
update spliceAI results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="assess-the-impact-of-different-priors-on-fine-mapping" class="section level2">
<h2>Assess the impact of different priors on fine mapping</h2>
<pre class="r"><code>library(data.table)
library(dplyr)
library(susieR)
library(bigsnpr)
library(ggplot2)
source(&quot;code/make_plots.R&quot;)
source(&quot;code/run_susie.R&quot;)</code></pre>
<div id="spliceai-predicted-variant-effects-on-splicing" class="section level3">
<h3>SpliceAI-predicted variant effects on splicing</h3>
<p><strong>Annotations</strong> * spliceAI predictions are not tissue-specific. * The predicted delta score of a variant can be interpreted as the probrability of the variant being splice-altering.</p>
<p><strong>Procedure</strong> * Build binary annotation by denoting delta score&gt;=0.01 as 1 and detal score &lt; 0.01 as 0. * Run TORUS jointly over spliceAI-predictions, iPSC-derived neuronal accessibility, CADD top5% and protein coding regions. * Use the derived prior probability for each SNP to further perform fine mapping with Susie.</p>
<p><strong>Torus enrichment estimates</strong></p>
<pre class="r"><code>est&lt;-read.table(&quot;output/splicing/torus_enrichment_joint_scz_spliceAI.est&quot;, header = F, skip = 2)
colnames(est)&lt;-c(&quot;term&quot;, &quot;estimate&quot;, &quot;low&quot;, &quot;high&quot;)
snp_enrichment_plot(est, y.label = c(&quot;top5% CADD&quot;, &quot;CDS&quot;, &quot;iPSC-derived&quot;, &quot;spliceAI_0.01&quot;))</code></pre>
<p><img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="results-summary" class="section level4">
<h4>Results Summary</h4>
<p><strong>Plot p-values against susie PIPs</strong> <img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /><img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-5-2.png" width="1152" style="display: block; margin: auto;" /> * For an initial QC, we did not observe SNPs with high PIPs but show no significant associations for both scenarios under the assumption of one causal variant per locus. However, at L=10 we did see a substantial fractions of SNPs with high PIPs and non-sigiciant p-values for the run with functional priors.</p>
<p>NOTE: The following analyses were based on L=1.</p>
<p><strong>Compare PIPs between uniform and functional priors</strong> <img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /> * We observed most variants have similar low PIPs with or without functional priors. However some variants show higher PIPs with functional prior while some show higher PIPs without functional priors.</p>
<p><strong>Compare the sizes of credible sets</strong></p>
<ul>
<li>Uniform priors</li>
</ul>
<pre class="r"><code>plot_cs_size(uniform_L1) </code></pre>
<p><img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /> * Functional priors</p>
<pre class="r"><code>plot_cs_size(annot_L1)</code></pre>
<p><img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /> * The numbers the stacked barplot represent the numbers of independent LD blocks within each range of credible sizes. * Overall, the credible sizes are similar between the runs with or without priors</p>
</div>
</div>
<div id="examine-the-snps-within-credible-set-of-size-less-than-20-variants" class="section level3">
<h3>Examine the snps within credible set of size less than 20 variants</h3>
<p><strong>Examine a few loci</strong></p>
<pre><code>                              snp      beta        se     pval    zscore locus
85212   2:200715388:G:T:rs2949006  0.100117 0.0118920 3.69e-17 -8.418853   251
46595   2:57987593:T:C:rs11682175 -0.066332 0.0095085 3.05e-12 -6.976074   171
240783 8:143316970:G:A:rs13262595  0.081672 0.0096177 1.99e-17 -8.491843   935
6764     1:8423510:G:A:rs10779702  0.057891 0.0101120 1.03e-08 -5.724980     6
97703    3:2547786:G:T:rs17194490 -0.092006 0.0131910 3.06e-12 -6.974907   279
6842       1:8484228:C:T:rs159961 -0.058975 0.0102170 7.83e-09 -5.772242     6
240768  8:143312933:C:A:rs4129585  0.082041 0.0095633 9.26e-18 -8.578733   935
97650    3:2519703:G:T:rs17014863 -0.085798 0.0123130 3.21e-12 -6.968083   279
6755     1:8418644:A:C:rs11121172  0.055245 0.0102990 8.25e-08 -5.364113     6
6802      1:8452725:C:T:rs2661863  0.056569 0.0101850 2.81e-08 -5.554148     6
       X_NUM_ID_  torus_pip cs cs_purity cs_size  susie_pip iPSC_derived_d
85212       2883 0.00482880  1 0.9912495       2 0.93320400              1
46595       2743 0.00023381  1 0.7441588       8 0.91643760              1
240783      1392 0.00127850  1 0.8849128       3 0.77486460              1
6764        4052 0.00482880  1 0.9317116      19 0.39688087              1
97703       6588 0.00058581  1 0.7786948      17 0.30755800              0
6842        4166 0.00127850  1 0.9317116      19 0.13671204              1
240768      1373 0.00010706  1 0.8849128       3 0.13482170              0
97650       6513 0.00023381  1 0.7786948      17 0.11716620              1
6755        4043 0.00816960  1 0.9317116      19 0.09668983              0
6802        4114 0.00221670  1 0.9317116      19 0.07165940              0
       CADD_d CDS_d spliceAI0.01_d_d all_d
85212       0     0                1     2
46595       0     0                0     1
240783      1     0                0     2
6764        0     0                1     2
97703       1     0                0     1
6842        1     0                0     2
240768      0     0                0     0
97650       0     0                0     1
6755        1     1                1     3
6802        0     0                1     1</code></pre>
<ul>
<li>Locus 251:
<ul>
<li>top snp(2:200715388:G:T:rs2949006|zscore=8.42|spliceAI=0.01|iPSC_derived=1|pip=0.93|cs_size=2)</li>
<li>Compared with the other SNP in credible set, rs2949006 has a PIP higher than 0.9 due to being annotated with having splice-altering effects.</li>
<li>found in GWAS catalog</li>
<li>not a GTEx splicing QTL <img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-10-1.png" width="576" style="display: block; margin: auto;" /><img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-10-2.png" width="576" style="display: block; margin: auto;" /></li>
</ul></li>
</ul>
<pre><code>                            snp     beta       se     pval    zscore locus
85212 2:200715388:G:T:rs2949006 0.100117 0.011892 3.69e-17 -8.418853   251
85213  2:200716119:A:C:rs796364 0.099031 0.011912 9.41e-17 -8.313549   251
      X_NUM_ID_  torus_pip cs cs_purity cs_size  susie_pip iPSC_derived_d
85212      2883 0.00482880  1 0.9912495       2 0.93320400              1
85213      2884 0.00023381  1 0.9912495       2 0.01895903              1
      CADD_d CDS_d spliceAI0.01_d_d all_d
85212      0     0                1     2
85213      0     0                0     1</code></pre>
<ul>
<li>Locus 6:
<ul>
<li>top snp(1:8423510:G:A:rs10779702|zscore=5.7|spliceAI=0.02|iPSC_derived=1|pip=0.4|cs_size=19)</li>
<li>not found in GWAS catalog</li>
<li>annotated as a GTEx eQTL from combined analysis <img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-11-1.png" width="576" style="display: block; margin: auto;" /><img src="figure/snp_finemapping_results.Rmd/unnamed-chunk-11-2.png" width="576" style="display: block; margin: auto;" /></li>
</ul></li>
</ul>
<pre><code>                            snp     beta       se     pval    zscore locus
85212 2:200715388:G:T:rs2949006 0.100117 0.011892 3.69e-17 -8.418853   251
85213  2:200716119:A:C:rs796364 0.099031 0.011912 9.41e-17 -8.313549   251
      X_NUM_ID_  torus_pip cs cs_purity cs_size  susie_pip iPSC_derived_d
85212      2883 0.00482880  1 0.9912495       2 0.93320400              1
85213      2884 0.00023381  1 0.9912495       2 0.01895903              1
      CADD_d CDS_d spliceAI0.01_d_d all_d
85212      0     0                1     2
85213      0     0                0     1</code></pre>
</div>
<div id="annotate-fine-mapped-results" class="section level3">
<h3>Annotate fine-mapped results</h3>
</div>
<div id="troubleshoot-toruss-inflation-esimation" class="section level3">
<h3>Troubleshoot Torus's inflation esimation</h3>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.4 (2021-02-15)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] RColorBrewer_1.1-2 ggplot2_3.3.3      bigsnpr_1.9.11     bigstatsr_1.5.6   
[5] susieR_0.11.92     dplyr_1.0.4        data.table_1.14.2 

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8         lattice_0.20-41    assertthat_0.2.1   rprojroot_2.0.2   
 [5] digest_0.6.27      foreach_1.5.1      utf8_1.2.2         R6_2.5.1          
 [9] bigsparser_0.6.0   plyr_1.8.6         evaluate_0.14      highr_0.8         
[13] pillar_1.5.0       flock_0.7          rlang_1.0.1        rstudioapi_0.13   
[17] irlba_2.3.3        whisker_0.4        jquerylib_0.1.3    Matrix_1.4-0      
[21] rmarkdown_2.7      labeling_0.4.2     bigparallelr_0.3.2 stringr_1.4.0     
[25] munsell_0.5.0      mixsqp_0.3-43      compiler_4.0.4     httpuv_1.5.5      
[29] xfun_0.21          pkgconfig_2.0.3    htmltools_0.5.1.1  tidyselect_1.1.1  
[33] tibble_3.0.6       workflowr_1.6.2    codetools_0.2-18   matrixStats_0.58.0
[37] reshape_0.8.8      fansi_1.0.2        crayon_1.4.1       withr_2.4.3       
[41] later_1.1.0.1      grid_4.0.4         jsonlite_1.7.2     gtable_0.3.0      
[45] lifecycle_1.0.0    DBI_1.1.1          git2r_0.28.0       magrittr_2.0.1    
[49] scales_1.1.1       cli_3.2.0          stringi_1.5.3      farver_2.1.0      
[53] fs_1.5.0           promises_1.2.0.1   doParallel_1.0.16  bslib_0.2.4       
[57] ellipsis_0.3.2     generics_0.1.0     vctrs_0.3.8        cowplot_1.1.1     
[61] iterators_1.0.13   tools_4.0.4        glue_1.6.1         purrr_0.3.4       
[65] parallel_4.0.4     yaml_2.2.1         colorspace_2.0-2   bigassertr_0.1.5  
[69] knitr_1.31         sass_0.3.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
