---
title: "asthma_prelim_results"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(knitr)
source("code/make_plots.R")
```
## cell-type resolved annotations
*Caldero et al. 2019* - ATAC-Seq data for FACS-sorted cells from whole blood   
I extracted ATAC-Seq peaks specific to each immune cell type from the count table that consists of a union set of peaks across cells based on the provided cutoffs.

### immune traits 
This analysis was done to replicate the enrichment results from figure 5b in the paper. Overall, the magnitude of enrichment estimates are much smaller than that in figure 5b. I could not find their resouce of GWAS dataset, but I believe that I used a different GWAS data set for RA. They claimed 20% per-SNP heritability for RA, while the dataset in 2014 I used only found 8% genome-wide heritability. They used LDSC to estimate enrichment estimates. The stimulated immune cells tend to have slightly higher enrichment estimates, but not as significant due to overlapped CIs. However, the figure in the paper demonstrates strong enrichment of stimulated-related peaks for cells such as Bulk B, naive B, CD8+ T and Th1. 
```{r echo=F}
immune_cells<-c("Bulk B", "Naive B", "Mem B","Plasmablasts",
              "CD8pos T", "Naive CD8 T", "Central memory CD8pos T", 
              "Effector memory CD8pos T", "Gamma delta T", "Regulatory T","Naive Tregs",
              "Memory Tregs","Effector CD4pos T","Naive Teffs", "Memory Teffs",
              "Th1 precursors", "Th2 precursors", "Th17 precursors",
             "Follicular T Helper","Immature NK", "Mature NK", "Memory NK",
             "Monocytes", "pDCs", "Myeloid DCs")
traits<-c("ra_2014", "asthma_adult", "allergy")
tbl<-data.frame()
for(trait in traits){
  rest<-read.table(sprintf("output/AAD/%s/torus_enrichment_all_rest.est", trait), header=F)
  stimu<-read.table(sprintf("output/AAD/%s/torus_enrichment_all_stimulated.est", trait), header=F)
  # name the terms in a simpler form (todo)
  rest["condition"]<-"resting"
  stimu["condition"]<-"stimulated"
  df<-data.frame(rbind(rest, stimu))
  colnames(df)<-c("raw_term", "estimate", "low", "high", "condition")
  df["term"]<-unlist(lapply(df$raw_term, function(i){paste(strsplit(strsplit(i, "[.]")[[1]][1], "_")[[1]], collapse = " ")}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}

tbl$condition<-factor(df$condition, levels=c("stimulated", "resting"))
snp_enrichment_plot(tbl, split.data = "condition", y.order = rev(immune_cells), trait="") + facet_grid(. ~ trait)
```
For Asthma, we did not see stronger enrichment of stimulated-related peaks compared with resting ones. But overall all immune cells whether or not stimulated (except for the stimulated mature NK) are significantly enriched for GWAS risk variants. 


## Joint Enrichment analysis   
### Immune cells grouped by lineages
Immune cells were grouped into six main categories and merged across two conditions. The peaks in these groups can be overlapped.

* NK
* B cells
* CD4+
* CD8+
* Myeloids
* Gamma delta T
```{r echo=F}
tbl<-data.frame()
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Caldero2019_regroup.est", trait), header=T)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  ### remove lines of FDR values for each loci
  df["label"] = unlist(lapply(df$raw_term, function(i){length(strsplit(i[1], "[.]")[[1]])>1}))
  df<-df[df$label==TRUE,]
  df["term"]<-unlist(lapply(df$raw_term, function(i){strsplit(i, "_merge")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}

snp_enrichment_plot(tbl, trait="", y.label = c("B cells", "CD4_T", "CD8_T", "Gamma_delta_T", "Myeloid cells", "NK cells")) + facet_grid(. ~ trait)
```
### Immune cells grouped by lineages
Immune cells were grouped into six main categories but compared across conditions.
```{r echo=F, fig.width=10, fig.height=6}
tbl<-data.frame()
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Caldero2019_condition.est", trait), header=T)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  ### remove lines of FDR values for each loci
  df["label"]<-unlist(lapply(df$raw_term, function(i){length(strsplit(i[1], "[.]")[[1]])>1}))
  df<-df[df$label==TRUE,]
  df["term"]<-unlist(lapply(df$raw_term, function(i){strsplit(i[1], "[.]")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}
tbl["condition"]<-unlist(lapply(tbl$term, function(i){rev(strsplit(i[1], "_")[[1]])[1]}))
tbl$condition<-factor(tbl$condition, levels = c("S", "U"), labels=c("stimulated","resting"))
tbl["term"]<-unlist(lapply(tbl$term, function(i){
            words<-strsplit(i[1], "_")[[1]]
            paste(words[1:(length(words)-1)], collapse = "_")}))
snp_enrichment_plot(tbl, trait="") + facet_grid(. ~ trait)
```
### disjoint groups by lineage 
```{r echo=F}
tbl<-data.frame()
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Caldero2019_lineage.est", trait), header=T)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  ### remove lines of FDR values for each loci
  df["label"] = unlist(lapply(df$raw_term, function(i){length(strsplit(i[1], "[.]")[[1]])>1}))
  df<-df[df$label==TRUE,]
  df["term"]<-unlist(lapply(df$raw_term, function(i){strsplit(i, "[.]")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}

snp_enrichment_plot(tbl, y.order = rev(c("B_stimulated", "T_stimulated", "BandT_stimulated",
                                     "B_resting", "T_resting","nk_resting","myeloid_resting",
                                     "thymo_resting","EPI_resting","progenitor_resting",
                                     "open")), trait="") + xlim(-15,15) + facet_grid(. ~ trait)

```

### Other datasets Ulirsch2019
test enrichment in other immune cell types such as granulocytes. 
```{r echo=F}
tbl<-data.frame()
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Ulirsch2019_torus_indiv.est", trait), header=F)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  df["term"]<-unlist(lapply(df$raw_term, function(i){strsplit(i, "[.]")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}

snp_enrichment_plot(tbl, trait="") + facet_grid(. ~ trait)
```

## Annotations - Differential accessible peaks 
*Caldero et al. 2019* 

  - Significant differentially accessible regions when compared to progenitor cells
  - Significant differentially accessible regions under stimulation

```{r echo=F}
df<-read.table("output/AAD/asthma_adult/stimu_adult_blood_torus.est",header = F)
colnames(df)<-c("raw_term", "est", "low", "high")
df["term"]<-unlist(lapply(df$raw_term, function(i){
                          term<-strsplit(i, "_S")[[1]][1]
                          return(paste(strsplit(term, "_")[[1]], collapse = " "))}))
df['highlight']<-df$low >=0
#omit cell types with very large standard deviations
df['omit']<-df$low < -10 | df$high > 10
plot_torus_enrichment(df[df['omit']==F,], text_angle=90, vjust=1) + ggtitle("Stimulation-specific & accessible regions in blood")
```
    

    
