---
title: "asthma_prelim_results"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(knitr)
library(dplyr)
library(RColorBrewer)
source("code/make_plots.R")
```
## Asthama and Allergy diseases
The main goal is to identify causal variants, genes and cell types relevant to AAD by integrating omics data of lung samples. We hypothesize that open chromatin regions of lung-resident immune cells can explain broader heritability for AAD than those of blood immune cells. The disease associated variants annotated by these regulatory functions are  more likely to contribute to disease risk, so this prior knowledge can be leveraged to prioritize risk variants in GWAS loci. 

## Dataset
### ATAC-seq data for Blood immune cells (Caldero2019)
**Dataset**: ATAC-Seq profiles for FACS-sorted cells from the peripheral blood of up to 4 healthy donors

**Output files**:  
1. An ATAC-seq count table with a union set of peaks as rows and individual cell as columns  
2. A Sample QC table that includes number of peaks and cell type identity for each cell   
3. Significant differentially accessible regions when compared to progenitor cells   
4. Significant differentially accessible regions under stimulation   

**Procedure**: I used the number of peaks after QC for each cell to extract its corresponding ATAC-Seq peaks from the count table. 
<details>
  <summary>Details of the procedure</summary>
  For the cells of interest, I sorted their corresponding columns of the count table and store the top N number of peaks according to the number of peaks from the QC table as cell type resolved peaks. 
</details> 

### ATAC-seq data for hematopoietic cells (Ulirsch2019)
**Dataset**: ATAC-Seq profiles for FACS-sorted cells from human peripheral blood or bone marrow. 

**Output files**:   
1.peak files downloaded from: https://github.com/caleblareau/singlecell_bloodtraits/tree/master/data/bulk/ATAC/narrowpeaks

### scATAC-seq data for lung tissues (Wang2020)
**Dataset**: scATAC-Seq and scRNA-seq profiles for small airway region of right middle lobe (RML) lung tissue from 3 donors at different ages 

**Output files**:   
1. peak files downloaded from web portal: https://www.lungepigenome.org/

### scATAC-seq data for fetal hematopoietic cells (Ronzoni2021)
**Dataset**: scATAC-Seq and scRNA-seq profiles of human immunophenotypic blood cells from fetal liver and bone marrow

**Output files**:  
Downloaded from gitlab page: to be added  
1. A merged normalized peak table with a union set of peaks as rows and individual cell as columns    
2. A meta table that includes number of peaks and predicted cell type identity for each cell     
3. A raw count table with peaks as rows and individual cell as columns   

**Procedure**: I used the number of peaks for each cell from the meta table to extract its corresponding ATAC-Seq peaks from the merged peak table. The number of peaks is equivalent to the amount of non-zero peaks.

## TORUS run for individual Blood annotation 
**Motivation**: 

1. perform QC check for the Caldero2019 dataset
2. have a senes of which cell types are potentially relevant to AAD

**Results**: 
Overall, the magnitude of enrichment estimates are much smaller than that in figure 5b. The discrepancy can be due to different GWAS datasets and enrichment method. They used LDSC to estimate enrichment coefficients, which have a better control on the overlapping peaks between many annotations.
```{r echo=F, fig.height=8, fig.width=12}
immune_cells<-c("Bulk B", "Naive B", "Mem B","Plasmablasts",
              "CD8pos T", "Naive CD8 T", "Central memory CD8pos T", 
              "Effector memory CD8pos T", "Gamma delta T", "Regulatory T","Naive Tregs",
              "Memory Tregs","Effector CD4pos T","Naive Teffs", "Memory Teffs",
              "Th1 precursors", "Th2 precursors", "Th17 precursors",
             "Follicular T Helper","Immature NK", "Mature NK", "Memory NK",
             "Monocytes", "pDCs", "Myeloid DCs")
traits<-c("ra_2014", "asthma_adult", "allergy")
tbl<-data.frame()
for(trait in traits){
  rest<-read.table(sprintf("output/AAD/%s/torus_enrichment_all_rest.est", trait), header=F)
  stimu<-read.table(sprintf("output/AAD/%s/torus_enrichment_all_stimulated.est", trait), header=F)
  # name the terms in a simpler form (todo)
  rest["condition"]<-"resting"
  stimu["condition"]<-"stimulated"
  df<-data.frame(rbind(rest, stimu))
  colnames(df)<-c("raw_term", "estimate", "low", "high", "condition")
  df["term"]<-unlist(lapply(df$raw_term, function(i){paste(strsplit(strsplit(i, "[.]")[[1]][1], "_")[[1]], collapse = " ")}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}

tbl$condition<-factor(df$condition, levels=c("stimulated", "resting"))
snp_enrichment_plot(tbl, split.data = "condition", y.order = rev(immune_cells), trait="", bar.width = 0) + facet_grid(. ~ trait)
```

## TORUS run for individual lung annotation
**Motivation**: 

1. perform QC check for this dataset
2. have a senes of which cell types in lungs are potentially relevant to AAD

**Results**: 
Data look good as the open chromatin regions of most cell types in lungs are enriched with disease risk variants. 
```{r echo=F, fig.width=12, fig.height=8}
tbl<-data.frame()
for(trait in c(traits, "asthma_child")){
  df<-read.table(sprintf("output/AAD/%s/Wang2020_indiv.est", trait), header=F)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  df["term"]<-unlist(lapply(df$raw_term, function(i){strsplit(i, "[.]")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}

snp_enrichment_plot(tbl, trait="") + facet_grid(. ~ trait)

```
## TORUS joint run for lung vs blood immune cells
**Motivation**:  
Test the hypothesis that open chromatin regions of lung-resident immune cells can explain broader heritability for AAD than those of blood immune cells.

**Procedure**:
For each immune group (B cells, T cells, Myeloids, NK cells), I ran TORUS over pairs of annotations from lung and blood one at a time. 
Pairs of annotations as follows:

* B cells (lung) vs B cells (blood)
* T cells (lung) vs CD4+ T (blood)
* T cells (lung) vs CD8+ T (blood)
* NK cells (lung) vs NK cells (blood)
* Fibroblast cells (lung) vs Myeloids (blood)

**Results**: 
Overall, we see lung-resident immune cells show significant enrichemnt conditional on blood immune cells in AAD, but not in RA. Compared with a control set of traits, there is a large difference between blood and lung enrichment estimates for Myeloid, CD4+ and CD8+ T cells in AAD. 

### Caldero2019 dataset
```{r echo=F}
tbl<-data.frame()
traits<-c("ra_2014", "asthma_adult", "allergy", "asthma_child", "LDL", "HDL", "height", "bmi")
test<-traits[1:4]
control<-traits[5:8]
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Caldero2019_compare.est", trait), header=F)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  cols<-data.frame(t(sapply(df$raw_term, function(i){
    strsplit(strsplit(i, "[.]")[[1]][1], "_compare_")[[1]]
  }))) 
  colnames(cols)<-c("term", "tissue")
  cols$tissue<-factor(cols$tissue, levels=c("lung", "blood"))
  tbl<-data.frame(rbind(tbl, cbind(df, cols, trait)), row.names = NULL)
}

snp_enrichment_plot(tbl[tbl$trait %in% test,], trait="", split.data="tissue",
                    bar.width=0) + facet_grid(. ~ trait)
snp_enrichment_plot(tbl[tbl$trait %in% control,], trait="", split.data="tissue",
                    bar.width=0) + facet_grid(. ~ trait)

```
### Ulirsch2019 dataset

**Motivation**: Replicate the above result in a different blood dataset

**Results**:   
Overall, the enrichment of lung annotations dwindles in a great extent when jointly run with blood annotations. There is one exception that CD8+ T cells in lungs show significant enrichment with child-onset asthma. 

```{r echo=F}
tbl<-data.frame()
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Ulirsch2019_compare.est", trait), header=F)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  cols<-data.frame(t(sapply(df$raw_term, function(i){
    words=strsplit(strsplit(i, "[.]")[[1]][1], "_")[[1]]
    return(c(paste(words[1:(length(words)-1)], collapse = "_"), words[length(words)]))
  }))) 
  colnames(cols)<-c("term", "tissue")
  cols$tissue<-factor(cols$tissue, levels=c("lung", "blood"))
  tbl<-data.frame(rbind(tbl, cbind(df, cols, trait)), row.names = NULL)
}

snp_enrichment_plot(tbl[tbl$trait %in% test,], trait="", split.data="tissue",
                    bar.width=0) + facet_grid(. ~ trait)
snp_enrichment_plot(tbl[tbl$trait %in% control,], trait="", split.data="tissue",
                    bar.width=0) + facet_grid(. ~ trait)
```

##Joint TORUS run for annotation sets
### major clusters in lung dataset 

* Mesenchymal cells
* Immune cells
* epithelial cells
* endothelial cells
* others(neuroendocrine cells, type I pneumocyte, ciliated cells)

**Motivation**: determine how each cluster with similar chromatin accessibility patterns contribute to AAD heritability. 

**Results**:  
We observed some non-immune cell types such as epithelial cells, endothelia cells do contribute to genetic risks of AAD, though not as much as immune cells. 
```{r echo=F, fig.width=12}
tbl<-data.frame()
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Wang2020_regroup.est", trait), header=T, skip=1)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  df["term"] = unlist(lapply(df$raw_term, function(i){strsplit(i[1], "[.]")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}

snp_enrichment_plot(tbl[tbl$trait %in% test,], trait="",
                    y.order=c("neuroendocrine_cell", "type_I_pneumocyte",
                               "ciliated_cell", "epithelial_cells",
                               "mesenchymal_cells", "endothelial_cells",
                               "immune_cells")
                    ) + facet_grid(. ~ trait)
snp_enrichment_plot(tbl[tbl$trait %in% control,], trait="", 
                    y.order=c("neuroendocrine_cell", "type_I_pneumocyte",
                               "ciliated_cell", "epithelial_cells",
                               "mesenchymal_cells", "endothelial_cells",
                               "immune_cells")                    
                    ) + facet_grid(. ~ trait)

```

### annotation sets for Caldero2019
#### grouped by lineages

**Motivation**: 
Running annotation sets in a joint model enables us to idenity the relevant contribution in open chromatin regions of each immune cell type to traits. LSDC is a better tool to use as we expected many overlaps of acccessible peaks across the sub-clusters of immune cells, which can make the estimation of TORUS's joint model unstable. Here we grouped the cell types and used TORUS to get a quick run of which group of cell types have significant enrichment.   

**Procedure**:     
For Caldero2019 dataset: Immune cells were grouped into six main categories and merged across two conditions. The peaks in these groups can be overlapped.     
For Ronzoni2021 dataset, I took a union set of peaks from all cells that were predicted to be granulocytes progenitors.

**Results**:  
Consistent with prior knowledge, we see enrichment of granulocyte progenitors with genetic risk of allergy. 
```{r echo=F, fig.width=12}
tbl<-data.frame()
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Caldero2019_addGPs.est", trait), header=T)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  ### remove lines of FDR values for each loci
  df["label"] = unlist(lapply(df$raw_term, function(i){length(strsplit(i[1], "[.]")[[1]])>1}))
  df<-df[df$label==TRUE,]
  df["term"]<-unlist(lapply(df$raw_term, function(i){strsplit(i, "_merge")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}

snp_enrichment_plot(tbl[tbl$trait %in% test,], trait="", y.label = c("Myeloid cells","B cells", "NK cells", "CD8_T", "CD4_T","Gamma_delta_T", "Granulocyte-progenitors")) + xlim(-3,5) + facet_grid(. ~ trait)
#snp_enrichment_plot(tbl[tbl$trait %in% control,], trait="", y.label = c("Myeloid cells","B cells", "NK cells", "CD8_T", "CD4_T","Gamma_delta_T", "Granulocyte-progenitors")) + facet_grid(. ~ trait)

```
**Summary**:

1. Overall, we see the differences in immune cell components that contribute to these three autoimmune diseases. 
2. GPs and CD4+ T cels are consistently significant in enrichment of risk variants for all three diseases. 

#### grouped by lineages and conditions

**Procedure**: 
Immune cells were grouped first into six main categories and then separated by conditions. This set of 12 annotations were jointly tested via TORUS.
```{r echo=F, fig.width=10, fig.height=6}
tbl<-data.frame()
for(trait in test[1:3]){
  df<-read.table(sprintf("output/AAD/%s/Caldero2019_condition.est", trait), header=T)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  ### remove lines of FDR values for each loci
  df["label"]<-unlist(lapply(df$raw_term, function(i){length(strsplit(i[1], "[.]")[[1]])>1}))
  df<-df[df$label==TRUE,]
  df["term"]<-unlist(lapply(df$raw_term, function(i){strsplit(i[1], "[.]")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}
tbl["condition"]<-unlist(lapply(tbl$term, function(i){rev(strsplit(i[1], "_")[[1]])[1]}))
tbl$condition<-factor(tbl$condition, levels = c("S", "U"), labels=c("stimulated","resting"))
tbl["term"]<-unlist(lapply(tbl$term, function(i){
            words<-strsplit(i[1], "_")[[1]]
            paste(words[1:(length(words)-1)], collapse = "_")}))
snp_enrichment_plot(tbl, trait="", bar.width = 0) + facet_grid(. ~ trait)
```
#### grouped by disjoint peaks  
*Enrichment of each disjoint set of peaks*

**Motivation**: 
Using disjoint groups of peaks from immune cell types to estimate separate contributions of immune components to disease heritability. These annotations are ideal predictors for the linear model used in either LDSC or TORUS to obtain unbiased enrichment estimators.     

**Procedure**: The disjoint peaks were directly downloaded from the paper.
```{r echo=F, fig.width=12, fig.height=8}
tbl<-data.frame()
for(trait in test){
  df<-read.table(sprintf("output/AAD/%s/Caldero2019_lineage.est", trait), header=T)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  ### remove lines of FDR values for each loci
  df["label"] = unlist(lapply(df$raw_term, function(i){length(strsplit(i[1], "[.]")[[1]])>1}))
  df<-df[df$label==TRUE,]
  df["term"]<-unlist(lapply(df$raw_term, function(i){strsplit(i, "[.]")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}
snpsIn<-read.table("output/AAD/Caldero2019_lineage_snps.sumstats", header=T)
cols<-rev(c("B_stimulated", "T_stimulated", "BandT_stimulated",
            "B_resting", "T_resting","nk_resting","myeloid_resting",  "EPI_resting","progenitor_resting","open"))
cols<-unlist(lapply(cols, function(i){
                    label<-snpsIn[snpsIn$term==i, "percent"]
                    return(sprintf("%s(%s)", i, label))
                    }))
snpsIn["term"]<-unlist(lapply(snpsIn$term, function(i){strsplit(i, "[.]")[[1]][1]}))
tbl<-inner_join(tbl, snpsIn, by="term")
tbl["term"]<-sprintf("%s(%s)", tbl$term, tbl$percent)
tbl["h2g"]<-round((tbl$percent_val * exp(tbl$estimate))*100, 1)
snp_enrichment_plot(tbl[tbl$raw_term!="thymo_resting.1" & tbl$trait %in% test,], y.order = cols , trait="")  + facet_grid(. ~ trait, scales="free_x")
#snp_enrichment_plot(tbl[tbl$term!="thymo_resting" & tbl$trait %in% control,], y.order = cols , trait="")  + facet_grid(. ~ trait, scales="free_x") 
```

*Percent of causal variants in each peak set*

**Motivation**: To not only test for enrichment, but also estimate the percent of genetic signals explained by each set of peaks 

**Procedure**:  
For each disjoint set of peaks, I multiplied the percent of genome-wide SNPs that occur in each peak set with the exponentials of Torus enrichment estimates. The number in parenthesis is the percent of SNPs in peaks. 

**Results**:
```{r echo=F, fig.width=10}

total_h2g<-unlist(lapply(split(tbl, tbl$trait), function(i){
  sprintf("%s(%s)", unique(i$trait),paste0(sum(i$h2g),"%"))
}))

ggplot(tbl, aes(x=term, y=h2g, fill= trait)) + 
    scale_fill_discrete(name="Trait (Total h2g)", labels=total_h2g, type= palette.colors(palette="Set2")) + 
    geom_bar(stat="identity", position="dodge") + ylab("Percent of h2g")+ 
    theme(axis.text.x=element_text(angle=45, vjust = 0.9, hjust=0.8)) 
```

### Disjoint immune peaks in blood vs. immune groups in lung
**Motivation**:  
We will do a joint analysis on three disjoint peak sets - only in blood, only in lung and shared. In this way, we can both estimate enrichments and compute for percent of genetic signals explained by each set.

**Procedure**:  
I started from the cell types defined with disjoint sets of immune peaks from Caldero2019 dataset. For each cell type, I use `bedtools intersect` to call the overlapped peaks and peaks that are unique to either side for lung and blood peaks. Then, the overlapped peaks will be merged.

*Enrichment of peak sets*
```{r echo=F, fig.width=12}
snpsIn<-read.table("output/AAD/Caldero2019_disjoint_snps.sumstats", header=F, row.names = 1)
colnames(snpsIn)<-c("counts", "percent", "term")
snpsIn["label"]<-paste0(round(snpsIn$percent*100, 1), "%")
tbl<-data.frame()
for(trait in traits){
  df<-read.table(sprintf("output/AAD/%s/Caldero2019/Caldero2019_compare.txt", trait), header=F)
  colnames(df)<-c("raw_term", "estimate", "low", "high")
  df["tmp"] = unlist(lapply(df$raw_term, function(i){strsplit(i[1], "[.]")[[1]][1]}))
  tbl<-data.frame(rbind(tbl, cbind(df, trait)))
}
tbl["term"]<-unlist(lapply(tbl$tmp, function(i){
  return(sprintf("%s(%s)", i, snpsIn[snpsIn$term == i, "label"]))
}))
tbl["percent"]<-unlist(lapply(tbl$tmp, function(i){
  return(snpsIn[snpsIn$term == i, "percent"])
}))
tbl_sub<-tbl[!(tbl$tmp %in% c("NK_common","Myeloid_blood", "NK_blood")),]
snp_enrichment_plot(tbl_sub[tbl_sub$trait %in% test,], trait="") + facet_grid(. ~ trait)
```
*Percent of causal variants in each peak set*
```{r echo=F}
tbl["celltype"]<-unlist(lapply(tbl$tmp, function(i){strsplit(i, "[_]")[[1]][1]}))
tbl["annot"]<-unlist(lapply(tbl$tmp, function(i){rev(strsplit(i, "[_]")[[1]])[1]}))
tbl["h2g"]=round(exp(tbl$estimate)*tbl$percent*100, 1)
tbl$celltype<-factor(tbl$celltype, levels = c("T", "B", "Myeloid", "NK"), 
                                    labels=c("T cells", "B cells", "Myeloid cells", "NK cells"))

ggplot(tbl[tbl$trait %in%test,], aes(x=celltype, y=h2g, fill= c(annot))) + 
    geom_bar(stat="identity") + ylab("Percent of h2g") + 
    scale_fill_discrete(name="Annotation sets", labels=c("Blood Only", "Common", "Lung Only"), type= palette.colors(palette="Set3")[c(1,6,5)]) + 
    theme(axis.text.x=element_text(angle=45, vjust = 0.9, hjust=0.8)) + facet_grid(.~trait)
```


<!-- ### [Archive] Disjoint immune peaks in blood vs. immune groups in lung -->

<!-- **Motivation**:  -->
<!-- There is a large fraction of overlaps in peaks across blood's immune groups. Thus we used disjoint peaks to build blood annotations and then ran TORUS together with main immune groups in lung to better estimate their enrichment separately.  -->

<!-- **Procedure**:   -->
<!-- I merged peaks from three groups of T-cells, which are T-stimulated, T-resting and BandT_stimulated cells. Same process for B cells. Then I ran TORUS on merged T cells, B cells, NK resting cells, myeloid resting cells from Caldero2019 with main immune groups in lungs.  -->

<!-- **Results**:     -->
<!-- NK disjoint peaks in blood was not displayed due to very large standard error. We can still see there is  difference in T cells between blood and lung and the extent of difference varies across traits. Other immune cell types in blood do not show significance probably due to small number of disjoint peaks. Since the peaks in lungs are not disjoint, it is expected to see a bias towards lung. Nonetheless, compared with controls, we observe a larger difference in enrichment of risk variants in T cells for AAD.  -->
<!-- ```{r echo=F, fig.width=12} -->
<!-- tbl<-data.frame() -->
<!-- for(trait in traits){ -->
<!--   df<-read.table(sprintf("output/AAD/%s/Caldero2019_disjoint_compare.est", trait), header=T, skip=1) -->
<!--   colnames(df)<-c("raw_term", "estimate", "low", "high") -->
<!--   df["term"] = unlist(lapply(df$raw_term, function(i){strsplit(i[1], "[.]")[[1]][1]})) -->
<!--   tbl<-data.frame(rbind(tbl, cbind(df, trait))) -->
<!-- } -->
<!-- tbl<-tbl[tbl$term!="NK_disjoint_compare_blood",] -->
<!-- snp_enrichment_plot(tbl[tbl$trait %in% test,], trait="",  -->
<!--                     y.label = c("B_cell_blood", "B_cell_lung", "Myeloid_blood", -->
<!--                                 "Myeloid_lung","NK_cell_lung", "T_cell_blood",  -->
<!--                                 "T_cell_lung" -->
<!--                                 )) + facet_grid(. ~ trait, scale="free_x") -->
<!-- ``` -->

```{r echo=F, eval=F}
#### Compute for the percent of SNPs in annotations
dataPath<-"/home/jinggu/cluster/data/features/formatted/Caldero2019/lineage"
tbl<-c()
#build IRange object for all QC SNPs in 1kg 
snpRanges<-make_ranges(seqname=bigSNP$map$chromosome, 
                       start = bigSNP$map$physical.pos,
                       end = bigSNP$map$physical.pos)
snpRanges<-plyranges::mutate(snpRanges, snp=bigSNP$map$marker.ID)
for (f in list.files(dataPath, pattern = "bed")){
  curr <- rtracklayer::import(paste(dataPath, f, sep='/'), format='bed')
  subdf <- IRanges::subsetByOverlaps(snpRanges, curr)
  tbl<-data.frame(rbind(tbl, c(f, length(unique(subdf$snp)), 
                    length(unique(subdf$snp))/dim(bigSNP$map)[1])
  ))
}
colnames(tbl)<-c("term", "count", "percent_val")
tbl["percent"]<-paste0(round(as.numeric(tbl$percent_val)*100, 1), "%")
write.table(tbl, "~/projects/funcFinemapping/output/AAD/Caldero2019_lineage_snps.sumstats",
            sep='\t', row.names = F, quote=F)
```




    
