---
title: "Post-transcriptional regulation"
Date: 08/30/21
output:
  html_document:
    df_print: paged
---
```{r echo=F}
library(ggplot2)
```

## Assessing the utility of novel annotations
In this analysis, we aim to assess how much disease heritability can be explained by post-transcritional regulation processes. GWAS SNPs were first annotated by either functional data or computational predictions. For each annotation category, S-LDSC computed partitioned LD scores using referencel panel and then regressed $\chi^2$ statistics against them to estimate the coefficient $\hat\tau$ per category. This estimate would be further used to compute heritability.

Several post-transcriptonal features will be explored:

* alternative splicing
* RNA modification: m6A
* RNA binding 
* Polyadenylation

## Method(S-LDSC)
Technical details:

1. HapMap3 SNPs (~1.2M): used as a proxy for well imputed SNPs and the regression SNPs in LDSC. They were generated by the international HapMap project on a collection of 1301 samples from a variaty of human populations.[https://www.sanger.ac.uk/resources/downloads/human/hapmap3.html]

2. Reference panel SNPs (~9M): the set of 1000G SNPs with MAF > 5% in ~500 Euproean samples

3. LDSC inputs/outputs:

  * .M/.l2.M_5_50: The .M file contains the total number of SNPs; the .l2.M_5_50 file contains the number of SNPs with minor allele frequency above 5%. By default, ldsc uses common SNPs (MAF > 5%) to estimate per SNP heritability, which is different for rare variants.

  * freqfile/w-ld-chr: computed on European of Phase 3 of 1kg Genomes

  * Regression weights: used to correct for non-independence and heteroskedasticity among the $\chi^2$ statistics. 

  * Regression coefficients: Quoted from the website that "They measure the additional contribution of one annotation to the model and are interpretable for both binary and continuous annotations"

### Outputs
The third column corresponds to the proportion of heritability explained. 
```{r echo=F}
tbl<-read.table("output/splicing/scz_PTR_annotations.results", header = T)
colnames(tbl)[1]<-"Category"
head(data.frame(tbl[, 1:5], row.names = NULL))

```

```{r echo=F}
plot_ldsc_enrichment<-function(df, tolabel=TRUE, tohighlight=NULL){
  if(is.null(tohighlight)){
    tohighlight=waiver()
    df["fill"]="TRUE"
  }else{
    #df$Category[(dim(df)[1]-length(tohighlight)+1):dim(df)[1]]<-tohighlight
    df["fill"]<-df$Category %in% tohighlight
  }
  # if tolabel=TRUE, label the plot with percent of heritability explained.
  if(tolabel){
    df["h2g"]<-paste0(round(df$Prop._h2*100,1), "%")
    p1<-ggplot(df, aes(x=Category, y=Enrichment, color=Category, label=h2g))
    p1<-p1 + geom_text(hjust=0, nudge_x=0.05) 
    }else{
      p1<-ggplot(df, aes(x=Category, y=Enrichment, color=Category))
    }
  
  p1<- p1 + geom_point() + 
            geom_errorbar(aes(ymin=Enrichment-Enrichment_std_error,
                              ymax=Enrichment+Enrichment_std_error,
                              width=.1)) + 
            theme(axis.text.x=element_text(angle=90), legend.position = "none") +
                      geom_hline(yintercept = 1, linetype="dashed")
                #scale_fill_manual(values=c("FALSE"="grey", "TRUE"="orange"), guide=F) 
  return(p1)
}
```
## Results

### test annotations one at a time
**Schizophrenia**

baseline model: baselineLD-v1.1
annotations: spidex, spliceai, fetal brain m6a, neuronal OCR, differentially methylated regions in brains (DMR)

Legend:

  * x-axis - annotations
  * y-axis - fold of enrichment 
  * label on the plot - percent of narrow-sense heritability 
  * dashed line - no enrichment
```{r echo=F, eval=F}
ldsc.dir<-"~/cluster/projects/ldsc_enrichment"
spidex.dir<-"/project2/xinhe/cdu/ldsc/results/splicing/Schizophrenia/baselineLDv1.1"
traits<-c("LDL", "HDL", "bmi", "height", "aFib", "allergy", "asthma_adult", "scz")
annotations<-c("m6a", "neuOCR", "DMR","spliceai_binary0.01", 
               "spliceai_binary0.03","spliceai_binary0.05",
               "spliceai_binary0.07","spliceai_binary0.1",
               "spliceai_binary0.2")
tbl<-c()
for(annot in annotations){
    f<-read.table(sprintf("%s/%s/baselineLD_v1.1/%s.results", ldsc.dir, "scz", annot), header=T)
    tbl<-rbind(tbl, cbind(annot, f[dim(f)[1],-1]))
}

annotations2<-c("top10/Schizophrenia_hg19_spidex_0.10_baselineLDv1.1",
                "top2/Schizophrenia_hg19_spidex_0.02_baselineLDv1.1",
                "top5/Schizophrenia_hg19_spidex_0.05_baselineLDv1.1")
for(annot in annotations2){
    f<-read.table(sprintf("%s/%s.results", spidex.dir, annot), header = T)
    annot<-paste0("spidex_", strsplit(annot, "/")[[1]][1])
    tbl<-rbind(tbl, cbind(annot, f[dim(f)[1],-1]))
}
```

```{r echo=F}
plot_ldsc_enrichment(tbl)
```

**Examine baseline annotations**

A list of annotations with >1.5 fold of enrichment were listed. GERP.NS is suspicious as both proportion of SNPs in category and heritability exceeds one. Besides that, we see heritability is mainly explained by conservation and transcriptional regulation.

test annotation:spliceai score >=0.03

```{r fig.width=12, fig.height=8}
f<-read.table("output/splicing/scz_spliceai_binary0.03.results", header = T)
plot_ldsc_enrichment(f[f$Enrichment>=1.5 & f$Enrichment<10  , ], tolabel = TRUE) 
```

**Across multiple traits**

baseline model: baselineLD-v1.1  
spidex annotations were not included.
```{r echo=F, eval=F}
ldsc.dir<-"~/cluster/projects/ldsc_enrichment"
spidex.dir<-"/project2/xinhe/cdu/ldsc/results/splicing/Schizophrenia/baselineLDv1.1"
traits<-c("LDL", "HDL", "bmi", "height", "aFib", "allergy", "asthma_adult", "scz")
annotations<-c("m6a", "neuOCR", "DMR","spliceai_binary0.01", 
               "spliceai_binary0.03","spliceai_binary0.05",
               "spliceai_binary0.07","spliceai_binary0.1")
tbl<-c()
for(annot in annotations){
  for(trait in traits){
    f<-read.table(sprintf("%s/%s/baselineLD_v1.1/%s.results", ldsc.dir, trait, annot), header=T)
    tbl<-rbind(tbl, cbind(trait, annot, f[dim(f)[1],-1]))
  }
  
}
colnames(tbl)[2]<-"Category"
```

```{r echo=F}
traits<-c("LDL", "HDL", "bmi", "height", "aFib", "allergy", "asthma_adult", "scz")
tbl<-read.table("output/splicing/PTR_across_traits_annotations.results", header = T)
plot_ldsc_enrichment(tbl[tbl$trait %in% traits[1:4], ], tolabel = FALSE) + facet_grid(. ~ trait)
plot_ldsc_enrichment(tbl[tbl$trait %in% traits[5:8], ], tolabel = FALSE) + facet_grid(. ~ trait)
```

### test annotations jointly

```{r echo=F}
annot<-"neuOCR_m6a_DMR"
f<-read.table(sprintf("output/splicing/%s_%s.results", "scz", annot), header=T)
Category<-strsplit(annot, "_")[[1]]
f.out<-f[(dim(f)[1]-2):dim(f)[1], -1]
f.out<-cbind(Category, f.out)
plot_ldsc_enrichment(f.out) + ggtitle("SCZ joint analysis")
```

### splicing predictions diagnosis

#### SCZ
A left-skewed p-value distribution for SNPs with spliceAI >=0.03 

```{r echo=F}
gwas<-readRDS("~/cluster/data/gwas/gwas_ldsc/scz_gwas_spliceai.rds")
pred.splicing<-gwas$spliceAI_varPred>=0.03

#histogram for pvals of SNPs with spliceAI scores above threshold
hist(gwas[which(pred.splicing==TRUE),]$pval, 
     main="SCZ",
     xlab="p-values for SNPs with SpliceAI score >=0.03")

#Scatter plot for comparing p-vals agasint spliceai scores
common<-gwas[gwas$spliceAI_varPred>0,]
plot(y=-log10(common$pval), x=common$spliceAI_varPred, 
     ylab="-log10(p-values)", xlab="spliceAI scores")
```
Note: Very few SNPs with high spliceAI scores also have significant p-values.

**QQ plot**   

```{r echo=F, fig.width=12, fig.height=10}
par(mfrow = c(1,2))
#QQ plot against uniform distribution
qqplot(y=-log10(gwas[which(pred.splicing==TRUE), ]$pval), 
       x=-log10((1:sum(pred.splicing, na.rm=TRUE))/sum(pred.splicing, na.rm=TRUE)),
       xlab="-log10(Expected p-values under uniform dist.)", ylab="-log10(Observed p-values)")
abline(a=0, b=1, col="red")


#QQ plot against the distribution of log p-values for randomly sampled SNPs
qqplot(y=-log10(gwas[which(pred.splicing==FALSE), ]$pval), 
        x=-log10(gwas[sample(1:dim(gwas)[1], sum(pred.splicing, na.rm=TRUE), replace=FALSE),]$pval),
        xlab="-log10(p-values for randomly sampled SNPs)", ylab="-log10(Observed p-values)")
abline(a=0, b=1, col="red")
```
Note: There is not much signal in SNPs that have spliceAI scores above threshold. 

#### AFib

**QQ plot **

[QQ plots for aFib GWAS SNPs with spliceAI>=0.03](splicing/aFib_spliceAI0.03_QQplot.png)

```{r echo=F, eval=F}
# gwas<-fread("~/cluster/data/gwas/0_gwas/nielsen-thorolfsdottir-willer-NG2018-AFib-gwas-summary-statistics_posGRCh37.tbl")
# colnames(gwas)[1:2]<-c("CHR", "BP")
# gwas$CHR<-as.numeric(gwas$CHR)
# gwas<-left_join(gwas, spliceai, by=c("CHR", "BP"))
# saveRDS(gwas, "~/cluster/data/gwas/gwas_ldsc/aFib_gwas_spliceai.rds")
```
```{r echo=F, fig.width=12, fig.height=10}
trait<-readRDS("~/cluster/data/gwas/gwas_ldsc/afib_gwas_spliceai.rds")
trait$pval<-as.numeric(trait$pval)
pred.splicing<-trait$spliceAI_varPred>=0.03

#Scatter plot for comparing p-vals agasint spliceai scores
common<-trait[trait$spliceAI_varPred>=0.03,]
plot(y=-log10(common$pval), x=common$spliceAI_varPred, 
     ylab="-log10(p-values)", xlab="spliceAI scores")

par(mfrow = c(1,2))

#QQ plot against uniform distribution
qqplot(y=-log10(trait[which(pred.splicing==TRUE), ]$pval), 
       x=-log10((1:sum(pred.splicing, na.rm=TRUE))/sum(pred.splicing, na.rm=TRUE)),
       xlab="-log10(Expected p-values under uniform dist.)", ylab="-log10(Observed p-values)")
abline(a=0, b=1, col="red")


#QQ plot against the distribution of log p-values for randomly sampled SNPs
qqplot(y=-log10(trait[which(pred.splicing==FALSE), ]$pval),
       x=-log10(trait[sample(1:dim(gwas)[1], sum(pred.splicing, na.rm=TRUE), replace=FALSE),]$pval),
       xlab="-log10(p-values for randomly sampled SNPs)", ylab="-log10(Observed p-values)")
abline(a=0, b=1, col="red")
```
