---
title: "LDSC_enrichment"
Date: 08/30/21
output:
  html_document:
    df_print: paged
---
```{r echo=F}
library(ggplot2)
source("code/make_plots.R")
```

## Partitioned-LDSC

### Technical details

HapMap3 SNPs (~1.2M): used as a proxy for well imputed SNPs and the regression SNPs in LDSC. They were generated by the international HapMap project on a collection of 1301 samples from a variaty of human populations.[https://www.sanger.ac.uk/resources/downloads/human/hapmap3.html]

Reference panel SNPs (~9M): the set of 1000G SNPs with MAF > 5% in ~500 Euproean samples

LDSC inputs/outputs:
* .M/.l2.M_5_50: The .M file contains the total number of SNPs; the .l2.M_5_50 file contains the number of SNPs with minor allele frequency above 5%. By default, ldsc uses common SNPs (MAF > 5%) to estimate per SNP heritability, which is different for rare variants.

* freqfile/w-ld-chr: computed on European of Phase 3 of 1kg Genomes

Regression weights: used to correct for non-independence and heteroskedasticity among the $\chi^2$ statistics. 

Regression coefficients: Quoted from the website that "They measure the additional contribution of one annotation to the model and are interpretable for both binary and continuous annotations"

GC correction:





Baseline annotations:  
DHS, H3K4me1, H3K4me3, and H3K9ac - peaks w/o flanking regions 
Coding, Conserved, CTCF, DGF   
FANTOM5-Enhancer, Enhancer, Fetal_DHS, H3K27ac    
Intron, PromoterFlanking, Promoter, Repressed, Super-enhancer, TFBS, Transcribed TSS    
3-prime UTR, 5-prime UTR, Weak Enhancer    

### Schizophrenia  
**NeuronalATAC+SpliceAI+baselineLD**
```{r echo=F}
outDIR<-"/home/jinggu/cluster/projects/ldsc_enrichment"
traits<-c("scz")
df<-read.table(sprintf("%s/%s/scz_spliceai_cutoff0.1.results", outDIR, traits[1]), header=T)
test<-c("test_neuronal_atac", "test_spliceAI")
plot_ldsc_enrichment(df,test)
```

### Across traits 

```{r echo = F}
traits=c("aFib", "bmi", "height", "LDL", "HDL", "allergy","scz")
h2g<-c(0.0239, 0.2162, 0.5294, 0.2854, 0.349, 0.0771, 0.4296)
M<-c(1171995, 1009244, 1007580, 1017901, 1018947,1165717, 1144496)
df<-c()
for(i in 1:length(traits)){
  cutoffs<-c(0.01, 0.03, 0.05, 0.07, 0.1)
  for(j in cutoffs){
    f<-read.table(sprintf("%s/%s/%s_spliceai_cutoff%s.results", outDIR,traits[i],traits[i],j), header =T)
    fsub<-tail(f,1)
    fsub["trait"]<-traits[i]
    fsub["cutoff"]<-j
    fsub["h2g"]<-h2g[i]
    fsub["M"]<-M[i]
    df<-data.frame(rbind(df, fsub), row.names = NULL)
  }
}
df["std_tau"]=df["Coefficient"]*df["Coefficient_std_error"]*df["M"]/df["h2g"]
```
per-standardized-annotation effect sizes 
```{r echo=F}
p<-ggplot(df, aes(x=trait, y=std_tau, fill=factor(cutoff))) + 
              geom_bar(stat="identity", position=position_dodge()) + ylab("Standardized Tau")
              scale_fill_discrete(name="Annotations", 
                                  labels=paste0("cutoff", c(0.01,0.03,0.05,0.07,0.1)))
p
```
Heritability Enrichment across traits and annotations
```{r echo=F}
p<-ggplot(df, aes(x=trait, y=Enrichment, fill=factor(cutoff))) + 
              geom_bar(stat="identity", position=position_dodge()) +
              geom_errorbar(aes(ymin=Enrichment-Enrichment_std_error,
                      ymax=Enrichment+Enrichment_std_error),
                      width=.2, position=position_dodge(.9)) + 
              scale_fill_discrete(name="Annotations", 
                                  labels=paste0("cutoff", c(0.01,0.03,0.05,0.07,0.1)))
p
```


