---
title: "results"
output: html_notebook
---

Fine-mapping with functional annotations as priors has shown improved results in identifying causal variants. This project is to evaluate the utility of novel annotation features and adopt ones that can improve fine mapping results. 

## Evaluation  
### GWAS summary statistics   
Schizopherenia - Pardinas et al., 2018 

* 40675 cases and 64643 controls
* CLOZUK sample + PGC sample (independent)
* 179 independent GWAS significant SNPs mapped to 145 independent loci
* SNPs were imputed using a combination of the 1KGPp3 and UK 10K datasets. 
* SNPs were filtered by NFO > 0.6 and MAF > 0.01
* LD-score regression analysis: An LD reference was generated from 1KGPp3 after restricting this dataset to strictly unrelated individuals and retaining only markers with MAF > 0.01.

### GWAS QC Procedures

* Current procedures was based on Alan's finemappeR pipeline
* Criteria for filtering gwas SNPs 

1. Remove all non-biallelic SNPs
2. Remove all SNPs with strand-ambiguous alleles (SNPs with A/T, C/G alleles)
3. Removed SNPs without rs IDs, duplicated rs IDs or base pair position.
4. Removed SNPs not in the reference panels
5. Removed SNPs whose base pair positions or alleles doesn’t match the reference panels
6. Removed all SNPs on chromosome X, Y, and MT

After filtering, there are around 6 million variants remained. 

![Plots for GWAS summary statistics](assets/gwas_summary_stats_hist_plot.png)
```{r echo=F}
# png("docs/assets/gwas_summary_stats_hist_plot.png", width=960, height=480)
# par(mfrow = c(1,2))
# hist(gwas$zscore, main="", xlab="Z scores")
# hist(gwas$pval, main="", xlab="P values")
# dev.off()
```

### Features   
1. Sequence constraints:  
    * context-dependent tolerance scores(CDTS) in percentiles
    * A score was computed for each 10bp bin in the genome. 
    * The lower the score is, the more intolerant to variation is the bin.

### Procedures
1. GWAS summary statistics was pre-processed to remove sex chromosomes, indels, ambiguous and duplicated SNPs. 
2. Currently, genotypes from 1kg European samples are used to compute LD between SNPs. 
3. SNPs in GWAS summary statistics were matched with the reference panel and assigned to in total 1687 independent LD blocks. 
4. Run TORUS to perform genome-wide enrichment analyses.

### Results  
All variants were catogrized into whether or not they occur in genomic bins with CDTS up to 1 percentile or 5 percentile.   

**Examine the CDTS feature** 

check the proportion of variants with high sequencing constraints that also have functional annotations in brain
```{r echo=F}
library(data.table)
library(knitr)
# f<-fread("data/joint_torus_annotations.txt.gz")
# fnames<-unlist(lapply(colnames(f)[-3:-1], function(i){strsplit(i, "_hg19")[[1]][1]}))
# colnames(f)[c(-3:-1)]<-fnames
# f[,"Any_OCR"]<-as.numeric(apply(f[,6:dim(f)[2]], 1, sum) > 0)
# f_sum<-apply(f[, c(-3:-1)], 2, sum)
# print("total number of variants overlapped with annotations")
# #CDTS_1per
# output<-unlist(lapply(6:dim(f)[2], function(i){
#   return(sum(apply(f[, c(4, i), with=F], 1, sum)>1))
# }))
# output2<-unlist(lapply(6:dim(f)[2], function(i){
#   return(sum(apply(f[, c(5, i), with=F], 1, sum)>1))
# }))
# tbl<-data.frame(
#               rbind(paste0(round(output/sum(f[,4])*100,1),"%"),
#               paste0(round(output2/sum(f[,5])*100,1), "%"))
#               )
# rownames(tbl)<-c("CDTS_1%", "CDTS_5%")
# colnames(tbl)<-names(f)[6:dim(f)[2]]
kable(read.table("output/prop_constrained_variants_with_OCR.txt", header = T), caption="Summary of percentage of genetic variants within up to one/five percentile of CDTS that overlapped with OCRs in brain")
```
Check the percent of constrained sequences that overlaps with open chromatin regions from neurons

* Overlaps between two sets of genomic features were identified using `bedtools intersect`. The constrained sequences were counted to be overlapped when at least 10%(>=1 bp) intersect with peaks called from ATAC-Seq profiles.  
```{r echo=F}
kable(read.table("data/constrained_and_OPC.txt", header = T),
      caption="Summary of the overlappings between constrained sequences and chromatin accessibility profiles across multiple tissues")
```
```{r echo=F}
kable(read.table("data/constrained_and_genomic_anno.txt", header = T),
      caption="Summary of overlapping between CDTS and genomic annotations")
# counts_1per=2056917
# counts_5per=10582523
# f<-fread("~/cluster/data/features/formatted/seq_constraint/CDTS_1per_overlapped_genomic_anno_full.bed")
# out<-unlist(lapply(split(f, f$V4), function(i){length(unique(i$V8))/counts_1per}))
# f<-fread("~/cluster/data/features/formatted/seq_constraint/CDTS_5per_overlapped_genomic_anno_full.bed")
# out2<-unlist(lapply(split(f, f$V4), function(i){length(unique(i$V8))/counts_5per}))
# tbl<-data.frame(rbind(out, out2))
# rownames(tbl)<-c("CDTS_1%", "CDTS_5%")
# write.table(round(tbl*100,1), "data/constrained_and_genomic_anno.txt", quote = F)
```
**Summary**:
The overlap of CDTS with open chromain regions in neurons much higher than immure cells. In our analysis, the overlap with enhancers is low. However, CDTS shows high overlapping with exons, introns and promoters. 


**Enrichment analysis for sequence constraints**
```{r echo=F}
df<-read.table("data/torus_enrichment.est", header = F)
colnames(df)<-c("term", "estimate", "low", "high")
library(ggplot2)
ggplot(df, aes(y=term, x=log2(exp(estimate)))) +
  geom_point(size=4) + 
  geom_errorbar(aes(xmax=log2(exp(high)), xmin=log2(exp(low)))) + 
  xlab("SCZ variant enrichment(log2)") + 
  scale_y_discrete(labels=c("CDTS_1Percentile", "CDTS_5Percentile",
                            "iN_Dopa OCR", "iN_GABA OCR", 
                            "iN_Glut OCR", "iPSC OCR", "NPC OCR"))
```
The enrichment estimate has a confience level above zero for CDTS and positive controls. This shows SNPs associated with SCZ are on average ~ 9 fold enriched in genomic bins with up to 5 percentile of CDTS. 
```{r echo=F}
## process other conservation related annotations - CADD, LINSIGNT, and GERP
# f<-fread("~/cluster/data/features/processed/scz_2018_CADD_chunk10.txt")
# colnames(f)<-c("chr", "pos", "a1", "a0", "score", "phred")
# out<-list()
# for(i in 1:11){
#   print(i)
#   f<-fread(sprintf("~/cluster/data/features/processed/scz_2018_CADD_chunk%d.txt", i))
#   colnames(f)<-c("chr", "pos", "a1", "a0", "score", "phred")
#   out[[i]]<-snp_match(gwas[gwas$chr %in% unique(f$chr),], f)
# }
# final_out<-data.frame(do.call(rbind, out))
# 
# out<-list()
# for(i in 1:6){
#   f<-fread(sprintf("/home/jinggu/cluster/data/features/processed/scz_2018_LS_chunk%d.txt", i))
#   f<-f[, c(1,2,5)]
#   colnames(f)<-c("chr", "pos", "LS_score")
#   f[, "chr"]<-as.numeric(unlist(lapply(f$chr, function(i){strsplit(i, "r")[[1]][2]})))
#   subgwas<-out_LS[out_LS$chr %in% unique(f$chr),]
#   out[[i]]<-merge(subgwas, f, by=c("chr", "pos"))
# }
# out<-data.frame(do.call(rbind, out))
# 
# cond1<-as.numeric(out$LS_score>=0.24)
# cond2<-as.numeric(out$phred>=13)
# cond3<-as.numeric(out$G_score>=2)
# out<-out[!is.na(out$LS_score),]
# write.table(out[, c("snp", "locus","zscore")], gzfile("/home/jinggu/cluster/projects/finemap_pipeline/output/scz_2018/joint/torus_zscores.txt.gz"), quote = F, row.names = F)
# 
# write.table(data.frame(out[, c("snp", "CDTS_5per_d")], cond1, cond2, cond3), gzfile("/home/jinggu/cluster/projects/finemap_pipeline/output/scz_2018/joint/torus_annotations.txt.gz"), quote = F, row.names = F, col.names = c("snp", "CDTS_5per_d", "LINSIGHT_24per_d", "CADD_5per_d", "GERP_2cutoff_d"))
```
**Compare with other conservation annotations**

* LINSIGHT 
    - predict how noncoding nucleotide sites are likely to have deleterious fitness consequences and hence be phenotypically important 
    - genome-wide average of LINSIGHT scores was ~0.07 (range: 0.03-0.99)
    - Estimated mean LINSIGHT score for conserved TFBSs was 0.24->used as cutoff for whether the nucelotide site is conserved
    - 2.5% of GWAS SNPs are above LINSIGHT threshold.    
  
* CADD - Combined Annotation–Dependent Depletion,
    - provides metrics of deleteriousness
    - scaled PHRED score [-10log10(P)]
    - 5 percent chosen as a cutoff, which represents top 5% of all possible reference genome SNVs
 
  
* GERP - Genomic Evolutionary Rate Profiling 
    - produce position-specific estimates of evolutionary constraint
    - constraint intensity quantified as a "rejection score" range from -12.3 to 6.17
    - UCSC suggests a RS score threshold of 2 which provides high sensitivity and strongly enriched for true constraint sites

Comparing the top k-th percentile of CDTS bins vs top k-th percentile scores of other annotations

* CDTS vs CADD (genome-wide top 5%)  
For the top 5% of CDTS bins, 15.4% of the bases have CADD scores above 5%. 
For the SNPs with top 5% CADD scores, 10.8% of them fall into top 5% CDTS bins. 
* CDTS vs LINSIGHT (per-chromosome top 10%)   
For the top 10% of CDTS bins, 1% of the bases have LINSIGHT scores above 10%. 
For the SNPs with top 10% CADD scores, 12% of them fall into top 10% CDTS bins. 
* CDTS vs GERP (per-chromosome top 10%)  
For the top 10% of CDTS bins, 8.7% of the bases have GERP scores above 10%. 
For the SNPs with top 10% CADD scores, 11.7% of them fall into top 10% CDTS bins. 

**Summary**:

1. Overall, there is a maximum of 15.4% overlapping between the bases within the top CDTS bins and those ranked among top annotation scores. 
2. Only 1% of bases from top 10% CDTS bins have LINSIGHT scores above 10%, probably due to many missing predictions in LINSIGHT. 

```{r echo=F}
#extract raw cdts score at each nucleotide position
# cdts<-fread("~/cluster/data/features/raw/coord_CDTS_percentile_N7794unrelated.txt.gz")
# snpRanges<-make_ranges(paste0("chr",out$chr), out$pos, out$pos)
# snpRanges <- plyranges::mutate(snpRanges, snp=out$snp)
# cdts_ranges<-make_ranges(cdts$chromosome, cdts$start, cdts$end)
# cdts_ranges<-plyranges::mutate(cdts_ranges, score=cdts$CDTS)
# subsetdf <- IRanges::mergeByOverlaps(cdts_ranges, snpRanges)
# df<-inner_join(out, data.frame(subsetdf)[,c("score", "snp")], by="snp")
#gwas<-readRDS("~/cluster/data/gwas/2_Filtered/scz_2018_conservation.rds")

############
# library(data.table)
# source("~/scripts/R/utility_func.R")
# cdts<-fread("~/cluster/data/features/raw/coord_CDTS_hg19.txt.gz")
# colnames(cdts)<-c("chromosome", "start", "end", "CDTS", "percentile")
# cdts<-cdts[cdts$chromosome %in% paste0("chr",seq(1, 22, 1)),]
# cdts_top5<-subset(cdts, percentile<=5)
# 
# cadd<-fread("~/cluster/data/features/raw/top5_unique_cadd.txt.gz")
# snpRanges<-make_ranges(cadd$V2, cadd$V3, cadd$V3)
# cdts_ranges<-make_ranges(unlist(lapply(cdts_top5$chromosome, function(i){strsplit(i, "r")[[1]][2]})), cdts_top5$start, cdts_top5$end-1)
# count<-IRanges::mergeByOverlaps(cdts_ranges, snpRanges)
# length(unique(count$snpRanges))/(dim(cdts_top5)[1]*10)
# length(unique(count$snpRanges))/dim(cadd)[1]
# 
# cdts_split<-split(cdts, cdts$chromosome)
# top10<-lapply(cdts_split, function(i){
#   cutoff = quantile(i$CDTS, probs=0.1)
#   return(i[i$CDTS<=cutoff,])
#   })
# 
# output<-matrix(NA, nrow = 22, ncol=3)
# for(i in 1:22){
#   print(i)
#   f <-read.table(sprintf("~/cluster/data/features/processed/LS_chr%d.bed", i), header=F)
#   snpRanges<-make_ranges(f$V1, f$V2, f$V2)
#   cdts_top10<-top10[[paste0("chr",i)]]
#   cdts_ranges<-make_ranges(cdts_top10$chromosome, cdts_top10$start, cdts_top10$end-1)
#   count<-IRanges::mergeByOverlaps(cdts_ranges, snpRanges)
#   output[i,]<-c(dim(cdts_top10)[1]*10, dim(f)[1], length(unique(count$snpRanges)))
# }
# 
# counts<-apply(output, 2, sum)
# counts[3]/counts[1:2]

############
#percent of CDTS overlapping other features
# tbl<-cor(data.frame(rank(gwas$phred), rank(gwas$LS_score), rank(gwas$G_score), rank(gwas$score)))
# row.names(tbl)<-c("CADD", "LINSIGHT", "GERP", "CDTS")
# colnames(tbl)<-c("CADD", "LINSIGHT", "GERP", "CDTS")
# write.table(tbl, "data/rank_correlations.txt", quote = F)
print("5 million (68%) SCZ GWAS SNPs have conservation scores across all annotations.")
kable(read.table("data/rank_correlations.txt", header = T),
      caption="Summary of the correlations between ranks for the conservation scores among annotations")
```
The correlation table shows the pair-wise correlations between ranks for the conservation scores on GWAS SNPs across all the annotations. CDTS seems to be uncorrelated to other conservation scores. CADD has a moderate correlation separately with LINSIGHT and GERP. A low correlation observed between GERP and LINSIGHT. 

**joint TORUS enrichment analysis over conservation-related annotations**
```{r echo=F}
df<-read.table("data/joint_torus_conservation_enrichment.est", header = F)
colnames(df)<-c("term", "estimate", "low", "high")
library(ggplot2)
ggplot(df[-1,], aes(y=term, x=log2(exp(estimate)))) +
  geom_point(size=4) + 
  geom_errorbar(aes(xmax=log2(exp(high)), xmin=log2(exp(low))))+xlab("SCZ variant enrichment(log2)")
```
With other conservation annotations as predictors in the model, we can see CDTS within top 5 percentile still shows around 8 fold enrichment. 
