---
title: "results"
output: html_notebook
---

Fine-mapping with functional annotations as priors has shown improved results in identifying causal variants. This project is to evaluate the utility of novel annotation features and adopt ones that can improve fine mapping results. 

## Evaluation  
### GWAS summary statistics   
Schizopherenia - Pardinas et al., 2018 

* 40675 cases and 64643 controls
* CLOZUK sample + PGC sample (independent)
* 179 independent GWAS significant SNPs mapped to 145 independent loci
* SNPs were imputed using a combination of the 1KGPp3 and UK 10K datasets. 
* SNPs were filtered by NFO > 0.6 and MAF > 0.01
* LD-score regression analysis: An LD reference was generated from 1KGPp3 after restricting this dataset to strictly unrelated individuals and retaining only markers with MAF > 0.01.

### GWAS QC Procedures

* Current procedures was based on Alan's finemappeR pipeline
* Criteria for filtering gwas SNPs 

1. Remove all non-biallelic SNPs
2. Remove all SNPs with strand-ambiguous alleles (SNPs with A/T, C/G alleles)
3. Removed SNPs without rs IDs, duplicated rs IDs or base pair position.
4. Removed SNPs not in the reference panels
5. Removed SNPs whose base pair positions or alleles doesn’t match the reference panels
6. Removed all SNPs on chromosome X, Y, and MT

After filtering, there are around 6 million variants remained. 
![Plots for GWAS summary statistics](assets/gwas_summary_stats_hist_plot.png)
```{r echo=F}
# png("docs/assets/gwas_summary_stats_hist_plot.png", width=960, height=480)
# par(mfrow = c(1,2))
# hist(gwas$zscore, main="", xlab="Z scores")
# hist(gwas$pval, main="", xlab="P values")
# dev.off()
```

### Features   
1. Sequence constraints:  
    * context-dependent tolerance scores(CDTS) in percentiles
    * A score was computed for each 10bp bin in the genome. 
    * The lower the score is, the more intolerant to variation is the bin.

### Procedures
1. GWAS summary statistics was pre-processed to remove sex chromosomes, indels, ambiguous and duplicated SNPs. 
2. Currently, genotypes from 1kg European samples are used to compute LD between SNPs. 
3. SNPs in GWAS summary statistics were matched with the reference panel and assigned to in total 1687 independent LD blocks. 
4. Run TORUS to perform genome-wide enrichment analyses.

 
All variants were catogrized into whether or not they occur in genomic bins with CDTS up to 1 percentile or 5 percentile.   

### Examine the CDTS feature
```{r echo=F}
library(data.table)
library(knitr)
# f<-fread("data/joint_torus_annotations.txt.gz")
# fnames<-unlist(lapply(colnames(f)[-3:-1], function(i){strsplit(i, "_hg19")[[1]][1]}))
# colnames(f)[c(-3:-1)]<-fnames
# f[,"Any_OCR"]<-as.numeric(apply(f[,6:dim(f)[2]], 1, sum) > 0)
# f_sum<-apply(f[, c(-3:-1)], 2, sum)
# print("total number of variants overlapped with annotations")
# #CDTS_1per
# output<-unlist(lapply(6:dim(f)[2], function(i){
#   return(sum(apply(f[, c(4, i), with=F], 1, sum)>1))
# }))
# output2<-unlist(lapply(6:dim(f)[2], function(i){
#   return(sum(apply(f[, c(5, i), with=F], 1, sum)>1))
# }))
# tbl<-data.frame(
#               rbind(paste0(round(output/sum(f[,4])*100,1),"%"),
#               paste0(round(output2/sum(f[,5])*100,1), "%"))
#               )
# rownames(tbl)<-c("CDTS_1%", "CDTS_5%")
# colnames(tbl)<-names(f)[6:dim(f)[2]]
# kable(read.table("output/prop_constrained_variants_with_OCR.txt", header = T), caption="Summary of percentage of genetic variants within up to one/five percentile of CDTS that overlapped with OCRs in brain")
```
*Check the percent of constrained sequences that overlaps with open chromatin regions in brain*

* Overlaps between two sets of genomic features were identified using `bedtools intersect`. The constrained sequences were counted to be overlapped when at least 20%(>=2 bp) intersect with peaks called from ATAC-Seq profiles. No reciprocal option used here due to the much shorter length of CDTS sequence. 
```{r echo=F}
kable(read.table("output/constrained_and_OCR.txt", header = T),
      caption="Table 1-Summary of the overlaps between constrained sequences and open chromatin regions across tissues")
```
*Example of top 1% CDTS sequences and genes nearby*   
Conserved sequences found in the noncoding regions nearby gene YPEL4, which is a member of the highly conserved YPEL gene family.  
![chr11:57405334-57407554](assets/chr11_57405334_57407554.png)

**Summary**:   
Overall, CDTS at top percentiles are more enriched in open chromatin regions from brain than immune cells. CDTS at 50% threshold can be regarded as a negative control, which implies that less constrained sequence bins have much less overlapping with OCRs in brain.

*Check how highly conserved CDTS bins overlap with genomic segments including exons, introns, UTRs, etc.*   
Genomic regions were extracted from gencode and processed based on the post from Dave Tang (https://davetang.org/muse/2013/01/18/defining-genomic-regions/). The non-coding CDTS at top 5% were obtained by combining CDTS in top 5% bins completely overlapping both intergenic and intron regions. In total, 91% of original CDTS at top 5% bins are located in the non-coding regions. 
I used `bedtools intersect -a in.bed -b *.regions.bed -wa -wb -f 1` to obtain CDTS bins that are completely overlapped with genomic regions that belong to each category.
```{r echo=F}
kable(read.table("output/constrained_and_genomic_segment.txt", header = T),
caption="Table 2-Summary of overlapping between CDTS and genomic segment annotations")
# counts_1per=2056917
# counts_5per=10582523
# #counts_50per=2398981
# 
# f<-fread("~/resources/genomes/CDTS_1per_genome_overlaps_all.bed")
# out1<-unlist(lapply(split(f, f$V5), function(i){length(unique(i$V1))/counts_1per}))
# 
# f<-fread("~/resources/genomes/CDTS_5per_genome_overlaps_all.bed")
# out2<-unlist(lapply(split(f, f$V5), function(i){length(unique(i$V1))/counts_5per}))
# tbl<-data.frame(rbind(out1, out2))
# tbl<-matrix(paste0(unlist(round(tbl*100,1)), "%"), nrow=2)
# rownames(tbl)<-c("CDTS_1%", "CDTS_5%")
# colnames(tbl)<-names(out1)
# write.table(tbl, "output/constrained_and_genomic_segment.txt", quote = F)

# 
# for(i in c("iNGlut", "iPSC", "b_cell")){
#   f<-fread(sprintf("~/resources/genomes/CDTS_5per_%s_genome_overlaps.bed",i))
#   counts<-length(unique(f$V4))
#   out<-unlist(lapply(split(f, f$V5), function(i){length(unique(i$V4))/counts}))
#   assign(i, out)
# }
# tbl<-data.frame(rbind(iNGlut, iPSC, b_cell))
# tbl<-matrix(paste0(unlist(round(tbl*100, 1)), "%"), nrow=3)
# rownames(tbl)<-c("CDTS5%_iN-Glut", "CDTS5%_iPSC", "CDTS5%_B-cell")
# colnames(tbl)<-names(iNGlut)
# write.table(tbl, "output/CDTS_OCR_overlaps_and_genomic_segment.txt", quote = F)
kable(read.table("output/CDTS_OCR_overlaps_and_genomic_segment.txt", header = T),
caption="Table 3-Breakdown of CDTS and OCR overlaps by genomic segments")
```
**Summary-table1**:   
1. Consistently, the overlap of CDTS with open chromain regions in neurons are 5-10 fold fold higher than immune cells. 
2. As comparison, CDTS at the end of 50% quantile shows much lower percentage of overlaps with OCRs. 

**Summary-table2**  
1. Overall, CDTS top percentile bins are mainly located in intron regions, which is consistent with the results in the paper. 
2. Around 20% of CDTS bins overlap with exons or intergenic regions. 
3. We may look into other annotations such as promoter and enhancers.

**Summary-table3**  
1. The compositions of the CDTS-OCR overlaps in terms of genomic regions are quite consistent across cell types. 
2. Generally, there are twice amount of CDTS-OCR overlaps resided in introns than in exons. 

**Enrichment analysis for sequence constraints**    
*QCs for brain OCR ATAC-seq peaks*  
There are very few peaks with size larger than 5kb across different brain ATAC-Seq profiles. The removal of large peaks did not improve the enrichment results. 
```{r echo =F}

kable(read.table("output/OCR_peaks_summary_stats.txt", header = T),
      caption="ATAC-seq peaks summary statistics")

# out<-unlist(lapply(unique(f$V5), function(i){length(unique(f[f$V5==i,]$V1))/sum(f$V5==i)}))
# out<-paste0(round(out*100,1), "%")
# names(out)<-unique(f$V5)

# fileDIR="/home/jinggu/cluster/data/features/formatted/asoc_science2020"
# annot_list<-read.table(sprintf("%s/annotation_list.txt", fileDIR))
# out_df<-t(sapply(annot_list$V1[-2:-1], function(i){
#                f<-fread(sprintf("%s/%s", fileDIR, i))
#                size<-f$V3-f$V2
#                coverage<-sum(size)
#                return(c(dim(f)[1], coverage, 100*coverage/(3*10^10),sum(size>5000)))
#                }))
# out_df
```

```{r echo=F}
df<-read.table("data/torus_enrichment.est", header = F)
colnames(df)<-c("term", "estimate", "low", "high")
library(ggplot2)
ggplot(df, aes(y=term, x=log2(exp(estimate)))) +
  geom_point(size=4) +
  geom_errorbar(aes(xmax=log2(exp(high)), xmin=log2(exp(low)))) +
  xlab("SCZ variant enrichment(log2)") +
  scale_y_discrete(labels=c("CDTS_1Percentile", "CDTS_5Percentile",
                            "iN_Dopa OCR", "iN_GABA OCR",
                            "iN_Glut OCR", "iPSC OCR", "NPC OCR"))
```
The enrichment estimate has a confience level above zero for CDTS and positive controls. This shows SNPs associated with SCZ are on average ~ 9 fold enriched in genomic bins with up to 5 percentile of CDTS. 
```{r echo=F}
## process other conservation related annotations - CADD, LINSIGNT, and GERP
# f<-fread("~/cluster/data/features/processed/scz_2018_CADD_chunk10.txt")
# colnames(f)<-c("chr", "pos", "a1", "a0", "score", "phred")
# out<-list()
# for(i in 1:11){
#   print(i)
#   f<-fread(sprintf("~/cluster/data/features/processed/scz_2018_CADD_chunk%d.txt", i))
#   colnames(f)<-c("chr", "pos", "a1", "a0", "score", "phred")
#   out[[i]]<-snp_match(gwas[gwas$chr %in% unique(f$chr),], f)
# }
# final_out<-data.frame(do.call(rbind, out))
# 
# out<-list()
# for(i in 1:6){
#   f<-fread(sprintf("/home/jinggu/cluster/data/features/processed/scz_2018_LS_chunk%d.txt", i))
#   f<-f[, c(1,2,5)]
#   colnames(f)<-c("chr", "pos", "LS_score")
#   f[, "chr"]<-as.numeric(unlist(lapply(f$chr, function(i){strsplit(i, "r")[[1]][2]})))
#   subgwas<-out_LS[out_LS$chr %in% unique(f$chr),]
#   out[[i]]<-merge(subgwas, f, by=c("chr", "pos"))
# }
# out<-data.frame(do.call(rbind, out))
# 
# cond1<-as.numeric(out$LS_score>=0.24)
# cond2<-as.numeric(out$phred>=13)
# cond3<-as.numeric(out$G_score>=2)
# out<-out[!is.na(out$LS_score),]
# write.table(out[, c("snp", "locus","zscore")], gzfile("/home/jinggu/cluster/projects/finemap_pipeline/output/scz_2018/joint/torus_zscores.txt.gz"), quote = F, row.names = F)
# 
# write.table(data.frame(out[, c("snp", "CDTS_5per_d")], cond1, cond2, cond3), gzfile("/home/jinggu/cluster/projects/finemap_pipeline/output/scz_2018/joint/torus_annotations.txt.gz"), quote = F, row.names = F, col.names = c("snp", "CDTS_5per_d", "LINSIGHT_24per_d", "CADD_5per_d", "GERP_2cutoff_d"))
```
**Compare with other conservation annotations**

* LINSIGHT 
    - predict how noncoding nucleotide sites are likely to have deleterious fitness consequences and hence be phenotypically important 
    - genome-wide average of LINSIGHT scores was ~0.07 (range: 0.03-0.99)
    - Estimated mean LINSIGHT score for conserved TFBSs was 0.24->used as cutoff for whether the nucelotide site is conserved
    - 2.5% of GWAS SNPs are above LINSIGHT threshold.    
  
* CADD - Combined Annotation–Dependent Depletion,
    - provides metrics of deleteriousness
    - scaled PHRED score [-10log10(P)]
    - 5 percent chosen as a cutoff, which represents top 5% of all possible reference genome SNVs
 
  
* GERP - Genomic Evolutionary Rate Profiling 
    - produce position-specific estimates of evolutionary constraint
    - constraint intensity quantified as a "rejection score" range from -12.3 to 6.17
    - UCSC suggests a RS score threshold of 2 which provides high sensitivity and strongly enriched for true constraint sites

*Compare top k-th percentile of CDTS bins vs top k-th percentile scores of other annotations*
```{r echo=F}
mat1<-matrix(c(1,0.154,0.108,1), nrow = 2, ncol = 2)
colnames(mat1)<-c("top5%_CDTS", "top5%_CADD")
rownames(mat1)<-c("top5%_CDTS", "top5%_CADD")
mat1
```
```{r echo=F}
mat2<-matrix(c(1, 0.01, 0.087, 0.12, 1, NA, 0.12, NA, 1), nrow=3, ncol=3, byrow = T)
colnames(mat2)<-c("top10%_CDTS", "top10%_LINSIGHT", "top10%_GERP")
rownames(mat2)<-c("top10%_CDTS", "top10%_LINSIGHT", "top10%_GERP")
mat2
```
**Summary**:

1. Overall, there is a maximum of 15.4% overlapping between the bases within the top CDTS bins and those ranked among top annotation scores. 

2. Only 1% of bases from top 10% CDTS bins have LINSIGHT scores above 10%, probably due to many missing predictions in LINSIGHT. 

```{r echo=F}
#extract raw cdts score at each nucleotide position
# cdts<-fread("~/cluster/data/features/raw/coord_CDTS_percentile_N7794unrelated.txt.gz")
# snpRanges<-make_ranges(paste0("chr",out$chr), out$pos, out$pos)
# snpRanges <- plyranges::mutate(snpRanges, snp=out$snp)
# cdts_ranges<-make_ranges(cdts$chromosome, cdts$start, cdts$end)
# cdts_ranges<-plyranges::mutate(cdts_ranges, score=cdts$CDTS)
# subsetdf <- IRanges::mergeByOverlaps(cdts_ranges, snpRanges)
# df<-inner_join(out, data.frame(subsetdf)[,c("score", "snp")], by="snp")
#gwas<-readRDS("~/cluster/data/gwas/2_Filtered/scz_2018_conservation.rds")

############
# library(data.table)
# source("~/scripts/R/utility_func.R")
# cdts<-fread("~/cluster/data/features/raw/coord_CDTS_hg19.txt.gz")
# colnames(cdts)<-c("chromosome", "start", "end", "CDTS", "percentile")
# cdts<-cdts[cdts$chromosome %in% paste0("chr",seq(1, 22, 1)),]
# cdts_top5<-subset(cdts, percentile<=5)
# 
# cadd<-fread("~/cluster/data/features/raw/top5_unique_cadd.txt.gz")
# snpRanges<-make_ranges(cadd$V2, cadd$V3, cadd$V3)
# cdts_ranges<-make_ranges(unlist(lapply(cdts_top5$chromosome, function(i){strsplit(i, "r")[[1]][2]})), cdts_top5$start, cdts_top5$end-1)
# count<-IRanges::mergeByOverlaps(cdts_ranges, snpRanges)
# length(unique(count$snpRanges))/(dim(cdts_top5)[1]*10)
# length(unique(count$snpRanges))/dim(cadd)[1]
# 
# cdts_split<-split(cdts, cdts$chromosome)
# top10<-lapply(cdts_split, function(i){
#   cutoff = quantile(i$CDTS, probs=0.1)
#   return(i[i$CDTS<=cutoff,])
#   })
# 
# output<-matrix(NA, nrow = 22, ncol=3)
# for(i in 1:22){
#   print(i)
#   f <-read.table(sprintf("~/cluster/data/features/processed/LS_chr%d.bed", i), header=F)
#   snpRanges<-make_ranges(f$V1, f$V2, f$V2)
#   cdts_top10<-top10[[paste0("chr",i)]]
#   cdts_ranges<-make_ranges(cdts_top10$chromosome, cdts_top10$start, cdts_top10$end-1)
#   count<-IRanges::mergeByOverlaps(cdts_ranges, snpRanges)
#   output[i,]<-c(dim(cdts_top10)[1]*10, dim(f)[1], length(unique(count$snpRanges)))
# }
# 
# counts<-apply(output, 2, sum)
# counts[3]/counts[1:2]

############
#percent of CDTS overlapping other features
# tbl<-cor(data.frame(rank(gwas$phred), rank(gwas$LS_score), rank(gwas$G_score), rank(gwas$score)))
# row.names(tbl)<-c("CADD", "LINSIGHT", "GERP", "CDTS")
# colnames(tbl)<-c("CADD", "LINSIGHT", "GERP", "CDTS")
# write.table(tbl, "data/rank_correlations.txt", quote = F)
# print("5 million (68%) SCZ GWAS SNPs have conservation scores across all annotations.")
# kable(read.table("data/", header = T),
#       caption="Summary of the correlations between ranks for the conservation scores among annotations")
```
The correlation table shows the pair-wise correlations between ranks for the conservation scores on GWAS SNPs across all the annotations. CDTS seems to be uncorrelated to other conservation scores. CADD has a moderate correlation separately with LINSIGHT and GERP. A low correlation observed between GERP and LINSIGHT. 

**joint TORUS enrichment analysis over conservation-related annotations**
```{r echo=F}
df<-read.table("data/joint_torus_conservation_enrichment.est", header = F)
colnames(df)<-c("term", "estimate", "low", "high")
library(ggplot2)
ggplot(df[-1,], aes(y=term, x=log2(exp(estimate)))) +
  geom_point(size=4) + 
  geom_errorbar(aes(xmax=log2(exp(high)), xmin=log2(exp(low))))+xlab("SCZ variant enrichment(log2)") 
```
With other conservation annotations as predictors in the model, we can see CDTS within top 5 percentile still shows around 8 fold enrichment. 

### DNA methylation

**procedure**  

1. concatenate and sort CG DNA methylation regions across 21 clusters    
2. use `bedtools merge -i merged.bed` to merge overlapped regions in the concatenated bed file   
3. check the distribution of peak lengths    
![Histogram for the sizes of DNA methylation regions](assets/human_CG_DMR_peak_length.png)
```{r echo = F}
##QC
# f<-fread("~/cluster/data/features/formatted/DMR/human_CG_DMR.bed")
# f2<-fread("~/cluster/data/features/formatted/DMR/human_large_CG_DMR.bed")
# png("docs/assets/human_CG_DMR_peak_length.png", width = 800, height=480)
# par(mfrow=c(1,2))
# hist(f$V3-f$V2, breaks = 50, main="CG-DMRs", xlab="peak length")
# hist(f2$V3-f2$V2, breaks = 50,main="Large CG-DMRs", xlab="peak length")
# dev.off()
```
**TORUS marginal enrichment results**
```{r echo=F}
df<-read.table("data/torus_enrichment_DMR.est", header = F)
colnames(df)<-c("term", "estimate", "low", "high")
library(ggplot2)
ggplot(df, aes(y=reorder(term, estimate), x=log2(exp(estimate)))) +
       geom_errorbar(aes(xmax=log2(exp(high)), xmin=log2(exp(low))),width=0.2)+
       geom_point(size=1.5) + xlab("SCZ variant enrichment(log2)") + 
       theme_bw() + ylab("") + 
       scale_y_discrete(labels=c("iPSC", "iN_Glut", "NPC", "iN_Dopa", "iN_GABA", 
                                 "CG_DMRs","large CG_DMRs", "CDTS_5%","CDTS_1%"))  
```

**TORUS joint enrichment results**
```{r echo=F}
df<-read.table("data/torus_joint_enrichment.est", header = T)
library(ggplot2)
ggplot(df[-1,], aes(y=reorder(term, estimate), x=log2(exp(estimate)))) +
       geom_errorbar(aes(xmax=log2(exp(high)), xmin=log2(exp(low))),width=0.2)+
       geom_point(size=1.5) + xlab("SCZ variant enrichment(log2)") + 
       theme_bw() + ylab("") + 
       scale_y_discrete(labels=c("iPSC", "iN_Dopa","NPC", "iN_Glut", "iN_GABA", 
                                 "CG_DMRs","CDTS_5%")) +
       geom_vline(xintercept=0, linetype="dashed", color="grey")

```

## Joint enrichment analyses across traits

**Joint enrichment results**
```{r echo=F, fig.width=8, fig.height=6}
library(ggplot2)
library(ggpubr)

outDIR<-"/home/jinggu/cluster/projects/finemap_pipeline/torus_output"
neg_controls<-c("bmi", "height", "ibd_IBD", "ibd_UC", "ibd_CD", "HDL", "LDL")
test<-"scz_2018"

outputList<-list()
for(i in c(neg_controls, test)){
  outputList[[i]]<-read.table(paste(outDIR, sprintf("%s/joint_refined/torus_joint_enrichment.est", i),sep="/"), header = T)
  outputList[[i]]<-outputList[[i]][-1,]
  outputList[[i]]$term<-unlist(lapply(outputList[[i]]$term, function(i){strsplit(i, "[.]")[[1]][1]}))
  outputList[[i]]<-cbind(rep(i, dim(outputList[[i]])[1]), outputList[[i]])
}
outputDF<-data.frame(do.call(rbind, outputList), row.names = NULL)
colnames(outputDF)[1]<-"trait"

### This function will generate a plot listing the panels of enrichment estimates from multiple traits. 
ref <- outputDF[outputDF$trait=="scz_2018",]
ref_order<-reorder(ref$term, ref$estimate)
outputDF$term<-factor(outputDF$term, levels(ref_order))
plot_enrichment<-function(traits, log_based){
  if(log_based==T){
    ggplot(outputDF[outputDF$trait %in% traits,],
         aes(y=term, x=log2(exp(estimate)))) +
    geom_errorbar(aes(xmax=log2(exp(high)), xmin=log2(exp(low))),width=0.1) +   
    geom_point(size=2) + xlab("variant enrichment(log2)") +
    theme_bw() + ylab("") +
    geom_vline(xintercept=0, linetype="dashed", color="grey") + facet_grid(.~ trait)
  }else{
    ggplot(outputDF[outputDF$trait %in% traits & outputDF$estimate>0,],
         aes(y=term, x=log2(estimate))) +
    geom_errorbar(aes(xmax=log2(high), xmin=log2(low)),width=0.1) +   
    geom_point(size=2) + xlab("variant enrichment(log2)") +
    theme_bw() + ylab("") +
    geom_vline(xintercept=0, linetype="dashed", color="grey") + facet_grid(.~ trait)
  }
}

plot_enrichment(c("scz_2018", "HDL", "LDL"), log_based = T)
#ggsave("~/projects/funcFinemapping/output/scz_lipid_joint_enrichment.png", width = 10, height=6)
plot_enrichment(c("scz_2018", "bmi", "height"), log_based = T)
#ggsave("~/projects/funcFinemapping/output/scz_bmi_joint_enrichment.png", width = 8, height=6)
plot_enrichment(c("scz_2018", "ibd_IBD", "ibd_UC", "ibd_CD"), log_based = T)
#ggsave("~/projects/funcFinemapping/output/scz_immune_joint_enrichment.png", width = 8, height=6)
```
## Stratified-LSDC results  
### **Procedures**   
1. Built the annotation sets by combining our own refined annotations with the baseline_LD annotations from S-LDSC.
2. Computed LD scores for this customized annotation set
3. Run S-LDSC using the ldsc.py code from the PolyFun. 
4. The functional enrichment estimates were based on all SNPs, instead of only commmon SNPs. 
5. Efficient sample size needs to be considered. (TO DO)

### **Schizophrenia** 
```{r echo=F}
outDIR<-"/home/jinggu/cluster/projects/finemap_pipeline/s_ldsc/output"
df<-read.table(paste(outDIR, "enrichment_scz_2018.results", sep="/"), header = T)
df[,"high"]<-df$Enrichment+2*df$Enrichment_std_error
df[,"low"]<-df$Enrichment-2*df$Enrichment_std_error
ucsc<-df[grepl("ucsc_", tolower(df$Category)),]
encode<-df[grepl("promoter_ucsc_0", tolower(df$Category)),]
peaks<-df[grepl("peaks", tolower(df$Category)),]
kb<-df[grepl("10kb", tolower(df$Category)),]
tbl<-rbind(df[1:6,], ucsc, encode, peaks, kb)
ggplot(tbl[tbl$low >0,], aes(y=reorder(Category, Enrichment), x=log2(Enrichment))) + 
       geom_errorbar(aes(xmax=log2(high), xmin=log2(low)),width=0.2) + 
       geom_point(size=1.5) + xlab("SCZ variant enrichment(log2)") + theme_bw() + 
       ylab("") + geom_vline(xintercept=0, linetype="dashed", color="grey")


```
By performing S-LDSC analysis, we see DNA methylation and CDTS annotations shows 6-8 fold enrichment with SCZ variants conditional on all other annotations. 

### **Across traits** 
```{r echo=F}
neg_controls<-c("bmi", "height", "ibd_IBD", "ibd_UC", "ibd_CD", "HDL", "LDL")
test<-"scz_2018"
outputList<-list()
for(i in c(neg_controls, test)){
  df<-read.table(paste(outDIR, sprintf("enrichment_%s.results", i),sep="/"), header = T)
  df[,"high"]<-df$Enrichment+2*df$Enrichment_std_error
  df[,"low"]<-df$Enrichment-2*df$Enrichment_std_error
  outputList[[i]]<-df[1:6,]
  outputList[[i]]<-cbind(rep(i, dim(outputList[[i]])[1]), outputList[[i]])
}
outputDF<-data.frame(do.call(rbind, outputList), row.names = NULL)
colnames(outputDF)[c(1, 2, 6)]<-c("trait", "term", "estimate")
```
**Notes**

* Annotations with no estimates mean they have negative values.
* If the lower end of 95% confidence interval of an estimate is negative, it will be assigned as 0.05 (-4.3 as log2 based).
```{r echo=F, fig.width=8, fig.height=6}
outputDF[outputDF$low<0,]$low <- 0.05
ref <- outputDF[outputDF$trait=="scz_2018",]
ref_order<-reorder(ref$term, ref$estimate)
outputDF$term<-factor(outputDF$term, levels(ref_order))

plot_enrichment(c("scz_2018", "HDL", "LDL"), log_based = F)
plot_enrichment(c("scz_2018", "bmi", "height"), log_based = F)
plot_enrichment(c("scz_2018", "ibd_IBD", "ibd_UC", "ibd_CD"), log_based = F)
```
Overall, we see Torus and S-LDSC generates similar enrichment estimates, except for ibd_UC trait. There seems to be some discrepancy in the estimates, which needs a closer look. We so see the performance of annotations varies across traits. 

* CDTS at top 5% bins that locate in non-coding regions can better distinguish SCZ from lipid-related trait, but no from bmi, height or immune related traits.
* Human CpG methylations can distinguish SCZ from both immune related traits and lipid-related traits.
* M6A can better distinguish SCZ from immune related traits.
* SCZ variants are not enriched in protein coding regions, while variants associated with all other traits are enriched in protein coding regions. 








