---
title: "SNP_finemapping_results"
output:
  html_document:
    df_print: paged
---

## Assess the impact of different priors on fine mapping
```{r message=FALSE}
setwd("~/projects/funcFinemapping/")
library(data.table)
library(dplyr)
library(susieR)
library(bigsnpr)
library(ggplot2)
source("code/make_plots.R")
source("code/run_susie.R")
```
### SpliceAI-predicted variant effects on splicing
**Annotations**

* spliceAI predictions are not tissue-specific.
* The predicted delta score of a variant can be interpreted as the probrability of the variant being splice-altering. 

**Procedure**

* Build binary annotation by denoting delta score>=0.01 as 1 and detal score < 0.01 as 0.
* Run TORUS jointly over spliceAI-predictions, iPSC-derived neuronal accessibility, CADD top5% and  protein coding regions.
* Use the derived prior probability for each SNP to further perform fine mapping with SuSiE. We used 10% of UK Biobank samples to build LD reference matrix, which was further integrated with GWAS summary statsistics to compute for PIPs at different assumptions of number of causal variants per LD block. 
**Torus enrichment estimates**
```{r}
#est<-read.table("output/splicing/torus_enrichment_joint_scz_mtsplice0.6_hypothalamus-brain.est", header = T)
getwd()
est<-read.table("output/splicing/", header = F, skip=1)
colnames(est)<-c("term", "estimate", "low", "high")
snp_enrichment_plot(est[-1,], y.label = c("top5% CADD", "CDS", "iPSC-derived", "spliceAI_0.01"))
```
```{r echo=F}
### load data ###
uniform.L1<-read.table("output/splicing/scz_unif_ukb_susieL1.txt.gz", header = T)
annot.L1<-read.table("output/splicing/scz_spliceAI0.01_ukb_susieL1.txt.gz", header = T)

uniform.L10<-read.table("output/splicing/scz_unif_ukb_susieL10.txt.gz", header = T)
annot.L10<-read.table("output/splicing/scz_spliceAI0.01_ukb_susieL10.txt.gz", header = T)

#load("~/cluster/projects/fine_mapping/scz_2018/scz_spliceAI0.01_ukb_differentLs.out.RData")
```
**Note**: ".cov" means a lower coverage 
```{r echo=F, eval=F}
work_dir<-"~/cluster/projects/torus_enrichment/scz_2018/joint"
gwas<-read.table("~/cluster/data/gwas/2_Filtered/scz_2018.txt.gz", header=T)
scz_loci <- rtracklayer::import("~/cluster/data/gwas/gwas_loci/SCZ_108loci_hg19.tsv", format='bed')
priors<-read.table(paste(work_dir, "spliceAI/torus_joint_prior_est.txt.gz", sep="/"), header=T)
gwas<-inner_join(gwas, priors, by="snp")
#focus on the 104 independent loci from previous paper
LDRanges<-make_ranges(paste0('chr', LD_Blocks$X1), LD_Blocks$X2, LD_Blocks$X3)
LDRanges <- plyranges::mutate(LDRanges, chunk=LD_Blocks$X4)
overlaps<-IRanges::subsetByOverlaps(LDRanges, scz_loci, type="any")

susie.L2<-list()
#susie_L10.cov<-list()
for(z in 1:length(overlaps$chunk)){
  print(paste0('Finemapping chunks.. ',z,' of ', length(overlaps$chunk)))
  z<-as.character(overlaps$chunk[z])
  print(z)
  susie.L2[[z]]<-run_susie(gwas, z, L=2)
}
joint.annot<-read.table("/home/jinggu/cluster/projects/torus_enrichment/scz_2018/joint/spliceAI/cutoff0.01/torus_annotations.txt.gz",header = T)
colnames(joint.annot)<-
joint.annot["all_d"]<-apply(joint.annot[,-1], 1, sum)

uniform.L2<-data.frame(do.call(rbind, lapply(susie_L2, function(i){return(i$unif)})))
uniform.L2<-left_join(uniform.L2, joint.annot, by="snp")
annot.L2<-data.frame(do.call(rbind, lapply(susie_L2, function(i){return(i$prior)})))
annot.L2<-left_join(annot.L2, joint.annot, by="snp")

save(susie.L2,
     file="~/cluster/projects/fine_mapping/scz_2018/scz_spliceAI0.03_ukb_differentLs.out.RData")


#uniform.L1<-merge_susie_sumstats(susie_L1,gwas[gwas$locus %in% overlaps$chunk,])
cols<-c(c("snp", "beta", "se", "pval", "zscore", "locus", "X_NUM_ID_", 
          "torus_pip", "cs", "cs_purity","cs_size", "susie_pip"),colnames(joint.annot)[-1])
uniform.L1<-read.table("~/cluster/projects/fine_mapping/scz_2018/scz_uniform_susieL1.txt.gz", header=T)
annot.L1<-read.table("~/cluster/projects/fine_mapping/scz_2018/scz_spliceAI0.01_ukb_susieL1.txt.gz")
#annot.L1<-merge_susie_sumstats(susie_L1,gwas[gwas$locus %in% overlaps$chunk,])
write.table(annot.L2[,cols],"~/cluster/projects/fine_mapping/scz_2018/scz_spliceAI0.03_susieL2.txt.gz", quote=F, row.names = F, sep="\t")
write.table(uniform.L2[,cols],"~/cluster/projects/fine_mapping/scz_2018/scz_unif_ukb_susieL2.txt.gz",quote=F, row.names = F, sep="\t")

```

#### Results Summary
**Plot p-values against susie PIPs**
```{r echo=F,fig.width=12, fig.height=6}
par(mfrow = c(1,2))
plot(y=uniform.L1$susie_pip, x=-log10(uniform.L1$pval), ylab="PIPs with uniform priors", xlab="-log10(pval)", main="L=1")
plot(y=annot.L1$susie_pip, x=-log10(annot.L1$pval), ylab="PIPs with functional priors", xlab="-log10(pval)")

par(mfrow = c(1,2))
plot(y=uniform.L10$susie_pip, x=-log10(uniform.L10$pval), ylab="PIPs with uniform priors", xlab="-log10(pval)", main="L=10")
plot(y=annot.L10$susie_pip, x=-log10(annot.L10$pval), ylab="PIPs with functional priors", xlab="-log10(pval)")

```
* For an initial QC, we did not observe SNPs with high PIPs but show no significant associations for both scenarios under the assumption of one causal variant per locus. The trend was the similar for L=2, but we did see a small fractions non-significant SNPs with elevated PIPs for the run with functional prior. At L=10 the distribution of PIPs against -log10 p-value shifted to the left, implying a substantial fraction of SNPs with non-sigiciant p-values show higher PIPs under functional priors. 

NOTE: The following analyses were based on L=1. 


**Compare PIPs between uniform and functional priors**
```{r echo=F, fig.width=12, fig.height=6}
par(mfrow = c(1,2))
plot(x=uniform.L1$susie_pip, 
     y=annot.L1$susie_pip,
     main="L=1",
     xlab="PIPs from uniform priors", ylab="PIPs from functional priors"
     )
abline(a=0, b=1, col='red')

plot(x=uniform.L10$susie_pip, 
     y=annot.L10$susie_pip,
     main="L=10",
     xlab="PIPs from uniform priors", ylab="PIPs from functional priors"
     )
abline(a=0, b=1, col='red')
```
* For L=1, we observed most variants have similar low PIPs with or without functional priors. However some variants show higher PIPs with functional prior while some show higher PIPs without functional priors. 
* For L=2, we observed some improvements using functional priors as more SNPs deviate above the diagonal line than below it.


**Compare the sizes of credible sets**

* Uniform priors vs. Functional priors
```{r}
plot_cs_size(uniform.L1) 
plot_cs_size(annot.L1)
```
* The numbers the stacked barplot represent the numbers of independent LD blocks within each range of credible sizes.
* Overall, the credible sizes are similar between the runs with or without priors 


**Distributions of spliceAI scores**
```{r}
###load spliceAI scores###
spliceAI.dir<-"/home/jinggu/cluster/data/ldsc/spliceAI"
scores<-c()
for(i in 1:22){
  f<-fread(sprintf("%s/spliceAI_varPred.%s.annot.gz", spliceAI.dir, i))
  scores<-rbind(scores, f)
}
annot.L1["SNP"]<-unlist(lapply(annot.L1$snp, function(i){strsplit(i, ":")[[1]][5]}))
annot.L1<-left_join(annot.L1, scores[, c("SNP", "spliceAI_varPred")], by="SNP")
annot.L10["spliceAI_varPred"]<-annot.L1$spliceAI_varPred
head(annot.L1[order(annot.L1$spliceAI_varPred, decreasing = T),])
```

```{r}
### compare PIPs between uniform priors and functional priors
# L=1
compare.tbl<-data.frame(annot.L1$susie_pip, uniform.L1$susie_pip, annot.L1$spliceAI_varPred)
colnames(compare.tbl)<-c("annot_pip", "uniform_pip", "spliceAI_varPred")

ggplot(data = compare.tbl[compare.tbl$spliceAI_varPred>0.01,], aes(y= annot_pip, x=uniform_pip, color=spliceAI_varPred)) + 
  geom_point() + scale_color_gradient(low="light grey", high="red") + theme_minimal() + ggtitle("L=1")

```
```{r}
#L=10
compare.tbl<-data.frame(annot.L10$susie_pip, uniform.L10$susie_pip, annot.L1$spliceAI_varPred)
colnames(compare.tbl)<-c("annot_pip", "uniform_pip", "spliceAI_varPred")

ggplot(data = compare.tbl[compare.tbl$spliceAI_varPred>0.05,], aes(y= annot_pip, x=uniform_pip, color=spliceAI_varPred)) + 
  geom_point() + scale_color_gradient(low="light grey", high="red") + theme_minimal() + ggtitle("L=10")

```
*** Examine the cases with higher PIP from functional priors and high spliceAI scores ***
```{r}
output<-annot.L10[annot.L10$susie_pip>0.2 & annot.L10$spliceAI_varPred > 0.3,]
output[,c("chr", "pos")]<-t(sapply(output$snp, function(i){as.numeric(strsplit(i, ":")[[1]][1:2])}))
tbl<-assign_genomic_annots(output)
```

```{r}
tbl
```




### Examine the snps within credible set of size less than 20 variants

**Examine a few loci**

```{r echo=F, fig.width=6, fig.height=7}
susie_check_plot(uniform.L10, "uniform prior-locus:313")
susie_check_plot(annot.L10, "functional prior-locus:313")
target.locus<-annot.L10[annot.L10$locus==313,]
target.locus[order(target.locus$susie_pip, decreasing = T),]

```

```{r echo=F}
unif.top20<-uniform.L1[uniform.L2$cs>0 & uniform.L1$cs_size<30,]
annot.top20<-annot.L1[annot.L2$cs>0 & annot.L1$cs_size<30,]
commons<-intersect(x=unif.top20$locus, y=annot.top20$locus)
unif.diff<-setdiff(unif.top20$locus, commons)
annot.diff<-setdiff(annot.top20$locus, commons)

annot.sub<-annot.L1[annot.L1$locus %in% annot.diff,]
annot.sub[order(annot.sub$susie_pip, decreasing = T),]
target.df<-annot.L1[annot.L1$spliceAI0.01_d>0,]
target.df[order(target.df$susie_pip, decreasing = T),]
```
```{r echo=F, fig.width=6, fig.height=7}
susie_check_plot(uniform.L1, "uniform prior-locus:1254")
susie_check_plot(annot.L1, "functional prior-locus:1254")
target.locus<-annot.L1[annot.L1$locus==1254,]
target.locus[order(target.locus$susie_pip, decreasing = T),]

```

```{r echo=F, fig.width=6, fig.height=7}
susie_check_plot(uniform.L1, "uniform prior-locus:27")
susie_check_plot(annot.L1, "functional prior-locus:27")
target.locus<-annot.L1[annot.L1$locus==27,]
target.locus[order(target.locus$susie_pip, decreasing = T),]

```


* Locus 251: 
    - top snp(2:200715388:G:T:rs2949006|zscore=8.42|spliceAI=0.01|iPSC_derived=1|pip=0.93|cs_size=2)
    - Compared with the other SNP in credible set, rs2949006 has a PIP higher than 0.9 due to being annotated with having splice-altering effects. 
    - found in GWAS catalog
    - not a GTEx splicing QTL
```{r echo=F, fig.width=6, fig.height=7}
susie_check_plot(uniform.L1, "uniform prior-locus:251")
susie_check_plot(annot.L1, "functional prior-locus:251")
target.locus<-annot.L1[annot.L1$locus==251,]
target.locus[order(target.locus$susie_pip, decreasing = T),]
```
* Locus 6: 
    - top snp(1:8423510:G:A:rs10779702|zscore=5.7|spliceAI=0.02|iPSC_derived=1|pip=0.4|cs_size=19)
    - not found in GWAS catalog
    - sQTL for skin and skeletal muscle
```{r echo=F, fig.width=6, fig.height=7}
susie_check_plot(uniform.L1, "uniform prior-locus:6")
susie_check_plot(annot.L1, "functional prior-locus:6")
target.locus<-annot.L1[annot.L1$locus==6,]
target.locus[order(target.locus$susie_pip, decreasing = T),]
```


### Annotate fine-mapped results
```{r echo=F, eval=F}
assign_genomic_annots<-function(sumstats){
      annot<-c()
      finemap.gr<-make_ranges(seqname=paste0("chr", sumstats$chr), start=sumstats$pos, end=sumstats$pos)
      snps.overlap.annots <- suppressWarnings(
        GRangesList(lapply(genomic.annots, function(x) { 
          overlaps<-plyranges::join_overlap_inner(y = x, x = finemap.gr) 
  })))
      for(c in c("exons", "promoters", "introns", "UTRs", "splice_junctions")){
        curr <- snps.overlap.annots[[c]]%>% as_tibble()
        curr <-plyranges::mutate(curr, category=c)
        annot<-rbind(annot, curr[,c("seqnames", "start", "category", "gene_name")])
      }
      
      colnames(annot)[2]<-"pos"
      sumstats[,"seqnames"]<-paste0("chr", sumstats$chr)
      output<-inner_join(sumstats, annot, by=c("seqnames", "pos"))
      return(output)
}
```
### Troubleshoot Torus's inflation esimation
```{r echo=F, eval=F}
wdir<-"~/cluster/projects/torus_enrichment/asthma_adult/Ulirsch2019"
gwas<-read.table("~/cluster/data/gwas/2_Filtered/asthma_adult.txt.gz", header=T)
est<-read.table(paste0(wdir, "/GMP_merge_compare/torus_enrichment.est"), header=F)
priors<-read.table(paste0(wdir,"/GMP_merge_compare/torus_snp_priors.txt.gz"), header=T)
gwas<-inner_join(gwas, priors, by="snp")
susie_L1<-list()
uniform.L1<-list()
for(z in unique(gwas$locus)){
  print(paste0('Finemapping chunks.. ',z,' of ', length(unique(gwas$locus))))
  susie.df<-gwas[gwas$locus==z,]
  uniform.L1[[z]]<-run.susie(susie.df, bigSNP, z, L=1, prior = FALSE)
  #susie_L1[[z]]<-run.susie(susie.df, bigSNP, z, L=1, prior = FALSE)
}
uniform.L1<-merge_susie_sumstats(uniform.L1, gwas)
#annot.L1<-merge_susie_sumstats(susie_L1, gwas)
snpRanges<-make_ranges(seqname=gwas$chr,
                       start = gwas$pos,
                       end = gwas$pos)
snpRanges<-plyranges::mutate(snpRanges, snp=gwas$snp)
dataPath<-"/home/jinggu/cluster/data/features/formatted/Ulirsch2019/compare/"
fnames<-c("blood", "common", "lung")
i = 1
for (f in list.files(dataPath, pattern = "*merge")){
  curr <- rtracklayer::import(paste(dataPath, f, sep='/'), format='bed')
  subdf <- IRanges::subsetByOverlaps(snpRanges, curr)
  #gwas <- dplyr::mutate(gwas, !!fnames[i] := ifelse(snp %in% unique(subdf$snp),1,0))
  #annot_sub<-annot.L1[annot.L1$snp %in% unique(subdf$snp),]
  annot_sub<-uniform.L1[uniform.L1$snp %in% unique(subdf$snp),]
  prop<-round(sum(annot_sub$susie_pip)/length(unique(annot_sub$locus))*100, 1)
  print(sprintf("Percent of genetic signal captured by %s is %s", f, paste0(prop, "%")))
}

```

